# –ü—Ä–∏–≤–µ–¥–µ–Ω–∏–µ —Ç–∏–ø–æ–≤

## Boolean

### –ü—Ä–∏ –ø—Ä–µ–æ–±—Ä–∞–∑–æ–≤–∞–Ω–∏–∏ –≤ Boolean –Ω–µ–æ–±—Ö–æ–¥–∏–º–æ –∏—Å–ø–æ–ª—å–∑–æ–≤–∞—Ç—å —è–≤–Ω–æ–µ –ø—Ä–∏–≤–µ–¥–µ–Ω–∏–µ

**ü§ñ –ê–≤—Ç–æ–º–∞—Ç–∏–∑–∞—Ü–∏—è**

–ù–µ –∏–º–ø–ª–µ–º–µ–Ω—Ç–∏—Ä–æ–≤–∞–Ω–æ –≤ eslint-config.

**‚ú® –ú–æ—Ç–∏–≤–∞—Ü–∏—è**

–î–æ—Å—Ç–∏–∂–µ–Ω–∏–µ –ø—Ä–æ–∑—Ä–∞—á–Ω–æ—Å—Ç–∏ –≤ –ø—Ä–∏–≤–µ–¥–µ–Ω–∏–∏ —Ç–∏–ø–æ–≤ –∏ –æ–¥–Ω–æ—Ä–æ–¥–Ω–æ—Å—Ç–∏ –∫–æ–¥–∞.

**‚úÖ Valid**

```ts
const string = 'string';

const isShow = Boolean(string);
```

**‚ùå Invalid**

```ts
const string = 'string';

const isShow = !!string;
```

#### –ü—Ä–µ–æ–±—Ä–∞–∑–æ–≤—ã–≤–∞—Ç—å –≤ Boolean –≤ —É—Å–ª–æ–≤–Ω—ã—Ö –æ–ø–µ—Ä–∞—Ç–æ—Ä–∞—Ö –Ω–µ –Ω–∞–¥–æ

**‚úÖ Valid**

```ts
const string = 'string';

...

if (string) {}

...

const viewString = string ? '–ù–æ–≤–æ–µ' : '–°—Ç–∞—Ä–æ–µ';
```

**‚ùå Invalid**

```ts
const string = 'string';

...

if (Boolean(string)) {}

...

const viewString = Boolean(string) ? '–ù–æ–≤–æ–µ' : '–°—Ç–∞—Ä–æ–µ';
```

## Number

### –ü—Ä–∏ –ø—Ä–µ–æ–±—Ä–∞–∑–æ–≤–∞–Ω–∏–∏ –≤ Number –Ω–µ–æ–±—Ö–æ–¥–∏–º–æ –∏—Å–ø–æ–ª—å–∑–æ–≤–∞—Ç—å —è–≤–Ω–æ–µ –ø—Ä–∏–≤–µ–¥–µ–Ω–∏–µ

**ü§ñ –ê–≤—Ç–æ–º–∞—Ç–∏–∑–∞—Ü–∏—è**

–ù–µ –∏–º–ø–ª–µ–º–µ–Ω—Ç–∏—Ä–æ–≤–∞–Ω–æ –≤ eslint-config.

**‚ú® –ú–æ—Ç–∏–≤–∞—Ü–∏—è**

–î–æ—Å—Ç–∏–∂–µ–Ω–∏–µ –ø—Ä–æ–∑—Ä–∞—á–Ω–æ—Å—Ç–∏ –≤ –ø—Ä–∏–≤–µ–¥–µ–Ω–∏–∏ —Ç–∏–ø–æ–≤ –∏ –æ–¥–Ω–æ—Ä–æ–¥–Ω–æ—Å—Ç–∏ –∫–æ–¥–∞.

**‚úÖ Valid**

```ts
const string = '22';

const number = Number(string);
const intNumber = parseInt(string);
const foatNumber = parseFloat(string);
```

**‚ùå Invalid**

```ts
const string = '22';

const number = +string;
```

```ts
const string = '22';

const number = 1 * string;
```

## String

### –ü—Ä–∏ –ø—Ä–µ–æ–±—Ä–∞–∑–æ–≤–∞–Ω–∏–∏ –≤ String –Ω–µ–æ–±—Ö–æ–¥–∏–º–æ –∏—Å–ø–æ–ª—å–∑–æ–≤–∞—Ç—å —è–≤–Ω–æ–µ –ø—Ä–∏–≤–µ–¥–µ–Ω–∏–µ

**ü§ñ –ê–≤—Ç–æ–º–∞—Ç–∏–∑–∞—Ü–∏—è**

–ù–µ –∏–º–ø–ª–µ–º–µ–Ω—Ç–∏—Ä–æ–≤–∞–Ω–æ –≤ eslint-config.

**‚ú® –ú–æ—Ç–∏–≤–∞—Ü–∏—è**

–î–æ—Å—Ç–∏–∂–µ–Ω–∏–µ –ø—Ä–æ–∑—Ä–∞—á–Ω–æ—Å—Ç–∏ –≤ –ø—Ä–∏–≤–µ–¥–µ–Ω–∏–∏ —Ç–∏–ø–æ–≤ –∏ –æ–¥–Ω–æ—Ä–æ–¥–Ω–æ—Å—Ç–∏ –∫–æ–¥–∞.

**‚úÖ Valid**

```ts
const number = 22;

const string = String(number);
```

**‚ùå Invalid**

```ts
const number = 22;

const string = '' + number;
```

### –î–ª—è —Å–∫–ª–µ–∏–≤–∞–Ω–∏—è —Å—Ç—Ä–æ–∫ –Ω–µ–æ–±—Ö–æ–¥–∏–º–æ –∏—Å–ø–æ–ª—å–∑–æ–≤–∞—Ç—å —Å—Ç—Ä–æ–∫–æ–≤—ã–π —à–∞–±–ª–æ–Ω

**ü§ñ –ê–≤—Ç–æ–º–∞—Ç–∏–∑–∞—Ü–∏—è**

–ù–µ –∏–º–ø–ª–µ–º–µ–Ω—Ç–∏—Ä–æ–≤–∞–Ω–æ –≤ eslint-config.

**‚ú® –ú–æ—Ç–∏–≤–∞—Ü–∏—è**

–û–¥–Ω–æ–∑–Ω–∞—á–Ω–æ—Å—Ç—å –ø—Ä–æ–≤–æ–¥–∏–º–æ–π –æ–ø–µ—Ä–∞—Ü–∏–∏.

**‚úÖ Valid**

```ts
const str1 = 'str1';
const str2 = 'str2';

const string = `${str1}${str2}`;
```

**‚ùå Invalid**

```ts
const str1 = 'str1';
const str2 = 'str2';

const string = str1 + str2;
```

#### –î–ª—è —Å–∫–ª–µ–∏–≤–∞–Ω–∏—è —Å—Ç—Ä–æ–∫–∏ –∏ Number –Ω–µ–æ–±—Ö–æ–¥–∏–º–æ –∏—Å–ø–æ–ª—å–∑–æ–≤–∞—Ç—å —Å—Ç—Ä–æ–∫–æ–≤—ã–π —à–∞–±–ª–æ–Ω

**‚úÖ Valid**

```ts
const str = 'str';
const number = 22;

const string = `${str} - ${number}`;
```

**‚ùå Invalid**

```ts
const str = 'str';
const number = 22;

const string = str + ' - ' + number;
```
