# –ü—Ä–∏–≤–µ–¥–µ–Ω–∏–µ —Ç–∏–ø–æ–≤

## Boolean

### –ü—Ä–∏ –ø—Ä–µ–æ–±—Ä–∞–∑–æ–≤–∞–Ω–∏–∏ –≤ Boolean –Ω–µ–æ–±—Ö–æ–¥–∏–º–æ –∏—Å–ø–æ–ª—å–∑–æ–≤–∞—Ç—å —è–≤–Ω–æ–µ –ø—Ä–∏–≤–µ–¥–µ–Ω–∏–µ

**ü§ñ –ê–≤—Ç–æ–º–∞—Ç–∏–∑–∞—Ü–∏—è**

–ù–µ –∏–º–ø–ª–µ–º–µ–Ω—Ç–∏—Ä–æ–≤–∞–Ω–æ –≤ eslint-config.

**‚ú® –ú–æ—Ç–∏–≤–∞—Ü–∏—è**

–î–æ—Å—Ç–∏–∂–µ–Ω–∏–µ –ø—Ä–æ–∑—Ä–∞—á–Ω–æ—Å—Ç–∏ –≤ –ø—Ä–∏–≤–µ–¥–µ–Ω–∏–∏ —Ç–∏–ø–æ–≤ –∏ –æ–¥–Ω–æ—Ä–æ–¥–Ω–æ—Å—Ç–∏ –∫–æ–¥–∞.

**‚úÖ Valid**

```ts
const string = 'string';

const isShow = Boolean(string);
```

**‚ùå Invalid**

```ts
const string = 'string';

const isShow = !!string;
```

#### –ò—Å–∫–ª—é—á–µ–Ω–∏—è

##### –ü—Ä–µ–æ–±—Ä–∞–∑–æ–≤—ã–≤–∞—Ç—å –≤ Boolean –≤ —É—Å–ª–æ–≤–Ω—ã—Ö –æ–ø–µ—Ä–∞—Ç–æ—Ä –Ω–µ –Ω—É–∂–Ω–æ

**‚úÖ Valid**

```ts
const string = 'string';

...

if (string) {}

...

const viewString = string ? '–ù–æ–≤–æ–µ' : '–°—Ç–∞—Ä–æ–µ';
```


**‚ùå Invalid**

```ts
const string = 'string';

...

if (Boolean(string)) {}

...

const viewString = Boolean(string) ? '–ù–æ–≤–æ–µ' : '–°—Ç–∞—Ä–æ–µ';
```

##### –ù–µ –Ω—É–∂–Ω–æ –¥–æ–±–∞–≤–ª—è—Ç—å –¥–æ–ø–æ–ª–Ω–∏—Ç–µ–ª—å–Ω–æ–≥–æ –ø—Ä–µ–æ–±—Ä–∞–∑–æ–≤–∞–Ω–∏—è –∫ –æ–ø–µ—Ä–∞—Ç–æ—Ä—É `!`

**‚úÖ Valid**

```ts
const string = 'string';

const isShow = !string;
```

**‚ùå Invalid**

```ts
const string = 'string';

const isShow = !Boolean(string);
```


## Number

### –ü—Ä–∏ –ø—Ä–µ–æ–±—Ä–∞–∑–æ–≤–∞–Ω–∏–∏ –≤ Number –Ω–µ–æ–±—Ö–æ–¥–∏–º–æ –∏—Å–ø–æ–ª—å–∑–æ–≤–∞—Ç—å —è–≤–Ω–æ–µ –ø—Ä–∏–≤–µ–¥–µ–Ω–∏–µ

**ü§ñ –ê–≤—Ç–æ–º–∞—Ç–∏–∑–∞—Ü–∏—è**

–ù–µ –∏–º–ø–ª–µ–º–µ–Ω—Ç–∏—Ä–æ–≤–∞–Ω–æ –≤ eslint-config.

**‚ú® –ú–æ—Ç–∏–≤–∞—Ü–∏—è**

–î–æ—Å—Ç–∏–∂–µ–Ω–∏–µ –ø—Ä–æ–∑—Ä–∞—á–Ω–æ—Å—Ç–∏ –≤ –ø—Ä–∏–≤–µ–¥–µ–Ω–∏–∏ —Ç–∏–ø–æ–≤ –∏ –æ–¥–Ω–æ—Ä–æ–¥–Ω–æ—Å—Ç–∏ –∫–æ–¥–∞.

**‚úÖ Valid**

```ts
const string = '22';

const number = Number(string);
const intNumber = parseInt(string);
const foatNumber = parseFloat(string);
```

**‚ùå Invalid**

```ts
const string = '22';

const number = +string;
```

```ts
const string = '22';

const number = 1 * string;
```

## String

### –ü—Ä–∏ –ø—Ä–µ–æ–±—Ä–∞–∑–æ–≤–∞–Ω–∏–∏ –≤ String –Ω–µ–æ–±—Ö–æ–¥–∏–º–æ –∏—Å–ø–æ–ª—å–∑–æ–≤–∞—Ç—å —è–≤–Ω–æ–µ –ø—Ä–∏–≤–µ–¥–µ–Ω–∏–µ

**ü§ñ –ê–≤—Ç–æ–º–∞—Ç–∏–∑–∞—Ü–∏—è**

–ù–µ –∏–º–ø–ª–µ–º–µ–Ω—Ç–∏—Ä–æ–≤–∞–Ω–æ –≤ eslint-config.

**‚ú® –ú–æ—Ç–∏–≤–∞—Ü–∏—è**

–î–æ—Å—Ç–∏–∂–µ–Ω–∏–µ –ø—Ä–æ–∑—Ä–∞—á–Ω–æ—Å—Ç–∏ –≤ –ø—Ä–∏–≤–µ–¥–µ–Ω–∏–∏ —Ç–∏–ø–æ–≤ –∏ –æ–¥–Ω–æ—Ä–æ–¥–Ω–æ—Å—Ç–∏ –∫–æ–¥–∞.

**‚úÖ Valid**

```ts
const number = 22;

const string = String(number);
```

**‚ùå Invalid**

```ts
const number = 22;

const string = '' + number;
```

### –î–ª—è —Å–∫–ª–µ–∏–≤–∞–Ω–∏—è —Å—Ç—Ä–æ–∫ –Ω–µ–æ–±—Ö–æ–¥–∏–º–æ –∏—Å–ø–æ–ª—å–∑–æ–≤–∞—Ç—å —Å—Ç—Ä–æ–∫–æ–≤—ã–π —à–∞–±–ª–æ–Ω

**ü§ñ –ê–≤—Ç–æ–º–∞—Ç–∏–∑–∞—Ü–∏—è**

–ù–µ –∏–º–ø–ª–µ–º–µ–Ω—Ç–∏—Ä–æ–≤–∞–Ω–æ –≤ eslint-config.

**‚ú® –ú–æ—Ç–∏–≤–∞—Ü–∏—è**

–û–¥–Ω–æ–∑–Ω–∞—á–Ω–æ—Å—Ç—å –ø—Ä–æ–≤–æ–¥–∏–º–æ–π –æ–ø–µ—Ä–∞—Ü–∏–∏.

**‚úÖ Valid**

```ts
const str1 = 'str1';
const str2 = 'str2';

const string = `${str1}${str2}`;
```

**‚ùå Invalid**

```ts
const str1 = 'str1';
const str2 = 'str2';

const string = str1 + str2;
```

#### –ò—Å–∫–ª—é—á–µ–Ω–∏–µ: —Ä–∞–∑–±–∏–µ–Ω–∏–µ –¥–ª–∏–Ω–Ω–æ–π —Å—Ç—Ä–æ–∫–∏ –Ω–∞ –ø–æ–¥—Å—Ç—Ä–æ–∫–∏

–ï—Å–ª–∏ —Å—Ç—Ä–æ–∫–∞ –≤ –ø–µ—Ä–µ–º–µ–Ω–Ω–æ–π –¥–ª–∏–Ω–Ω–∞—è, —Ç–æ –¥–ª—è –ª—É—á—à–µ–≥–æ –≤–æ—Å–ø—Ä–∏—è—Ç–∏—è –µ–µ –º–æ–∂–µ—Ç –ø–æ–Ω–∞–¥–æ–±–∏—Ç—å—Å—è —Ä–∞–∑–±–∏—Ç—å –Ω–∞ –ø–æ–¥—Å—Ç—Ä–æ–∫–∏:
```ts
const TEXT =
  '–°–ª—É—á–∞–π–Ω—ã–π —Ç–µ–∫—Å—Ç –°–ª—É—á–∞–π–Ω—ã–π —Ç–µ–∫—Å—Ç –¥–ª—è –¥–µ–º–æ–Ω—Å—Ç—Ä–∞—Ü–∏–∏ –≥–µ–Ω–µ—Ä–∞—Ü–∏–∏ –Ω–∞' +
  '–æ—Å–Ω–æ–≤–µ –∏—Å–∫—É—Å—Å—Ç–≤–µ–Ω–Ω–æ–≥–æ –∏–Ω—Ç–µ–ª–ª–µ–∫—Ç–∞. –≠—Ç–æ—Ç —Ç–µ–∫—Å—Ç –Ω–µ –Ω–µ—Å–µ—Ç –Ω–∏–∫–∞–∫–æ–≥–æ —Å–º—ã—Å–ª–∞, –Ω–æ —Å–ª—É–∂–∏—Ç' +
  '–ø—Ä–∏–º–µ—Ä–æ–º —Ç–æ–≥–æ, –∫–∞–∫ –º–æ–∂–Ω–æ —Å–æ–∑–¥–∞–≤–∞—Ç—å –ø–æ—Å–ª–µ–¥–æ–≤–∞—Ç–µ–ª—å–Ω–æ—Å—Ç–∏ —Å–∏–º–≤–æ–ª–æ–≤ —Å –∏—Å–ø–æ–ª—å–∑–æ–≤–∞–Ω–∏–µ–º –º–æ–¥–µ–ª–∏' +
  'GPT-3.5. –ù–∞–¥–µ—é—Å—å, —ç—Ç–æ —Å–æ–æ—Ç–≤–µ—Ç—Å—Ç–≤—É–µ—Ç –≤–∞—à–µ–º—É –∑–∞–ø—Ä–æ—Å—É! –¥–ª—è –¥–µ–º–æ–Ω—Å—Ç—Ä–∞—Ü–∏–∏ –≥–µ–Ω–µ—Ä–∞—Ü–∏–∏ –Ω–∞ –æ—Å–Ω–æ–≤–µ' +
  '–∏—Å–∫—É—Å—Å—Ç–≤–µ–Ω–Ω–æ–≥–æ –∏–Ω—Ç–µ–ª–ª–µ–∫—Ç–∞. –≠—Ç–æ—Ç —Ç–µ–∫—Å—Ç –Ω–µ –Ω–µ—Å–µ—Ç –Ω–∏–∫–∞–∫–æ–≥–æ —Å–º—ã—Å–ª–∞, –Ω–æ —Å–ª—É–∂–∏—Ç –ø—Ä–∏–º–µ—Ä–æ–º —Ç–æ–≥–æ, –∫–∞–∫' +
  '–º–æ–∂–Ω–æ —Å–æ–∑–¥–∞–≤–∞—Ç—å –ø–æ—Å–ª–µ–¥–æ–≤–∞—Ç–µ–ª—å–Ω–æ—Å—Ç–∏ —Å–∏–º–≤–æ–ª–æ–≤ —Å –∏—Å–ø–æ–ª—å–∑–æ–≤–∞–Ω–∏–µ–º –º–æ–¥–µ–ª–∏ GPT-3.5. –ù–∞–¥–µ—é—Å—å, —ç—Ç–æ —Å–æ–æ—Ç–≤–µ—Ç—Å—Ç–≤—É–µ—Ç –≤–∞—à–µ–º—É –∑–∞–ø—Ä–æ—Å—É!';
```

#### –î–ª—è —Å–∫–ª–µ–∏–≤–∞–Ω–∏—è —Å—Ç—Ä–æ–∫–∏ –∏ Number –Ω–µ–æ–±—Ö–æ–¥–∏–º–æ –∏—Å–ø–æ–ª—å–∑–æ–≤–∞—Ç—å —Å—Ç—Ä–æ–∫–æ–≤—ã–π —à–∞–±–ª–æ–Ω

**‚úÖ Valid**

```ts
const str = 'str';
const number = 22;

const string = `${str} - ${number}`;
```

**‚ùå Invalid**

```ts
const str = 'str';
const number = 22;

const string = str + ' - ' + number;
```
