# Ğ£ÑĞ»Ğ¾Ğ²Ğ½Ñ‹Ğµ Ğ¾Ğ¿ĞµÑ€Ğ°Ñ‚Ğ¾Ñ€Ñ‹

## Ğ—Ğ°Ğ¿Ñ€ĞµÑ‰ĞµĞ½Ğ¾ Ğ¸ÑĞ¿Ğ¾Ğ»ÑŒĞ·Ğ¾Ğ²Ğ°Ñ‚ÑŒ Ğ²Ğ»Ğ¾Ğ¶ĞµĞ½Ğ½Ñ‹Ğµ Ñ‚ĞµÑ€Ğ½Ğ°Ñ€Ğ½Ñ‹Ğµ Ğ¾Ğ¿ĞµÑ€Ğ°Ñ‚Ğ¾Ñ€Ñ‹

**ğŸ¤– ĞĞ²Ñ‚Ğ¾Ğ¼Ğ°Ñ‚Ğ¸Ğ·Ğ°Ñ†Ğ¸Ñ**

ĞĞµ Ğ¸Ğ¼Ğ¿Ğ»ĞµĞ¼ĞµĞ½Ñ‚Ğ¸Ñ€Ğ¾Ğ²Ğ°Ğ½Ğ¾ Ğ² eslint-config.

**âœ¨ ĞœĞ¾Ñ‚Ğ¸Ğ²Ğ°Ñ†Ğ¸Ñ**

Ğ’Ğ»Ğ¾Ğ¶ĞµĞ½Ğ½Ñ‹Ğµ Ñ‚ĞµÑ€Ğ½Ğ°Ñ€Ğ½Ñ‹Ğµ Ğ¾Ğ¿ĞµÑ€Ğ°Ñ‚Ğ¾Ñ€Ñ‹ Ğ·Ğ½Ğ°Ñ‡Ğ¸Ñ‚ĞµĞ»ÑŒĞ½Ğ¾ ÑƒÑĞ»Ğ¾Ğ¶Ğ½ÑÑÑ‚ Ğ²Ğ¾ÑĞ¿Ñ€Ğ¸ÑÑ‚Ğ¸Ğµ ĞºĞ¾Ğ´Ğ°.

**âŒ Invalid**

```ts
const getColor = (variant: Variant, darkMode: boolean) =>
  darkMode
    ? variant === 'primary'
      ? darkTheme.primary
      : darkTheme.secondary
    : variant === 'primary'
    ? lightTheme.primary
    : lightTheme.secondary;
```

```tsx
const Info = ({ list, type }: Props) => {
  return (
    <div>
      {list.length ? (
        type === 'auth' ? (
          <AuthInfo />
        ) : (
          <NoAuthInfo />
        )
      ) : (
        <NoData />
      )}
    </div>
  );
};
```

**âœ… Valid**

```ts
const getColor = (variant: Variant, darkMode: boolean) => {
  if (darkMode) {
    return variant === 'primary' ? darkTheme.primary : darkTheme.secondary;
  }

  return variant === 'primary' ? lightTheme.primary : lightTheme.secondary;
};
```

```tsx
const Info = ({ list, type }: Props) => {
  return (
    <div>
      {list.length && <NoData />}
      {!list.length && type === 'auth' ? <AuthInfo /> : <NoAuthInfo />}
    </div>
  );
};
```
