# Constants

## –ö–æ–Ω—Å—Ç–∞–Ω—Ç—ã –≤–Ω–µ –±–ª–æ—á–Ω–æ–π –æ–±–ª–∞—Å—Ç–∏ –≤–∏–¥–∏–º–æ—Å—Ç–∏ –¥–æ–ª–∂–Ω—ã –±—ã—Ç—å –≤ UPPER_CASE

**ü§ñ –ê–≤—Ç–æ–º–∞—Ç–∏–∑–∞—Ü–∏—è**

–ù–µ –∏–º–ø–ª–µ–º–µ–Ω—Ç–∏—Ä–æ–≤–∞–Ω–æ –≤ eslint-config.

**‚ú® –ú–æ—Ç–∏–≤–∞—Ü–∏—è**

–ò–¥–µ–Ω—Ç–∏—Ñ–∏–∫–∞—Ü–∏—è –≥–ª–æ–±–∞–ª—å–Ω–æ–π –∫–æ–Ω—Å—Ç–∞–Ω—Ç—ã.

**‚úÖ Valid**

```tsx
// –ö–æ–Ω—Å—Ç–∞–Ω—Ç–∞ –≤–Ω–µ –±–ª–æ—á–Ω–æ–π –æ–±–ª–∞—Å—Ç–∏ –≤–∏–¥–∏–º–æ—Å—Ç–∏
const DEFAULT_NAME = '–í–∞—Å—è';

const Info = ({ list }: Props) => {
  // –ö–æ–Ω—Å—Ç–∞–Ω—Ç–∞ –≤–Ω—É—Ç—Ä–∏ –±–ª–æ—á–Ω–æ–π –æ–±–ª–∞—Å—Ç–∏ –≤–∏–¥–∏–º–æ—Å—Ç–∏
  const isShowList = Boolean(list.length);

  return (
    <section>
      {isShowList && <List list={list} />}
      <span>{DEFAULT_NAME}</span>
    </section>
  );
};
```

```ts
// –ö–æ–Ω—Å—Ç–∞–Ω—Ç–∞ –≤–Ω–µ –±–ª–æ—á–Ω–æ–π –æ–±–ª–∞—Å—Ç–∏ –≤–∏–¥–∏–º–æ—Å—Ç–∏
const DEFAULT_FACTOR = 2;

const calc = (a: number, b: number, factor: number = DEFAULT_FACTOR) => {
  // –ö–æ–Ω—Å—Ç–∞–Ω—Ç–∞ –≤–Ω—É—Ç—Ä–∏ –±–ª–æ—á–Ω–æ–π –æ–±–ª–∞—Å—Ç–∏ –≤–∏–¥–∏–º–æ—Å—Ç–∏
  const summ = a + b;

  return summ * factor;
};
```

–ö–æ–Ω—Å—Ç–∞–Ω—Ç–∞ –≤ –æ—Ç–¥–µ–ª—å–Ω–æ–º —Ñ–∞–π–ª–µ:

```constants.ts```
```ts
export const API_URL = 'https://astral.ru';
```

**‚ùå Invalid**

```ts
// –ö–æ–Ω—Å—Ç–∞–Ω—Ç–∞ –≤–Ω–µ –±–ª–æ—á–Ω–æ–π –æ–±–ª–∞—Å—Ç–∏ –≤–∏–¥–∏–º–æ—Å—Ç–∏
const defaultFactor = 2;

const calc = (a: number, b: number, factor: number = defaultFactor) => {
  // –ö–æ–Ω—Å—Ç–∞–Ω—Ç–∞ –≤–Ω—É—Ç—Ä–∏ –±–ª–æ—á–Ω–æ–π –æ–±–ª–∞—Å—Ç–∏ –≤–∏–¥–∏–º–æ—Å—Ç–∏
  const summ = a + b;

  return summ * factor;
};
```

–ö–æ–Ω—Å—Ç–∞–Ω—Ç–∞ –≤ –æ—Ç–¥–µ–ª—å–Ω–æ–º —Ñ–∞–π–ª–µ:

```constants.ts```
```ts
export const apiUrl = 'https://astral.ru';
```
