# Import/Export ES Modules

## –ù–µ–æ–±—Ö–æ–¥–∏–º–æ –æ—Ç–¥–∞–≤–∞—Ç—å –ø—Ä–µ–¥–ø–æ—á—Ç–µ–Ω–∏–µ –∏–º–µ–Ω–æ–≤–∞–Ω–Ω—ã–º —ç–∫—Å–ø–æ—Ä—Ç–∞–º

–î–µ—Ñ–æ–ª—Ç–Ω—ã–π —ç–∫—Å–ø–æ—Ä—Ç –∏—Å–ø–æ–ª—å–∑—É–µ—Ç—Å—è —Ç–æ–ª—å–∫–æ –≤ —Ç–æ–º —Å–ª—É—á–∞–µ, –µ—Å–ª–∏ —ç—Ç–æ –Ω–µ–æ–±—Ö–æ–¥–∏–º–æ –ø–æ —Ç–µ—Ö–Ω–∏—á–µ—Å–∫–∏–º –ø—Ä–∏—á–∏–Ω–∞–º (–Ω–∞–ø—Ä–∏–º–µ—Ä, **react lazy**).

**ü§ñ –ê–≤—Ç–æ–º–∞—Ç–∏–∑–∞—Ü–∏—è**

–ù–µ –∏–º–ø–ª–µ–º–µ–Ω—Ç–∏—Ä–æ–≤–∞–Ω–æ –≤ eslint-config.

**‚ú® –ú–æ—Ç–∏–≤–∞—Ü–∏—è**

–î–µ—Ñ–æ–ª—Ç–Ω—ã–µ —ç–∫—Å–ø–æ—Ä—Ç—ã –∑–∞—Å—Ç–∞–≤–ª—è—é—Ç –ø—Ä–æ–∏–∑–≤–æ–¥–∏—Ç—å –¥–æ–ø–æ–ª–Ω–∏—Ç–µ–ª—å–Ω—ã–µ –∑–∞—Ç—Ä–∞—Ç—ã –Ω–∞ –Ω–µ–π–º–∏–Ω–≥ –ø—Ä–∏ –∫–∞–∂–¥–æ–º –∏–º–ø–æ—Ä—Ç–µ.

**‚úÖ Valid**

```ts
export const Header = () => {...};

export class CartStore {
  ...
};

export const API_URLS = { ... } 
```

```ts
import { Header } from './Header';
import { CartStore } from './store';
import { API_URLS } from './constants';
```

**‚ùå Invalid**

```ts
export default = () => {...};
```

```ts
class CartStore {
  ...
};

export default CartStore;
```

```ts
export default {
  main: '/main'
} 
```

```ts
import Header from './Header';
import CartStore from './store';
import API_URLS from './constants';
```


## `Export` —É–∫–∞–∑—ã–≤–∞–µ—Ç—Å—è —Å—Ä–∞–∑—É –¥–ª—è —ç–∫—Å–ø–æ—Ä—Ç–∏—Ä—É–µ–º–æ–π —Å—É—â–Ω–æ—Å—Ç–∏

**ü§ñ –ê–≤—Ç–æ–º–∞—Ç–∏–∑–∞—Ü–∏—è**

–ù–µ –∏–º–ø–ª–µ–º–µ–Ω—Ç–∏—Ä–æ–≤–∞–Ω–æ –≤ eslint-config.

**‚ú® –ú–æ—Ç–∏–≤–∞—Ü–∏—è**

–£–º–µ–Ω—å—à–µ–Ω–∏–µ –∫–æ–ª–∏—á–µ—Å—Ç–≤–∞ –¥–æ–ø–æ–ª–Ω–∏—Ç–µ–ª—å–Ω–æ–≥–æ –∫–æ–¥–∞, –Ω–µ –Ω–µ—Å—É—â–µ–≥–æ —Å–º—ã—Å–ª–æ–≤–æ–π –Ω–∞–≥—Ä—É–∑–∫–∏.

**‚úÖ Valid**

```ts
export const API_URL = {};

export class CartStore {};

export const UserCard = () => {};
```

**‚ùå Invalid**

```ts
const API_URL = {};

class CartStore {};

const UserCard = () => {};

export {
  API_URL,
  CartStore,
  UserCard
};
```

## –ò–º–ø–æ—Ä—Ç –∏–∑ –¥–∏—Ä–µ–∫—Ç–æ—Ä–∏–π –æ—Å—É—â–µ—Å—Ç–≤–ª—è–µ—Ç—Å—è –ø–æ—Å—Ä–µ–¥—Å—Ç–≤–æ–º —Ñ–∞–π–ª–∞ `index`

**ü§ñ –ê–≤—Ç–æ–º–∞—Ç–∏–∑–∞—Ü–∏—è**

–ù–µ –∏–º–ø–ª–µ–º–µ–Ω—Ç–∏—Ä–æ–≤–∞–Ω–æ –≤ eslint-config.

**‚ú® –ú–æ—Ç–∏–≤–∞—Ü–∏—è**

–ò–º–ø–æ—Ä—Ç –ø–æ—Å—Ä–µ–¥—Å—Ç–≤–æ–º `index` —Ñ–∞–π–ª–æ–≤ –ø–æ–∑–≤–æ–ª—è–µ—Ç –¥–æ—Å—Ç–∏—á—å —Ö–æ—Ä–æ—à–µ–≥–æ —É—Ä–æ–≤–Ω—è –∏–Ω–∫–∞–ø—Å—É–ª—è—Ü–∏–∏ –º–æ–¥—É–ª–µ–π –∏ –∫–æ–Ω—Ç—Ä–æ–ª–∏—Ä–æ–≤–∞—Ç—å –ø—É–±–ª–∏—á–Ω—ã–π –∏–Ω—Ç–µ—Ä—Ñ–µ–π—Å.
–í—Å–µ, —á—Ç–æ —ç–∫—Å–ø–æ—Ä—Ç–∏—Ä—É–µ—Ç—Å—è –∏–∑ `index` —Ñ–∞–π–ª–∞, —è–≤–ª—è–µ—Ç—Å—è –ø—É–±–ª–∏—á–Ω—ã–º–∏ —Å—É—â–Ω–æ—Å—Ç—è–º–∏, —Å –∫–æ—Ç–æ—Ä—ã–º–∏ –º–æ–∂–Ω–æ –≤–∑–∞–∏–º–æ–¥–µ–π—Å—Ç–≤–æ–≤–∞—Ç—å –∏–∑ –≤–Ω–µ.
–í—Å–µ, —á—Ç–æ –Ω–µ —ç–∫—Å–ø–æ—Ä—Ç–∏—Ä—É–µ—Ç—Å—è –∏–∑ `index` —Ñ–∞–π–ª–∞, —è–≤–ª—è–µ—Ç—Å—è –ø—Ä–∏–≤–∞—Ç–Ω—ã–º–∏ —Å—É—â–Ω–æ—Å—Ç—è–º–∏, —Å –Ω–∏–º–∏ –≤–∑–∞–∏–º–æ–¥–µ–π—Å—Ç–≤–æ–≤–∞—Ç—å –∏–∑ –≤–Ω–µ –Ω–µ–ª—å–∑—è.

**‚úÖ Valid**

```
‚îú‚îÄ‚îÄ features/
|    ‚îú‚îÄ‚îÄ UserCard/ 
|    ‚îú‚îÄ‚îÄ Profile/
|    ‚îî‚îÄ‚îÄ index.ts
```

```features/index.ts```
```ts
export { UserCard, type UserCardProps } from './UserCard';
export { Profile } from './Profile';
```

```ts
import { UserCard } from './features';

...
```

---

```
‚îú‚îÄ‚îÄ UserService/
|    ‚îú‚îÄ‚îÄ utils/ 
|    ‚îú‚îÄ‚îÄ constants.ts
|    ‚îú‚îÄ‚îÄ enums.ts
|    ‚îú‚îÄ‚îÄ types.ts
|    ‚îú‚îÄ‚îÄ UserService.ts
|    ‚îú‚îÄ‚îÄ UserService.test.ts
|    ‚îî‚îÄ‚îÄ index.ts
```

```UserService/index.ts```
```ts
export { UserService } from './UserService';
export { type UserInfo } from './types';
```

```ts
import { type UserInfo } from './UserService';

...
```

**‚ùå Invalid**

–û—Ç—Å—É—Ç—Å—Ç–≤—É–µ—Ç `index.ts`
```
‚îú‚îÄ‚îÄ UserService/
|    ‚îú‚îÄ‚îÄ utils/ 
|    ‚îú‚îÄ‚îÄ constants.ts
|    ‚îú‚îÄ‚îÄ enums.ts
|    ‚îú‚îÄ‚îÄ types.ts
|    ‚îú‚îÄ‚îÄ UserService.ts
|    ‚îî‚îÄ‚îÄ UserService.test.ts
```

---

–ò–º–ø–æ—Ä—Ç "–ª–µ–∑–µ—Ç" –≤–æ –≤–Ω—É—Ç—Ä–µ–Ω–Ω—é—é —Å—Ç—Ä—É–∫—Ç—É—Ä—É –º–æ–¥—É–ª—è 
```ts
import { UserCard } from './featues/UserCard/UserCard';
import { type UserInfo } from './UserService/types';

...
```

## –ò–º–ø–æ—Ä—Ç—ã –≤–∏–¥–∞ `../` –∑–∞–ø—Ä–µ—â–µ–Ω—ã

–ò–º–ø–æ—Ä—Ç—ã, –Ω–µ —É–∫–∞–∑—ã–≤–∞—é—â–∏–µ –Ω–∞ –∫–æ–Ω–∫—Ä–µ—Ç–Ω—ã–π es module, –∑–∞–ø—Ä–µ—â–µ–Ω—ã.

**ü§ñ –ê–≤—Ç–æ–º–∞—Ç–∏–∑–∞—Ü–∏—è**

–ù–µ –∏–º–ø–ª–µ–º–µ–Ω—Ç–∏—Ä–æ–≤–∞–Ω–æ –≤ eslint-config.

**‚ú® –ú–æ—Ç–∏–≤–∞—Ü–∏—è**

–ò–º–ø–æ—Ä—Ç—ã –≤–∏–¥–∞ `../` —Å–æ–∑–¥–∞—é—Ç —Ü–∏–∫–ª–∏—á–µ—Å–∫–∏–µ –∑–∞–≤–∏—Å–∏–º–æ—Å—Ç–∏.

**‚úÖ Valid**

```
‚îú‚îÄ‚îÄ CartService/
‚îú‚îÄ‚îÄ UserService/
‚îî‚îÄ‚îÄ index.ts
```

```index.ts```
```ts
export * from './CartService';
export * from './UserService';
```

```CartService/CartService.ts```

```ts
import { UserService } from '../UserService';

class CartService {
  constructor(private readonly userService: UserService) {}
}
```

**‚ùå Invalid**

```
‚îú‚îÄ‚îÄ CartService/
‚îú‚îÄ‚îÄ UserService/
‚îî‚îÄ‚îÄ index.ts
```

```index.ts```
```ts
export * from './CartService';
export * from './UserService';
```

```CartService/CartService.ts```
```ts
// –ò–∑-–∑–∞ —Ç–æ–≥–æ, —á—Ç–æ –∏–º–ø–æ—Ä—Ç –∏–¥–µ—Ç —á–µ—Ä–µ–∑ index.ts —Ñ–∞–π–ª –º–æ–∂–µ—Ç –æ–±—Ä–∞–∑–æ–≤–∞—Ç—å—Å—è —Ü–∏–∫–ª–∏—á–µ—Å–∫–∞—è –∑–∞–≤–∏—Å–∏–º–æ—Å—Ç—å
import { UserService } from '../';

class CartService {
  constructor(private readonly userService: UserService) {}
}
```
