# Stores

## ĞšĞ°Ğ¶Ğ´Ñ‹Ğ¹ store Ğ´Ğ¾Ğ»Ğ¶ĞµĞ½ Ğ½Ğ°Ñ…Ğ¾Ğ´Ğ¸Ñ‚ÑŒÑÑ Ğ¾Ñ‚Ğ´ĞµĞ»ÑŒĞ½Ğ¾Ğ¹ Ğ´Ğ¸Ñ€ĞµĞºÑ‚Ğ¾Ñ€Ğ¸Ğ¸ Ğ¸ Ğ¸Ğ¼ĞµÑ‚ÑŒ Ğ¿Ğ¾ÑÑ‚Ñ„Ğ¸ĞºÑ `Store`

**âœ¨ ĞœĞ¾Ñ‚Ğ¸Ğ²Ğ°Ñ†Ğ¸Ñ**

ĞÑ‚Ğ´ĞµĞ»ÑŒĞ½Ğ°Ñ Ğ´Ğ¸Ñ€ĞµĞºÑ‚Ğ¾Ñ€Ğ¸Ñ Ğ´Ğ»Ñ ĞºĞ°Ğ¶Ğ´Ğ¾Ğ³Ğ¾ store Ğ¿Ğ¾Ğ·Ğ²Ğ¾Ğ»ÑĞµÑ‚ Ğ±ĞµÑĞ¿Ñ€ĞµĞ¿ÑÑ‚ÑÑ‚Ğ²ĞµĞ½Ğ½Ğ¾ Ñ€Ğ°ÑÑˆĞ¸Ñ€ÑÑ‚ÑŒ store Ğ¸ ÑƒÑÑ‚Ğ°Ğ½Ğ°Ğ²Ğ»Ğ¸Ğ²Ğ°Ñ‚ÑŒ Ğ¾Ğ´Ğ½Ğ¾Ğ·Ğ½Ğ°Ñ‡Ğ½Ñ‹Ğµ ÑĞ²ÑĞ·Ğ¸ Ñ ĞµĞ³Ğ¾ Ğ²Ğ½ÑƒÑ‚Ñ€ĞµĞ½Ğ½Ğ¸Ğ¼Ğ¸ Ğ·Ğ°Ğ²Ğ¸ÑĞ¸Ğ¼Ğ¾ÑÑ‚ÑĞ¼Ğ¸.

**ğŸ¤– ĞĞ²Ñ‚Ğ¾Ğ¼Ğ°Ñ‚Ğ¸Ğ·Ğ°Ñ†Ğ¸Ñ**

ĞĞµ Ğ¸Ğ¼Ğ¿Ğ»ĞµĞ¼ĞµĞ½Ñ‚Ğ¸Ñ€Ğ¾Ğ²Ğ°Ğ½Ğ¾ Ğ² eslint-config.

**âœ… Valid**

```
â”œâ”€â”€ domain/
|    â”œâ”€â”€ stores/
|    |    |â”€â”€ PaymentStore/
|    |    |    |â”€â”€ utils/
|    |    |    |â”€â”€ PaymentStore.ts
|    |    |    |â”€â”€ useUserData.test.ts
|    |    |    |â”€â”€ constants.ts
|    |    |    |â”€â”€ types.ts
|    |    |    â””â”€â”€ index.ts
|    |    â””â”€â”€ index.ts
|    â””â”€â”€ index.ts
```

```domain/stores/PaymentStore/PaymentStore.ts```
```ts
export class PaymentStore {}
```

**âŒ Invalid**

```
â”œâ”€â”€ domain/
|    â”œâ”€â”€ stores/
|    |    |â”€â”€ PaymentStore.ts
|    |    â””â”€â”€ index.ts
|    â””â”€â”€ index.ts
```

## Mobx Stores

Mobx Stores Ğ¿Ğ¾Ğ´Ñ‡Ğ¸Ğ½ÑÑÑ‚ÑÑ [ĞµĞ´Ğ¸Ğ½Ñ‹Ğ¼ Ğ¿Ñ€Ğ°Ğ²Ğ¸Ğ»Ğ°Ğ¼ js classes](./classes).

