# Ğ Ğ°Ğ±Ğ¾Ñ‚Ğ° Ñ TODO Ğ² ĞºĞ¾Ğ´Ğµ

## TODO Ğ² ĞºĞ¾Ğ´Ğµ Ğ´Ğ¾Ğ»Ğ¶ĞµĞ½ ÑĞ¾Ğ´ĞµÑ€Ğ¶Ğ°Ñ‚ÑŒ ÑÑÑ‹Ğ»ĞºÑƒ Ğ½Ğ° Ñ‚ĞµÑ…. Ğ´Ğ¾Ğ»Ğ³Ğ¾Ğ²ÑƒÑ Ğ·Ğ°Ğ´Ğ°Ñ‡Ñƒ

Ğ”Ğ»Ñ ĞºĞ°Ğ¶Ğ´Ğ¾Ğ³Ğ¾ TODO Ğ´Ğ¾Ğ»Ğ¶Ğ½Ğ° Ğ±Ñ‹Ñ‚ÑŒ ÑĞ¾Ğ·Ğ´Ğ°Ğ½Ğ° Ñ‚ĞµÑ…. Ğ´Ğ¾Ğ»Ğ³Ğ¾Ğ²Ğ°Ñ Ğ·Ğ°Ğ´Ğ°Ñ‡Ğ° Ğ² Jira Ğ¿Ğ¾ [ÑˆĞ°Ğ±Ğ»Ğ¾Ğ½Ñƒ](https://track.astral.ru/soft/wiki/pages/viewpage.action?pageId=3849335550).

**ğŸ¤– ĞĞ²Ñ‚Ğ¾Ğ¼Ğ°Ñ‚Ğ¸Ğ·Ğ°Ñ†Ğ¸Ñ**

ĞĞµ Ğ¸Ğ¼Ğ¿Ğ»ĞµĞ¼ĞµĞ½Ñ‚Ğ¸Ñ€Ğ¾Ğ²Ğ°Ğ½Ğ¾ Ğ² eslint-config.

**âœ¨ ĞœĞ¾Ñ‚Ğ¸Ğ²Ğ°Ñ†Ğ¸Ñ**

- Ğ¡Ğ¾Ğ·Ğ´Ğ°Ğ½Ğ¸Ğµ Ğ·Ğ°Ğ´Ğ°Ñ‡Ğ¸ Ğ¿Ğ¾Ğ´ TODO Ğ¿Ğ¾Ğ·Ğ²Ğ¾Ğ»ÑĞµÑ‚ Ğ¿Ğ»Ğ°Ğ½Ğ¸Ñ€Ğ¾Ğ²Ğ°Ñ‚ÑŒ Ğ·Ğ°ĞºÑ€Ñ‹Ñ‚Ğ¸Ğµ Ñ‚ĞµÑ…. Ğ´Ğ¾Ğ»Ğ³Ğ°
- Ğ’Ğ¾Ğ·Ğ¼Ğ¾Ğ¶Ğ½Ğ¾ÑÑ‚ÑŒ Ğ±Ñ‹ÑÑ‚Ñ€Ğ¾Ğ³Ğ¾ Ğ¿ĞµÑ€ĞµÑ…Ğ¾Ğ´Ğ° Ğ¸Ğ· Ğ±ÑĞºĞ»Ğ¾Ğ³Ğ° Ñ‚ĞµÑ…. Ğ´Ğ¾Ğ»Ğ³Ğ° Ğ² Ğ±Ğ»Ğ¾Ğº ĞºĞ¾Ğ´Ğ°, ĞºĞ¾Ñ‚Ğ¾Ñ€Ñ‹Ğ¹ Ñ‚Ñ€ĞµĞ±ÑƒĞµÑ‚ Ñ€ĞµÑ„Ğ°ĞºÑ‚Ğ¾Ñ€Ğ¸Ğ½Ğ³Ğ°.

**âœ… Valid**

```ts
// TODO: Ğ¸Ğ·Ğ±Ğ°Ğ²Ğ¸Ñ‚ÑŒÑÑ Ğ¾Ñ‚ any. https://track.astral.ru/soft/browse/EXAMPLE-000
const calc = (params: any) => params.startDate + params.endDate;
```

**âŒ Invalid**

```ts
// TODO: Ğ¸Ğ·Ğ±Ğ°Ğ²Ğ¸Ñ‚ÑŒÑÑ Ğ¾Ñ‚ any
const calc = (params: any) => params.startDate + params.endDate;
```
