# Typescript

## –ù–µ–æ–±—Ö–æ–¥–∏–º–æ –æ—Ç–¥–∞–≤–∞—Ç—å –ø—Ä–µ–¥–ø–æ—á—Ç–µ–Ω–∏–µ `type` –ø–µ—Ä–µ–¥ `interface`

`interface` –Ω–µ–æ–±—Ö–æ–¥–∏–º–æ –∏—Å–ø–æ–ª—å–∑–æ–≤–∞—Ç—å —Ç–æ–ª—å–∫–æ –≤ —Å–ª—É—á–∞—è—Ö, –∫–æ–≥–¥–∞ –µ–≥–æ –∏—Å–ø–æ–ª—å–∑–æ–≤–∞–Ω–∏–µ –Ω–µ–æ–±—Ö–æ–¥–∏–º–æ: –¥–ª—è –∫–ª–∞—Å—Å–æ–≤, —Ä–∞—Å—à–∏—Ä–µ–Ω–∏–µ `interface` —Å—Ç–æ—Ä–æ–Ω–Ω–∏—Ö –±–∏–±–ª–∏–æ—Ç–µ–∫.

**ü§ñ –ê–≤—Ç–æ–º–∞—Ç–∏–∑–∞—Ü–∏—è**

–ù–µ –∏–º–ø–ª–µ–º–µ–Ω—Ç–∏—Ä–æ–≤–∞–Ω–æ –≤ eslint-config.

**‚ú® –ú–æ—Ç–∏–≤–∞—Ü–∏—è**

–û–¥–Ω–æ—Ä–æ–¥–Ω–æ—Å—Ç—å ts-–∫–æ–¥–∞.

**‚úÖ Valid**

```tsx
type Props = {
  title: string;
  userName: string;
  onClick: () => void;
};

export const UserInfo = ({ title, userName, onClick }: Props) => { ... };
```

```ts
type UseLogicParams = {
  data: unknown[];
};

type UseLogicResult = {
    isShow: boolean;
};

const useLogic = (params: UseLogicParams): UseLogicResult => {};
```

```ts
export type UserDTO = {
  data: unknown[];
};
```

```ts
interface IUserService {
  name: string;
  logout: () => void;
}

class UserService implements IUserService {
  ...
}
```

**‚ùå Invalid**

```tsx
interface Props {
  title: string;
  userName: string;
  onClick: () => void;
};

export const UserInfo = ({ title, userName, onClick }: Props) => { ... };
```

```ts
interface UseLogicParams {
  data: unknown[];
};

interface UseLogicResult {
    isShow: boolean;
};

const useLogic = (params: UseLogicParams): UseLogicResult => {};
```

```ts
export interface UserDTO {
  data: unknown[];
};
```

## –ù–∞–∑–≤–∞–Ω–∏—è –ø–∞—Ä–∞–º–µ—Ç—Ä–æ–≤ generic –¥–æ–ª–∂–Ω—ã –∏–º–µ—Ç—å –ø—Ä–µ—Ñ–∏–∫—Å `T` –∏ —É—Ç–æ—á–Ω—è—é—â–µ–µ —Å–ª–æ–≤–æ

**ü§ñ –ê–≤—Ç–æ–º–∞—Ç–∏–∑–∞—Ü–∏—è**

–ù–µ –∏–º–ø–ª–µ–º–µ–Ω—Ç–∏—Ä–æ–≤–∞–Ω–æ –≤ eslint-config.

**‚ú® –ú–æ—Ç–∏–≤–∞—Ü–∏—è**

- –í–æ–∑–º–æ–∂–Ω–æ—Å—Ç—å –≤–∏–∑—É–∞–ª—å–Ω–æ –æ—Ç–ª–∏—á–∏–π –ø–∞—Ä–∞–º–µ—Ç—Ä–∞ generic –æ—Ç —Å–∞–º–æ—Å—Ç–æ—è—Ç–µ–ª—å–Ω–æ–≥–æ —Ç–∏–ø–∞
- –ü–æ–∑–≤–æ–ª—è–µ—Ç –∏–∑–±–µ–∂–∞—Ç—å –ø–µ—Ä–µ—Å–µ—á–µ–Ω–∏–π –∏–º–µ–Ω
- –£—Ç–æ—á–Ω—è—é—â–µ–µ —Å–ª–æ–≤–æ –ø–æ–∑–≤–æ–ª—è–µ—Ç –ø–æ–Ω—è—Ç—å –Ω–∞–∑–Ω–∞—á–µ–Ω–∏–µ –ø–∞—Ä–∞–º–µ—Ç—Ä–∞

**‚úÖ Valid**

```ts
const formatErrorToGlobal = <TError, TResultError>(error: TError): TResultError => {
  ...
}
```

**‚ùå Invalid**

```ts
// –ò–∑ –Ω–∞–∑–≤–∞–Ω–∏–π –ø–∞—Ä–∞–º–µ—Ç—Ä–æ–≤ generic –Ω–µ–≤–æ–∑–º–æ–∂–Ω–æ –ø–æ–Ω—è—Ç—å –∑–∞ —á—Ç–æ –æ—Ç–≤–µ—á–∞—é—Ç –ø–∞—Ä–∞–º–µ—Ç—Ä—ã
const formatErrorToGlobal = <T, D>(error: T): D => {
  ...
}
```

## –ù–∞–∑–≤–∞–Ω–∏—è enum –∏ –µ–≥–æ —Å–≤–æ–π—Å—Ç–≤ –¥–æ–ª–∂–Ω–æ –±—ã—Ç—å –≤ PascalCase

**ü§ñ –ê–≤—Ç–æ–º–∞—Ç–∏–∑–∞—Ü–∏—è**

–ù–µ –∏–º–ø–ª–µ–º–µ–Ω—Ç–∏—Ä–æ–≤–∞–Ω–æ –≤ eslint-config.

**‚ú® –ú–æ—Ç–∏–≤–∞—Ü–∏—è**

–û–¥–Ω–æ—Ä–æ–¥–Ω–æ—Å—Ç—å –∏–º–µ–Ω –¥–ª—è enums.

**‚úÖ Valid**

```ts
enum UserType {
  NoAuth,
  Auth
}
```

**‚ùå Invalid**

```ts
enum userType {
  noAuth,
  auth
}
```

## –ù–µ–æ–±—Ö–æ–¥–∏–º–æ –æ—Ç–¥–∞–≤–∞—Ç—å –ø—Ä–µ–¥–ø–æ—á—Ç–µ–Ω–∏–µ `Record` –ø–µ—Ä–µ–¥ –æ–±—ä–µ–∫—Ç–Ω–æ–π –Ω–æ—Ç–∞—Ü–∏–µ–π

```ts
type DataRecord = Record<string, unknown>;

type DataObject = {
  [key: string]: unknown;
}
```

–û–±—ä–µ–∫—Ç–Ω—É—é –Ω–æ—Ç–∞—Ü–∏—é –Ω–µ–æ–±—Ö–æ–¥–∏–º–æ –∏—Å–ø–æ–ª—å–∑–æ–≤–∞—Ç—å —Ç–æ–ª—å–∫–æ –ø–æ –Ω–µ–æ–±—Ö–æ–¥–∏–º–æ—Å—Ç–∏. –ù–∞–ø—Ä–∏–º–µ—Ä, –¥–ª—è **Mapped Types**.

**ü§ñ –ê–≤—Ç–æ–º–∞—Ç–∏–∑–∞—Ü–∏—è**

–ù–µ –∏–º–ø–ª–µ–º–µ–Ω—Ç–∏—Ä–æ–≤–∞–Ω–æ –≤ eslint-config.

**‚ú® –ú–æ—Ç–∏–≤–∞—Ü–∏—è**

–ò—Å–ø–æ–ª—å–∑–æ–≤–∞–Ω–∏–µ –±–æ–ª–µ–µ –∫—Ä–∞—Ç–∫–æ–≥–æ –æ–ø–∏—Å–∞–Ω–∏—è –æ–±—ä–µ–∫—Ç–æ–≤.

**‚úÖ Valid**

```ts
type Data = Record<string, unknown>;
```

Mapped Types:
```ts
type OptionsFlags<TFields> = {
  [Property in keyof TFields]: boolean;
};
```

**‚ùå Invalid**

```ts
type Data = {
  [key: string]: unknown;
};
```

## –ò—Å–ø–æ–ª—å–∑–æ–≤–∞–Ω–∏–µ `any` –∑–∞–ø—Ä–µ—â–µ–Ω–æ

**ü§ñ –ê–≤—Ç–æ–º–∞—Ç–∏–∑–∞—Ü–∏—è**

–ò–º–ø–ª–µ–º–µ–Ω—Ç–∏—Ä–æ–≤–∞–Ω–æ –≤:
- [@astral/eslint-config-react-ts](https://npmjs.com/package/@astral/eslint-config-react-ts)
- [@astral/eslint-config-node-ts](https://npmjs.com/package/@astral/eslint-config-node-ts)

**‚ú® –ú–æ—Ç–∏–≤–∞—Ü–∏—è**

`any` –æ—Ç–∫–ª—é—á–∞–µ—Ç —Ç–∏–ø–∏–∑–∞—Ü–∏—é. –ï—Å–ª–∏ –≤ –∫–æ–¥–µ –∏—Å–ø–æ–ª—å–∑—É–µ—Ç—Å—è `any`, –∑–Ω–∞—á–∏—Ç –∫–æ–¥ —Å–æ–¥–µ—Ä–∂–∏—Ç –æ—à–∏–±–∫—É.

### –ü—Ä–∏ –∏—Å–ø–æ–ª—å–∑–æ–≤–∞–Ω–∏–∏ `any` –Ω–µ–æ–±—Ö–æ–¥–∏–º–æ –æ—Å—Ç–∞–≤–ª—è—Ç—å –∫–æ–º–º–µ–Ω—Ç–∞—Ä–∏–π: –ø—Ä–∏—á–∏–Ω—É –∏—Å–ø–æ–ª—å–∑–æ–≤–∞–Ω–∏—è

**‚ú® –ú–æ—Ç–∏–≤–∞—Ü–∏—è**

–û–±–æ—Å–Ω–æ–≤–∞–Ω–∏–µ –æ—Ç–∫–ª—é—á–µ–Ω–∏—è —Ç–∏–ø–∏–∑–∞—Ü–∏–∏ –ø–æ–∑–≤–æ–ª–∏—Ç –ø–æ–Ω—è—Ç—å –ø—Ä–æ–±–ª–µ–º—É –∏ –∏—Å–ø—Ä–∞–≤–∏—Ç—å –æ—à–∏–±–∫—É –≤ –±—É–¥—É—â–µ–º.

**‚úÖ Valid**

```ts
// –¢–∏–ø—ã —Ç–µ–º –Ω–µ—Å–æ–≤–º–µ—Å—Ç–∏–º—ã, –Ω–æ –ø–æ —Ñ–∞–∫—Ç—É –∏–¥–µ–Ω—Ç–∏—á–Ω—ã
// eslint-disable-next-line @typescript-eslint/no-explicit-any
export const useTheme = (): Theme => useEmotionTheme() as any;
```

**‚ùå Invalid**

–ö–æ–º–º–µ–Ω—Ç–∞—Ä–∏–π –æ—Ç—Å—É—Ç—Å—Ç–≤—É–µ—Ç:

```ts
// eslint-disable-next-line @typescript-eslint/no-explicit-any
export const useTheme = (): Theme => useEmotionTheme() as any;
```

## –ó–∞–ø—Ä–µ—â–µ–Ω–æ –∏—Å–ø–æ–ª—å–∑–æ–≤–∞–Ω–∏–µ `@ts-ignore`

`@ts-ignore` –Ω–µ–æ–±—Ö–æ–¥–∏–º–æ –∏—Å–ø–æ–ª—å–∑–æ–≤–∞—Ç—å —Ç–æ–ª—å–∫–æ –ø—Ä–∏ –∫—Ä–∞–π–Ω–µ–π –Ω–µ–æ–±—Ö–æ–¥–∏–º–æ—Å—Ç–∏.

**ü§ñ –ê–≤—Ç–æ–º–∞—Ç–∏–∑–∞—Ü–∏—è**

–ù–µ –∏–º–ø–ª–µ–º–µ–Ω—Ç–∏—Ä–æ–≤–∞–Ω–æ –≤ eslint-config.

**‚ú® –ú–æ—Ç–∏–≤–∞—Ü–∏—è**

`@ts-ignore` –ø–æ–ª–Ω–æ—Å—Ç—å—é –æ—Ç–∫–ª—é—á–∞–µ—Ç —Ç–∏–ø–∏–∑–∞—Ü–∏—é.

### –†–µ–∫–æ–º–µ–Ω–¥—É–µ—Ç—Å—è –∏—Å–ø–æ–ª—å–∑–æ–≤–∞—Ç—å `any` –≤–º–µ—Å—Ç–æ `@ts-ignore`

**‚ú® –ú–æ—Ç–∏–≤–∞—Ü–∏—è**

`any` –æ—Ç–∫–ª—é—á–∞–µ—Ç –ø—Ä–æ–≤–µ—Ä–∫—É —Ç–æ–ª—å–∫–æ –¥–ª—è –∫–æ–Ω–∫—Ä–µ—Ç–Ω–æ–≥–æ —Ç–∏–ø–∞, –∞ `@ts-ignore` –≤—ã–∫–ª—é—á–∞–µ—Ç –ø—Ä–æ–≤–µ—Ä–∫—É –¥–ª—è –≤—Å–µ–≥–æ –±–ª–æ–∫–∞ –∫–æ–¥–∞.

**‚úÖ Valid**

```tsx
// LegacyComponent –Ω–∞–ø–∏—Å–∞–Ω –Ω–∞ js, –±–µ–∑ any –Ω–µ –æ–±–æ–π—Ç–∏—Å—å
// eslint-disable-next-line @typescript-eslint/no-explicit-any
<LegacyComponent data={data as any} onClick={handleClick} />
```

**‚ùå Invalid**

`@ts-ignore` –≤—ã–∫–ª—é—á–∏—Ç –ø—Ä–æ–≤–µ—Ä–∫—É –Ω–µ —Ç–æ–ª—å–∫–æ –¥–ª—è `data`, –Ω–æ –∏ –¥–ª—è `onClick`:

```tsx
// @ts-ignore
<LegacyComponent data={data as any} onClick={handleClick} />
```

### –ü—Ä–∏ –∏—Å–ø–æ–ª—å–∑–æ–≤–∞–Ω–∏–∏ `@ts-ignore` –Ω–µ–æ–±—Ö–æ–¥–∏–º–æ –æ—Å—Ç–∞–≤–ª—è—Ç—å –∫–æ–º–º–µ–Ω—Ç–∞—Ä–∏–π: –ø—Ä–∏—á–∏–Ω—É –∏—Å–ø–æ–ª—å–∑–æ–≤–∞–Ω–∏—è

**‚ú® –ú–æ—Ç–∏–≤–∞—Ü–∏—è**

–û–±–æ—Å–Ω–æ–≤–∞–Ω–∏–µ –æ—Ç–∫–ª—é—á–µ–Ω–∏—è —Ç–∏–ø–∏–∑–∞—Ü–∏–∏ –ø–æ–∑–≤–æ–ª–∏—Ç –ø–æ–Ω—è—Ç—å –ø—Ä–æ–±–ª–µ–º—É –∏ –∏—Å–ø—Ä–∞–≤–∏—Ç—å –æ—à–∏–±–∫—É –≤ –±—É–¥—É—â–µ–º.

**‚úÖ Valid**

```tsx
// LegacyComponent –Ω–∞–ø–∏—Å–∞–Ω –Ω–∞ js –∏ –ø—Ä–∏ —Ä–µ–Ω–¥–µ—Ä–µ –≤—ã–¥–∞–µ—Ç –æ—à–∏–±–∫—É
// @ts-ignore
<LegacyComponent data={data as any} onClick={handleClick} />
```

**‚ùå Invalid**

```tsx
// @ts-ignore
<LegacyComponent data={data as any} onClick={handleClick} />
```
