# Utils

## ÐšÐ°Ð¶Ð´Ð°Ñ ÑƒÑ‚Ð¸Ð»Ð¸Ñ‚Ð° Ð´Ð¾Ð»Ð¶Ð½Ð° Ð¸Ð¼ÐµÑ‚ÑŒ ÑÐ²Ð¾ÑŽ Ð´Ð¸Ñ€ÐµÐºÑ‚Ð¾Ñ€Ð¸ÑŽ

**âœ¨ ÐœÐ¾Ñ‚Ð¸Ð²Ð°Ñ†Ð¸Ñ**

ÐžÑ‚Ð´ÐµÐ»ÑŒÐ½Ð°Ñ Ð´Ð¸Ñ€ÐµÐºÑ‚Ð¾Ñ€Ð¸Ñ Ð´Ð»Ñ ÐºÐ°Ð¶Ð´Ð¾Ð¹ ÑƒÑ‚Ð¸Ð»Ð¸Ñ‚Ñ‹ Ð¿Ð¾Ð·Ð²Ð¾Ð»ÑÐµÑ‚:
- Ð£ÑÑ‚Ð°Ð½Ð¾Ð²Ð¸Ñ‚ÑŒ Ð¾Ð´Ð½Ð¾Ð·Ð½Ð°Ñ‡Ð½ÑƒÑŽ ÑÐ²ÑÐ·ÑŒ ÑƒÑ‚Ð¸Ð»Ð¸Ñ‚Ñ‹ Ñ ÐµÐµ Ñ‚ÐµÑÑ‚Ð°Ð¼Ð¸
- Ð”ÐµÐºÐ¾Ð¼Ð¿Ð¾Ð·Ð¸Ñ€Ð¾Ð²Ð°Ñ‚ÑŒ Ð¸ Ñ€Ð°ÑÑˆÐ¸Ñ€ÑÑ‚ÑŒ ÑƒÑ‚Ð¸Ð»Ð¸Ñ‚Ñ‹, Ð±ÐµÐ· Ð½ÐµÐ¾Ð±Ñ…Ð¾Ð´Ð¸Ð¼Ð¾ÑÑ‚Ð¸ Ñ€ÐµÑ„Ð°ÐºÑ‚Ð¾Ñ€Ð¸Ð½Ð³Ð° (Ð´Ð¾Ð±Ð°Ð²Ð»ÐµÐ½Ð¸Ðµ Ñ‚ÐµÑÑ‚Ð¾Ð², ÐºÐ¾Ð½ÑÑ‚Ð°Ð½Ñ‚...)
- ÐŸÐµÑ€ÐµÐ½Ð¾ÑÐ¸Ñ‚ÑŒ ÑƒÑ‚Ð¸Ð»Ð¸Ñ‚Ñ‹ ÑÐ¾ ÑÐ²ÑÐ·Ð°Ð½Ð½Ñ‹Ð¼Ð¸ ÑÑƒÑ‰Ð½Ð¾ÑÑ‚ÑÐ¼Ð¸ (ÐºÐ¾Ð½ÑÑ‚Ð°Ð½Ñ‚Ñ‹, Ñ‚ÐµÑÑ‚Ñ‹) Ð² Ð´Ñ€ÑƒÐ³Ð¸Ðµ ÑÐ»Ð¾Ð¸ Ð¿Ñ€Ð¸Ð»Ð¾Ð¶ÐµÐ½Ð¸Ñ Ð±ÐµÐ· Ð½ÐµÐ¾Ð±Ñ…Ð¾Ð´Ð¸Ð¼Ð¾ÑÑ‚Ð¸ Ñ€ÐµÑ„Ð°ÐºÑ‚Ð¾Ñ€Ð¸Ð½Ð³Ð°

**ðŸ¤– ÐÐ²Ñ‚Ð¾Ð¼Ð°Ñ‚Ð¸Ð·Ð°Ñ†Ð¸Ñ**

ÐÐµ Ð¸Ð¼Ð¿Ð»ÐµÐ¼ÐµÐ½Ñ‚Ð¸Ñ€Ð¾Ð²Ð°Ð½Ð¾ Ð² eslint-config.

**âœ… Valid**

```
â”œâ”€â”€ utils/
|    |â”€â”€ formatPriceToView/
|    |    |â”€â”€ formatPriceToView.ts
|    |    |â”€â”€ constants.ts
|    |    |â”€â”€ enums.ts
|    |    |â”€â”€ formatPriceToView.test.ts
|    |    â””â”€â”€ index.ts
|    |â”€â”€ formatDate/
|    â””â”€â”€ index.ts
```

**âŒ Invalid**

```
â”œâ”€â”€ utils/
|    |â”€â”€ formatPriceToView.ts
|    |â”€â”€ formatDate.ts
|    â””â”€â”€ index.ts
```

#### Ð˜ÑÐºÐ»ÑŽÑ‡ÐµÐ½Ð¸Ñ

ÐŸÑ€Ð¸Ð²Ð°Ñ‚Ð½Ñ‹Ðµ Ñ„ÑƒÐ½ÐºÑ†Ð¸Ð¸, ÑÐ²Ð»ÑÑŽÑ‰Ð¸ÐµÑÑ Ñ‡Ð°ÑÑ‚ÑŒÑŽ Ñ€ÐµÐ°Ð»Ð¸Ð·Ð°Ñ†Ð¸Ð¸ ÑÐºÑÐ¿Ð¾Ñ€Ñ‚Ð¸Ñ€ÑƒÐµÐ¼Ð¾Ð¹ Ñ„ÑƒÐ½ÐºÑ†Ð¸Ð¸, Ð¼Ð¾Ð³ÑƒÑ‚ Ð½Ð°Ñ…Ð¾Ð´Ð¸Ñ‚ÑŒÑÑ Ð² Ð¾Ð´Ð½Ð¾Ð¼ Ñ„Ð°Ð¹Ð»Ðµ.

```
const parseVersion = () => {}
...
export const compareVersion = () => {
    const v = parseVersion()
    ...
}
```