# Utils

## Каждая утилита должна иметь свою директорию

**✨ Мотивация**

Отдельная директория для каждой утилиты позволяет:
- Установить однозначную связь утилиты с ее тестами
- Декомпозировать и расширять утилиты, без необходимости рефакторинга (добавление тестов, констант...)
- Переносить утилиты со связанными сущностями (константы, тесты) в другие слои приложения без необходимости рефакторинга

**🤖 Автоматизация**

Не имплементировано в eslint-config.

**✅ Valid**

```
├── utils/
|    |── formatPriceToView/
|    |    |── formatPriceToView.ts
|    |    |── constants.ts
|    |    |── enums.ts
|    |    |── formatPriceToView.test.ts
|    |    └── index.ts
|    |── formatDate/
|    └── index.ts
```

**❌ Invalid**

```
├── utils/
|    |── formatPriceToView.ts
|    |── formatDate.ts
|    └── index.ts
```

#### Исключения

Приватные функции, являющиеся частью реализации экспортируемой функции, могут находиться в одном файле.

```
const parseVersion = () => {}
...
export const compareVersion = () => {
    const v = parseVersion()
    ...
}
```