# Utils

## –ö–∞–∂–¥–∞—è —É—Ç–∏–ª–∏—Ç–∞ –¥–æ–ª–∂–Ω–∞ –∏–º–µ—Ç—å —Å–≤–æ—é –¥–∏—Ä–µ–∫—Ç–æ—Ä–∏—é

**‚ú® –ú–æ—Ç–∏–≤–∞—Ü–∏—è**

–û—Ç–¥–µ–ª—å–Ω–∞—è –¥–∏—Ä–µ–∫—Ç–æ—Ä–∏—è –¥–ª—è –∫–∞–∂–¥–æ–π —É—Ç–∏–ª–∏—Ç—ã –ø–æ–∑–≤–æ–ª—è–µ—Ç:
- –£—Å—Ç–∞–Ω–æ–≤–∏—Ç—å –æ–¥–Ω–æ–∑–Ω–∞—á–Ω—É—é —Å–≤—è–∑—å —É—Ç–∏–ª–∏—Ç—ã —Å –µ–µ —Ç–µ—Å—Ç–∞–º–∏
- –î–µ–∫–æ–º–ø–æ–∑–∏—Ä–æ–≤–∞—Ç—å –∏ —Ä–∞—Å—à–∏—Ä—è—Ç—å —É—Ç–∏–ª–∏—Ç—ã, –±–µ–∑ –Ω–µ–æ–±—Ö–æ–¥–∏–º–æ—Å—Ç–∏ —Ä–µ—Ñ–∞–∫—Ç–æ—Ä–∏–Ω–≥–∞ (–¥–æ–±–∞–≤–ª–µ–Ω–∏–µ —Ç–µ—Å—Ç–æ–≤, –∫–æ–Ω—Å—Ç–∞–Ω—Ç...)
- –ü–µ—Ä–µ–Ω–æ—Å–∏—Ç—å —É—Ç–∏–ª–∏—Ç—ã —Å–æ —Å–≤—è–∑–∞–Ω–Ω—ã–º–∏ —Å—É—â–Ω–æ—Å—Ç—è–º–∏ (–∫–æ–Ω—Å—Ç–∞–Ω—Ç—ã, —Ç–µ—Å—Ç—ã) –≤ –¥—Ä—É–≥–∏–µ —Å–ª–æ–∏ –ø—Ä–∏–ª–æ–∂–µ–Ω–∏—è –±–µ–∑ –Ω–µ–æ–±—Ö–æ–¥–∏–º–æ—Å—Ç–∏ —Ä–µ—Ñ–∞–∫—Ç–æ—Ä–∏–Ω–≥–∞

**ü§ñ –ê–≤—Ç–æ–º–∞—Ç–∏–∑–∞—Ü–∏—è**

–ù–µ –∏–º–ø–ª–µ–º–µ–Ω—Ç–∏—Ä–æ–≤–∞–Ω–æ –≤ eslint-config.

**‚úÖ Valid**

```
‚îú‚îÄ‚îÄ utils/
|    |‚îÄ‚îÄ formatPriceToView/
|    |    |‚îÄ‚îÄ formatPriceToView.ts
|    |    |‚îÄ‚îÄ constants.ts
|    |    |‚îÄ‚îÄ enums.ts
|    |    |‚îÄ‚îÄ formatPriceToView.test.ts
|    |    ‚îî‚îÄ‚îÄ index.ts
|    |‚îÄ‚îÄ formatDate/
|    ‚îî‚îÄ‚îÄ index.ts
```

**‚ùå Invalid**

```
‚îú‚îÄ‚îÄ utils/
|    |‚îÄ‚îÄ formatPriceToView.ts
|    |‚îÄ‚îÄ formatDate.ts
|    ‚îî‚îÄ‚îÄ index.ts
```
