# Ð¤ÑƒÐ½ÐºÑ†Ð¸Ð¸ Ð¸ Ð¼ÐµÑ‚Ð¾Ð´Ñ‹

## Ð¤ÑƒÐ½ÐºÑ†Ð¸Ð¸ Ð¸ Ð¼ÐµÑ‚Ð¾Ð´Ñ‹ Ð´Ð¾Ð»Ð¶Ð½Ñ‹ Ð½Ð°Ñ‡Ð¸Ð½Ð°Ñ‚ÑŒÑÑ Ñ Ð³Ð»Ð°Ð³Ð¾Ð»Ð°

Ð˜ÑÐºÐ»ÑŽÑ‡ÐµÐ½Ð¸Ðµ: [Ð¤ÑƒÐ½ÐºÑ†Ð¸Ð¸ Ð¸ Ð¼ÐµÑ‚Ð¾Ð´Ñ‹, Ð²Ð¾Ð·Ð²Ñ€Ð°Ñ‰Ð°ÑŽÑ‰Ð¸Ðµ Boolean](#Ñ„ÑƒÐ½ÐºÑ†Ð¸Ð¸-Ð¸-Ð¼ÐµÑ‚Ð¾Ð´Ñ‹-Ð²Ð¾Ð·Ð²Ñ€Ð°Ñ‰Ð°ÑŽÑ‰Ð¸Ðµ-boolean-Ð´Ð¾Ð»Ð¶Ð½Ñ‹-ÑÐ¾Ð´ÐµÑ€Ð¶Ð°Ñ‚ÑŒ-Ð¿Ñ€ÐµÑ„Ð¸ÐºÑ-ÑƒÐºÐ°Ð·Ñ‹Ð²Ð°ÑŽÑ‰Ð¸Ð¹-Ð½Ð°-Ð¿Ñ€Ð¸Ð½Ð°Ð´Ð»ÐµÐ¶Ð½Ð¾ÑÑ‚ÑŒ-Ðº-boolean).

**âœ¨ ÐœÐ¾Ñ‚Ð¸Ð²Ð°Ñ†Ð¸Ñ**

Ð“Ð»Ð°Ð³Ð¾Ð» Ð² Ð½Ð°Ð·Ð²Ð°Ð½Ð¸Ð¸ Ñ„ÑƒÐ½ÐºÑ†Ð¸Ð¸ Ð¸Ð»Ð¸ Ð¼ÐµÑ‚Ð¾Ð´Ð° Ð¿Ð¾Ð·Ð²Ð¾Ð»ÑÐµÑ‚ Ð±Ñ‹ÑÑ‚Ñ€Ð¾ Ð¿Ð¾Ð½ÑÑ‚ÑŒ Ð²Ñ‹Ð¿Ð¾Ð»Ð½ÑÐµÐ¼Ð¾Ðµ Ð´ÐµÐ¹ÑÑ‚Ð²Ð¸Ðµ.

**ðŸ¤– ÐÐ²Ñ‚Ð¾Ð¼Ð°Ñ‚Ð¸Ð·Ð°Ñ†Ð¸Ñ**

ÐÐµ Ð¸Ð¼Ð¿Ð»ÐµÐ¼ÐµÐ½Ñ‚Ð¸Ñ€Ð¾Ð²Ð°Ð½Ð¾ Ð² eslint-config.

**âœ… Valid**

```ts
const calcSum = () => {};

const getUserData = () => {};

const removeID = () => {};

const refreshData = () => {};

const formatToView = () => {};

const createRequest = () => {};
```

```ts
class Example {
  public calcSum = () => {};

  public getUserData = () => {};

  public removeID = () => {};

  public refresh = () => {};

  public formatToView = () => {};
}
```

**âŒ Invalid**
```ts
// Ð“Ð»Ð°Ð³Ð¾Ð» Ð² ÐºÐ¾Ð½Ñ†Ðµ
const sumCalc = () => {};

// Ð¡ÑƒÐ´Ñ Ð¸Ð· Ð½Ð°Ð·Ð²Ð°Ð½Ð¸Ñ ÑÑ‚Ð¾ Ð½Ðµ Ñ„ÑƒÐ½ÐºÑ†Ð¸Ñ, Ð° Ð¾Ð±ÑŠÐµÐºÑ‚
const userDataGetter = () => {};

// Ð¡ÑƒÐ´Ñ Ð¸Ð· Ð½Ð°Ð·Ð²Ð°Ð½Ð¸Ñ ÑÑ‚Ð¾ Ð½Ðµ Ñ„ÑƒÐ½ÐºÑ†Ð¸Ñ, Ð° Ð¾Ð±ÑŠÐµÐºÑ‚
const toViewFormatter = () => {};

// ÐÐ°Ð·Ð²Ð°Ð½Ð¸Ðµ ÑƒÐºÐ°Ð·Ñ‹Ð²Ð°ÐµÑ‚, Ñ‡Ñ‚Ð¾ ÑÑ‚Ð¾ Ð¿Ñ€Ð¾Ñ†ÐµÑÑ, Ð° Ð½Ðµ Ñ„ÑƒÐ½ÐºÑ†Ð¸Ñ
const refreshingData = () => {};
```

## Ð¤ÑƒÐ½ÐºÑ†Ð¸Ð¸ Ð¸ Ð¼ÐµÑ‚Ð¾Ð´Ñ‹, Ð²Ð¾Ð·Ð²Ñ€Ð°Ñ‰Ð°ÑŽÑ‰Ð¸Ðµ Boolean Ð´Ð¾Ð»Ð¶Ð½Ñ‹ ÑÐ¾Ð´ÐµÑ€Ð¶Ð°Ñ‚ÑŒ Ð¿Ñ€ÐµÑ„Ð¸ÐºÑ, ÑƒÐºÐ°Ð·Ñ‹Ð²Ð°ÑŽÑ‰Ð¸Ð¹ Ð½Ð° Ð¿Ñ€Ð¸Ð½Ð°Ð´Ð»ÐµÐ¶Ð½Ð¾ÑÑ‚ÑŒ Ðº Boolean

**âœ¨ ÐœÐ¾Ñ‚Ð¸Ð²Ð°Ñ†Ð¸Ñ**

Ð˜Ð´ÐµÐ½Ñ‚Ð¸Ñ„Ð¸ÐºÐ°Ñ†Ð¸Ñ Ñ„ÑƒÐ½ÐºÑ†Ð¸Ð¹ Ð¸ Ð¼ÐµÑ‚Ð¾Ð´Ð¾Ð², Ð²Ð¾Ð·Ð²Ñ€Ð°Ñ‰Ð°ÑŽÑ‰Ð¸Ñ… Boolean.
Ð˜Ð¼ÐµÐ½Ð¾Ð²Ð°Ð½Ð¸Ðµ Ñ„ÑƒÐ½ÐºÑ†Ð¸Ð¹ Ð¸Ð´ÐµÐ½Ñ‚Ð¸Ñ‡Ð½Ð¾ [Ð¸Ð¼ÐµÐ½Ð¾Ð²Ð°Ð½Ð¸ÑŽ Boolean Ð¿ÐµÑ€ÐµÐ¼ÐµÐ½Ð½Ñ‹Ñ…](./vars#boolean-Ð¿ÐµÑ€ÐµÐ¼ÐµÐ½Ð½Ñ‹Ðµ-Ð´Ð¾Ð»Ð¶Ð½Ñ‹-Ð¸Ð¼ÐµÑ‚ÑŒ-Ð¿Ñ€ÐµÑ„Ð¸ÐºÑ-ÑƒÐºÐ°Ð·Ñ‹Ð²Ð°ÑŽÑ‰Ð¸Ð¹-Ð½Ð°-Ð¿Ñ€Ð¸Ð½Ð°Ð´Ð»ÐµÐ¶Ð½Ð¾ÑÑ‚ÑŒ-Ðº-boolean), 
Ð½Ð¾ Ð¿Ñ€Ð°Ð²Ð¸Ð»Ð¾ ÑÑ‡Ð¸Ñ‚Ð°ÐµÐ¼ ÐºÐ°Ðº ÑÑ‚Ð°Ð½Ð´Ð°Ñ€Ñ‚, ÐºÐ¾Ñ‚Ð¾Ñ€Ñ‹Ð¹ Ð¿Ñ€Ð¸Ð²Ñ‹Ñ‡Ð½Ð¾ Ð¸ÑÐ¿Ð¾Ð»ÑŒÐ·ÑƒÐµÑ‚ÑÑ Ð² ÑÑ‚Ð°Ð½Ð´Ð°Ñ€Ñ‚Ðµ ÑÐ·Ñ‹ÐºÐ° (Ð½Ð°Ð¿Ñ€Ð¸Ð¼ÐµÑ€: `isNaN`, `hasOwnProperty`).

**ðŸ¤– ÐÐ²Ñ‚Ð¾Ð¼Ð°Ñ‚Ð¸Ð·Ð°Ñ†Ð¸Ñ**

ÐÐµ Ð¸Ð¼Ð¿Ð»ÐµÐ¼ÐµÐ½Ñ‚Ð¸Ñ€Ð¾Ð²Ð°Ð½Ð¾ Ð² eslint-config.

**âœ… Valid**

```ts
const isEmpty = (value: unknown) => remeda.isEmpty(value);

const wasRemoved = (obj?: Record<string, unknown>): obj is undefined =>
    obj === undefined;

const hasID = (obj: { id?: string }) =>
    Boolean(obj.id);
```

```ts
class Example {
  public isEmpty = (value: unknown) => remeda.isEmpty(value);

  public wasRemoved = (obj?: Record<string, unknown>): obj is undefined =>
    obj === undefined;

  public hasID = (obj: { id?: string }) => Boolean(obj.id);
}
```

**âŒ Invalid**

```ts
const empty = (value: unknown) => remeda.isEmpty(value);

const checkRemoved = (obj?: Record<string, unknown>): obj is undefined =>
    obj === undefined;
```

## Ð—Ð°Ð¿Ñ€ÐµÑ‰ÐµÐ½Ð¾ Ð¸ÑÐ¿Ð¾Ð»ÑŒÐ·Ð¾Ð²Ð°Ñ‚ÑŒ Ð² Ð½Ð°Ð·Ð²Ð°Ð½Ð¸ÑÑ… ÑÑƒÑ‰ÐµÑÑ‚Ð²Ð¸Ñ‚ÐµÐ»ÑŒÐ½Ñ‹Ðµ Ð²Ð¾ Ð¼Ð½Ð¾Ð¶ÐµÑÑ‚Ð²ÐµÐ½Ð½Ð¾Ð¼ Ñ‡Ð¸ÑÐ»Ðµ Ñ‚Ð¾Ð»ÑŒÐºÐ¾ Ñ Ð´Ð¾Ð±Ð°Ð²Ð»ÐµÐ½Ð¸ÐµÐ¼ Ð¾ÐºÐ¾Ð½Ñ‡Ð°Ð½Ð¸Ñ

ÐÐµÐ¾Ð±Ñ…Ð¾Ð´Ð¸Ð¼Ð¾ Ð´Ð¾Ð±Ð°Ð²Ð»ÑÑ‚ÑŒ ÑƒÑ‚Ð¾Ñ‡Ð½ÑÑŽÑ‰ÐµÐµ ÑÐ»Ð¾Ð²Ð¾: `getDrafts` -> `getDraftList`.

**âœ¨ ÐœÐ¾Ñ‚Ð¸Ð²Ð°Ñ†Ð¸Ñ**

Ð˜Ð·-Ð·Ð° ÑÑƒÑ‰ÐµÑÑ‚Ð²Ð¸Ñ‚ÐµÐ»ÑŒÐ½Ñ‹Ñ… Ð²Ð¾ Ð¼Ð½Ð¾Ð¶ÐµÑÑ‚Ð²ÐµÐ½Ð½Ð¾Ð¼ Ñ‡Ð¸ÑÐ»Ðµ Ñ Ð´Ð¾Ð±Ð°Ð²Ð»ÐµÐ½Ð¸ÐµÐ¼ Ð¾ÐºÐ¾Ð½Ñ‡Ð°Ð½Ð¸Ñ (Ð½Ð°Ð¿Ñ€Ð¸Ð¼ÐµÑ€, `s`) ÑÐ»Ð¾Ð¶Ð½Ð¾ Ð¾Ñ‚Ð»Ð¸Ñ‡Ð¸Ñ‚ÑŒ Ñ„ÑƒÐ½ÐºÑ†Ð¸ÑŽ, Ñ€Ð°Ð±Ð¾Ñ‚Ð°ÑŽÑ‰ÑƒÑŽ Ñ Ð¾Ð´Ð½Ð¾Ð¹ ÑÑƒÑ‰Ð½Ð¾ÑÑ‚ÑŒÑŽ Ð¸ Ð½ÐµÑÐºÐ¾Ð»ÑŒÐºÐ¸Ð¼Ð¸.

**ðŸ¤– ÐÐ²Ñ‚Ð¾Ð¼Ð°Ñ‚Ð¸Ð·Ð°Ñ†Ð¸Ñ**

ÐÐµ Ð¸Ð¼Ð¿Ð»ÐµÐ¼ÐµÐ½Ñ‚Ð¸Ñ€Ð¾Ð²Ð°Ð½Ð¾ Ð² eslint-config.

**âœ… Valid**

```ts
class Example {
  public getDraft = () => {};
  public getDraftList = () => {};
  
  public sendDraft = () => {};
  public sendDraftList = () => {};
  public sendAllDraftList = () => {};

  public calcDraftSum = () => {};
  public calcDraftListSum = () => {};
}
```

---

```tsx
const DraftScreen = () => {};

const DraftListScreen = () => {};
```

**âŒ Invalid**

Ð¡Ð»Ð¾Ð¶Ð½Ð¾ Ð²Ð¸Ð·ÑƒÐ°Ð»ÑŒÐ½Ð¾ Ð½Ð°Ð¹Ñ‚Ð¸ Ð¾Ñ‚Ð»Ð¸Ñ‡Ð¸Ñ Ð² Ð½Ð°Ð·Ð²Ð°Ð½Ð¸ÑÑ… Ð¸Ð·-Ð·Ð° Ñ€Ð°Ð·Ð½Ð¸Ñ†Ñ‹ Ð² Ð¾Ð´Ð½Ñƒ Ð±ÑƒÐºÐ²Ñƒ.

```ts
class Example {
  public getDraft = () => {};
  public getDrafts = () => {};
  
  public sendDraft = () => {};
  public sendDrafts = () => {};
  public sendAllDrafts = () => {};

  public calcDraftSum = () => {};
  public calcDraftsSum = () => {};
}
```

---

```tsx
const DraftScreen = () => {};

const DraftsScreen = () => {};
```

## Ð˜ÑÐ¿Ð¾Ð»ÑŒÐ·Ð¾Ð²Ð°Ð½Ð¸Ðµ `Remove` Ð¸ `Delete` Ð² Ð¸Ð¼ÐµÐ½Ð°Ñ…

**âœ¨ ÐœÐ¾Ñ‚Ð¸Ð²Ð°Ñ†Ð¸Ñ**

Ð˜ÑÐºÐ»ÑŽÑ‡ÐµÐ½Ð¸Ðµ ÑÐ¿Ð¾Ñ€Ð¾Ð² Ð¾ Ð½ÐµÐ¹Ð¼Ð¸Ð½Ð³Ðµ Ñ„ÑƒÐ½ÐºÑ†Ð¸Ð¹ Ð¸ Ð¼ÐµÑ‚Ð¾Ð´Ð¾Ð² ÑƒÐ´Ð°Ð»ÐµÐ½Ð¸Ñ.

**ðŸ¤– ÐÐ²Ñ‚Ð¾Ð¼Ð°Ñ‚Ð¸Ð·Ð°Ñ†Ð¸Ñ**

ÐÐµ Ð¸Ð¼Ð¿Ð»ÐµÐ¼ÐµÐ½Ñ‚Ð¸Ñ€Ð¾Ð²Ð°Ð½Ð¾ Ð² eslint-config.

### Ð”Ð»Ñ Ð±Ð¸Ð·Ð½ÐµÑ-Ð»Ð¾Ð³Ð¸ÐºÐ¸

Ð•ÑÐ»Ð¸ Ð¾Ð¿ÐµÑ€Ð°Ñ†Ð¸Ñ ÑƒÐ´Ð°Ð»ÐµÐ½Ð¸Ñ Ð¿Ñ€Ð¾Ð¸Ð·Ð²Ð¾Ð´Ð¸Ñ‚ÑÑ Ñ Ð²Ð¾Ð·Ð¼Ð¾Ð¶Ð½Ð¾ÑÑ‚ÑŒÑŽ Ð²Ð¾ÑÑÑ‚Ð°Ð½Ð¾Ð²Ð»ÐµÐ½Ð¸Ñ, Ñ‚Ð¾ Ð½ÐµÐ¾Ð±Ñ…Ð¾Ð´Ð¸Ð¼Ð¾ Ð¸ÑÐ¿Ð¾Ð»ÑŒÐ·Ð¾Ð²Ð°Ñ‚ÑŒ `Remove`.
Ð•ÑÐ»Ð¸ Ð²Ð¾ÑÑÑ‚Ð°Ð½Ð¾Ð²Ð»ÐµÐ½Ð¸Ðµ Ð½Ðµ Ð²Ð¾Ð·Ð¼Ð¾Ð¶Ð½Ð¾, Ñ‚Ð¾ Ð½ÐµÐ¾Ð±Ñ…Ð¾Ð´Ð¸Ð¼Ð¾ Ð¸ÑÐ¿Ð¾Ð»ÑŒÐ·Ð¾Ð²Ð°Ñ‚ÑŒ `Delete`.

**âœ… Valid**

`removeAccount` - Ñ„ÑƒÐ½ÐºÑ†Ð¸Ñ ÑƒÐ´Ð°Ð»ÐµÐ½Ð¸Ñ Ð°ÐºÐºÐ°ÑƒÐ½Ñ‚Ð° Ð¿Ð¾Ð»ÑŒÐ·Ð¾Ð²Ð°Ñ‚ÐµÐ»Ñ Ñ Ð²Ð¾Ð·Ð¼Ð¾Ð¶Ð½Ð¾ÑÑ‚ÑŒÑŽ Ð² Ð±ÑƒÐ´ÑƒÑ‰ÐµÐ¼ ÐµÐ³Ð¾ Ð²Ð¾ÑÑÑ‚Ð°Ð½Ð¾Ð²Ð»ÐµÐ½Ð¸Ñ.

`removeRequest` - Ñ„ÑƒÐ½ÐºÑ†Ð¸Ñ, ÑƒÐ´Ð°Ð»ÑÑŽÑ‰Ð°Ñ Ð¸Ð· Ñ€ÐµÐµÑÑ‚Ñ€Ð° Ð·Ð°ÑÐ²Ð»ÐµÐ½Ð¸Ðµ, Ð½Ð¾ Ð¿Ñ€Ð¸ ÑÑ‚Ð¾Ð¼ Ð¿ÐµÑ€ÐµÐ¼ÐµÑ‰Ð°ÑŽÑ‰Ð°Ñ ÐµÐ³Ð¾ Ð² "Ð°Ñ€Ñ…Ð¸Ð²".

`deleteRequest` - Ñ„ÑƒÐ½ÐºÑ†Ð¸Ñ Ð¿Ð¾Ð»Ð½Ð¾ÑÑ‚ÑŒÑŽ ÑƒÐ´Ð°Ð»ÑÑŽÑ‰Ð°Ñ Ð·Ð°ÑÐ²Ð»ÐµÐ½Ð¸Ðµ Ð¸Ð· ÑÐ¸ÑÑ‚ÐµÐ¼Ñ‹.

### Ð”Ð»Ñ Ð¸Ð½Ñ„Ñ€Ð°ÑÑ‚Ñ€ÑƒÐºÑ‚ÑƒÑ€Ð½Ð¾Ð³Ð¾ ÐºÐ¾Ð´Ð°

Ð”Ð»Ñ Ð¼ÑƒÑ‚Ð¸Ñ€ÑƒÑŽÑ‰Ð¸Ñ… Ð¾Ð¿ÐµÑ€Ð°Ñ†Ð¸Ð¹ Ð¸ÑÐ¿Ð¾Ð»ÑŒÐ·ÑƒÐµÑ‚ÑÑ `Delete`.
Ð”Ð»Ñ Ð¸Ð¼Ð¼ÑƒÑ‚Ð°Ð±ÐµÐ»ÑŒÐ½Ñ‹Ñ… Ð¾Ð¿ÐµÑ€Ð°Ñ†Ð¸Ð¹ Ð¸ÑÐ¿Ð¾Ð»ÑŒÐ·ÑƒÐµÑ‚ÑÑ `Remove`.

**âœ… Valid**

```ts
// ÐœÑƒÑ‚Ð¸Ñ€ÑƒÑŽÑ‰Ð°Ñ Ð¾Ð¿ÐµÑ€Ð°Ñ†Ð¸Ð¸
const deleteID = (user: User) => {
  delete user.id;
};

// Ð˜Ð¼Ð¼ÑƒÑ‚Ð°Ð±ÐµÐ»ÑŒÐ½Ð°Ñ Ð¾Ð¿ÐµÑ€Ð°Ñ†Ð¸Ñ
const removeID = ({ id, ...user }: User) => user;

// ÐœÑƒÑ‚Ð¸Ñ€ÑƒÑŽÑ‰Ð°Ñ Ð¾Ð¿ÐµÑ€Ð°Ñ†Ð¸Ð¸
const deleteItem = (array: unknown[]) => {
    array.splice(2);
};

// Ð˜Ð¼Ð¼ÑƒÑ‚Ð°Ð±ÐµÐ»ÑŒÐ½Ð°Ñ Ð¾Ð¿ÐµÑ€Ð°Ñ†Ð¸Ñ
const removeItem = (array: unknown[]) => array.slice(2);
```

```ts
class Example {
  private id?: string;
  
  private deleteID = () => {
    this.id = undefined;
  }
}
```
