# ÐÐµÐ¹Ð¼Ð¸Ð½Ð³ Ñ„ÑƒÐ½ÐºÑ†Ð¸Ð¹ Ð¸ Ð¼ÐµÑ‚Ð¾Ð´Ð¾Ð²

## Ð¤ÑƒÐ½ÐºÑ†Ð¸Ð¸ Ð¸ Ð¼ÐµÑ‚Ð¾Ð´Ñ‹ Ð´Ð¾Ð»Ð¶Ð½Ñ‹ Ð½Ð°Ñ‡Ð¸Ð½Ð°Ñ‚ÑŒÑÑ Ñ Ð³Ð»Ð°Ð³Ð¾Ð»Ð°

Ð˜ÑÐºÐ»ÑŽÑ‡ÐµÐ½Ð¸Ðµ: [Ð¤ÑƒÐ½ÐºÑ†Ð¸Ð¸ Ð¸ Ð¼ÐµÑ‚Ð¾Ð´Ñ‹, Ð²Ð¾Ð·Ð²Ñ€Ð°Ñ‰Ð°ÑŽÑ‰Ð¸Ðµ Boolean](#Ñ„ÑƒÐ½ÐºÑ†Ð¸Ð¸-Ð¸-Ð¼ÐµÑ‚Ð¾Ð´Ñ‹-Ð²Ð¾Ð·Ð²Ñ€Ð°Ñ‰Ð°ÑŽÑ‰Ð¸Ðµ-boolean-Ð´Ð¾Ð»Ð¶Ð½Ñ‹-ÑÐ¾Ð´ÐµÑ€Ð¶Ð°Ñ‚ÑŒ-Ð¿Ñ€ÐµÑ„Ð¸ÐºÑ-ÑƒÐºÐ°Ð·Ñ‹Ð²Ð°ÑŽÑ‰Ð¸Ð¹-Ð½Ð°-Ð¿Ñ€Ð¸Ð½Ð°Ð´Ð»ÐµÐ¶Ð½Ð¾ÑÑ‚ÑŒ-Ðº-boolean).

**âœ¨ ÐœÐ¾Ñ‚Ð¸Ð²Ð°Ñ†Ð¸Ñ**

Ð“Ð»Ð°Ð³Ð¾Ð» Ð² Ð½Ð°Ð·Ð²Ð°Ð½Ð¸Ð¸ Ñ„ÑƒÐ½ÐºÑ†Ð¸Ð¸ Ð¸Ð»Ð¸ Ð¼ÐµÑ‚Ð¾Ð´Ð° Ð¿Ð¾Ð·Ð²Ð¾Ð»ÑÐµÑ‚ Ð±Ñ‹ÑÑ‚Ñ€Ð¾ Ð¿Ð¾Ð½ÑÑ‚ÑŒ Ð²Ñ‹Ð¿Ð¾Ð»Ð½ÑÐµÐ¼Ð¾Ðµ Ð´ÐµÐ¹ÑÑ‚Ð²Ð¸Ðµ.

**ðŸ¤– ÐÐ²Ñ‚Ð¾Ð¼Ð°Ñ‚Ð¸Ð·Ð°Ñ†Ð¸Ñ**

ÐÐµ Ð¸Ð¼Ð¿Ð»ÐµÐ¼ÐµÐ½Ñ‚Ð¸Ñ€Ð¾Ð²Ð°Ð½Ð¾ Ð² eslint-config.

**âœ… Valid**

```ts
const calcSum = () => {};

const getUserData = () => {};

const removeID = () => {};

const refreshData = () => {};

const formatToView = () => {};

const createRequest = () => {};
```

```ts
class Example {
  public calcSum = () => {};

  public getUserData = () => {};

  public removeID = () => {};

  public refresh = () => {};

  public formatToView = () => {};
}
```

**âŒ Invalid**
```ts
// Ð“Ð»Ð°Ð³Ð¾Ð» Ð² ÐºÐ¾Ð½Ñ†Ðµ
const sumCalc = () => {};

// Ð¡ÑƒÐ´Ñ Ð¸Ð· Ð½Ð°Ð·Ð²Ð°Ð½Ð¸Ñ ÑÑ‚Ð¾ Ð½Ðµ Ñ„ÑƒÐ½ÐºÑ†Ð¸Ñ, Ð° Ð¾Ð±ÑŠÐµÐºÑ‚
const userDataGetter = () => {};

// Ð¡ÑƒÐ´Ñ Ð¸Ð· Ð½Ð°Ð·Ð²Ð°Ð½Ð¸Ñ ÑÑ‚Ð¾ Ð½Ðµ Ñ„ÑƒÐ½ÐºÑ†Ð¸Ñ, Ð° Ð¾Ð±ÑŠÐµÐºÑ‚
const toViewFormatter = () => {};

// ÐÐ°Ð·Ð²Ð°Ð½Ð¸Ðµ ÑƒÐºÐ°Ð·Ñ‹Ð²Ð°ÐµÑ‚, Ñ‡Ñ‚Ð¾ ÑÑ‚Ð¾ Ð¿Ñ€Ð¾Ñ†ÐµÑÑ, Ð° Ð½Ðµ Ñ„ÑƒÐ½ÐºÑ†Ð¸Ñ
const refreshingData = () => {};
```

## Ð¤ÑƒÐ½ÐºÑ†Ð¸Ð¸ Ð¸ Ð¼ÐµÑ‚Ð¾Ð´Ñ‹, Ð²Ð¾Ð·Ð²Ñ€Ð°Ñ‰Ð°ÑŽÑ‰Ð¸Ðµ Boolean Ð´Ð¾Ð»Ð¶Ð½Ñ‹ ÑÐ¾Ð´ÐµÑ€Ð¶Ð°Ñ‚ÑŒ Ð¿Ñ€ÐµÑ„Ð¸ÐºÑ, ÑƒÐºÐ°Ð·Ñ‹Ð²Ð°ÑŽÑ‰Ð¸Ð¹ Ð½Ð° Ð¿Ñ€Ð¸Ð½Ð°Ð´Ð»ÐµÐ¶Ð½Ð¾ÑÑ‚ÑŒ Ðº Boolean

**âœ¨ ÐœÐ¾Ñ‚Ð¸Ð²Ð°Ñ†Ð¸Ñ**

Ð˜Ð´ÐµÐ½Ñ‚Ð¸Ñ„Ð¸ÐºÐ°Ñ†Ð¸Ñ Ñ„ÑƒÐ½ÐºÑ†Ð¸Ð¹ Ð¸ Ð¼ÐµÑ‚Ð¾Ð´Ð¾Ð², Ð²Ð¾Ð·Ð²Ñ€Ð°Ñ‰Ð°ÑŽÑ‰Ð¸Ñ… Boolean.
Ð˜Ð¼ÐµÐ½Ð¾Ð²Ð°Ð½Ð¸Ðµ Ñ„ÑƒÐ½ÐºÑ†Ð¸Ð¹ Ð¸Ð´ÐµÐ½Ñ‚Ð¸Ñ‡Ð½Ð¾ [Ð¸Ð¼ÐµÐ½Ð¾Ð²Ð°Ð½Ð¸ÑŽ Boolean Ð¿Ð°Ñ€Ð°Ð¼ÐµÑ‚Ñ€Ð¾Ð² Ð¸ ÑÐ²Ð¾Ð¹ÑÑ‚Ð²](./props), 
Ð½Ð¾ Ð¿Ñ€Ð°Ð²Ð¸Ð»Ð¾ ÑÑ‡Ð¸Ñ‚Ð°ÐµÐ¼ ÐºÐ°Ðº ÑÑ‚Ð°Ð½Ð´Ð°Ñ€Ñ‚, ÐºÐ¾Ñ‚Ð¾Ñ€Ñ‹Ð¹ Ð¿Ñ€Ð¸Ð²Ñ‹Ñ‡Ð½Ð¾ Ð¸ÑÐ¿Ð¾Ð»ÑŒÐ·ÑƒÐµÑ‚ÑÑ Ð² Ð¿Ð¾Ð¿ÑƒÐ»ÑÑ€Ð½Ñ‹Ñ… npm-Ð¿Ð°ÐºÐµÑ‚Ð°Ñ… (Ð½Ð°Ð¿Ñ€Ð¸Ð¼ÐµÑ€, `remeda`).

**ðŸ¤– ÐÐ²Ñ‚Ð¾Ð¼Ð°Ñ‚Ð¸Ð·Ð°Ñ†Ð¸Ñ**

ÐÐµ Ð¸Ð¼Ð¿Ð»ÐµÐ¼ÐµÐ½Ñ‚Ð¸Ñ€Ð¾Ð²Ð°Ð½Ð¾ Ð² eslint-config.

**âœ… Valid**

```ts
const isEmpty = (value: unknown) => remeda.isEmpty(value);

const wasRemoved = (obj?: Record<string, unknown>): obj is undefined =>
    obj === undefined;

const hasID = (obj: { id?: string }) =>
    Boolean(obj.id);
```

```ts
class Example {
  public isEmpty = (value: unknown) => remeda.isEmpty(value);

  public wasRemoved = (obj?: Record<string, unknown>): obj is undefined =>
    obj === undefined;

  public hasID = (obj: { id?: string }) => Boolean(obj.id);
}
```

**âŒ Invalid**

```ts
const empty = (value: unknown) => remeda.isEmpty(value);

const checkRemoved = (obj?: Record<string, unknown>): obj is undefined =>
    obj === undefined;
```

