# –§—É–Ω–∫—Ü–∏–∏ –∏ –º–µ—Ç–æ–¥—ã

## –§—É–Ω–∫—Ü–∏–∏ –∏ –º–µ—Ç–æ–¥—ã –¥–æ–ª–∂–Ω—ã –Ω–∞—á–∏–Ω–∞—Ç—å—Å—è —Å –≥–ª–∞–≥–æ–ª–∞

**‚ú® –ú–æ—Ç–∏–≤–∞—Ü–∏—è**

–ì–ª–∞–≥–æ–ª –≤ –Ω–∞–∑–≤–∞–Ω–∏–∏ —Ñ—É–Ω–∫—Ü–∏–∏ –∏–ª–∏ –º–µ—Ç–æ–¥–∞ –ø–æ–∑–≤–æ–ª—è–µ—Ç –±—ã—Å—Ç—Ä–æ –ø–æ–Ω—è—Ç—å –≤—ã–ø–æ–ª–Ω—è–µ–º–æ–µ –¥–µ–π—Å—Ç–≤–∏–µ.

**ü§ñ –ê–≤—Ç–æ–º–∞—Ç–∏–∑–∞—Ü–∏—è**

–ù–µ –∏–º–ø–ª–µ–º–µ–Ω—Ç–∏—Ä–æ–≤–∞–Ω–æ –≤ eslint-config.

**‚úÖ Valid**

```ts
const calcSum = () => {};

const getUserData = () => {};

const removeID = () => {};

const refreshData = () => {};

const formatToView = () => {};

const createRequest = () => {};
```

```ts
class Example {
  public calcSum = () => {};

  public getUserData = () => {};

  public removeID = () => {};

  public refresh = () => {};

  public formatToView = () => {};
}
```

**‚ùå Invalid**
```ts
// –ì–ª–∞–≥–æ–ª –≤ –∫–æ–Ω—Ü–µ
const sumCalc = () => {};

// –°—É–¥—è –∏–∑ –Ω–∞–∑–≤–∞–Ω–∏—è —ç—Ç–æ –Ω–µ —Ñ—É–Ω–∫—Ü–∏—è, –∞ –æ–±—ä–µ–∫—Ç
const userDataGetter = () => {};

// –°—É–¥—è –∏–∑ –Ω–∞–∑–≤–∞–Ω–∏—è —ç—Ç–æ –Ω–µ —Ñ—É–Ω–∫—Ü–∏—è, –∞ –æ–±—ä–µ–∫—Ç
const toViewFormatter = () => {};

// –ù–∞–∑–≤–∞–Ω–∏–µ —É–∫–∞–∑—ã–≤–∞–µ—Ç, —á—Ç–æ —ç—Ç–æ –ø—Ä–æ—Ü–µ—Å—Å, –∞ –Ω–µ —Ñ—É–Ω–∫—Ü–∏—è
const refreshingData = () => {};
```

### –§—É–Ω–∫—Ü–∏–∏ –∏ –º–µ—Ç–æ–¥—ã, –≤–æ–∑–≤—Ä–∞—â–∞—é—â–∏–µ Boolean –¥–æ–ª–∂–Ω—ã –Ω–∞—á–∏–Ω–∞—Ç—å—Å—è —Å –≥–ª–∞–≥–æ–ª–∞

–ì–ª–∞–≥–æ–ª –≤ –Ω–∞—á–∞–ª–µ —Ñ—É–Ω–∫—Ü–∏–∏ –¥–æ–ª–∂–µ–Ω —É–∫–∞–∑—ã–≤–∞—Ç—å –Ω–∞ —Ü–µ–ª—å –ø—Ä–æ–≤–æ–¥–∏–º–æ–π –ø—Ä–æ–≤–µ—Ä–∫–∏.

**‚ú® –ú–æ—Ç–∏–≤–∞—Ü–∏—è**

–î–æ–±–∞–≤–ª–µ–Ω–∏–µ –≥–ª–∞–≥–æ–ª–∞ –≤ –Ω–∞—á–∞–ª–æ —Ñ—É–Ω–∫—Ü–∏–∏ –ø–æ–∑–≤–æ–ª—è–µ—Ç –∏–∑–±–µ–∂–∞—Ç—å –ø–µ—Ä–µ—Å–µ—á–µ–Ω–∏–π —Å Boolean –ø–µ—Ä–µ–º–µ–Ω–Ω—ã–º–∏.
–°–∞–º—ã–π —Ä–∞—Å–ø—Ä–æ—Å—Ç—Ä–∞–Ω–µ–Ω–Ω—ã–π –≥–ª–∞–≥–æ–ª: check.

**ü§ñ –ê–≤—Ç–æ–º–∞—Ç–∏–∑–∞—Ü–∏—è**

–ù–µ –∏–º–ø–ª–µ–º–µ–Ω—Ç–∏—Ä–æ–≤–∞–Ω–æ –≤ eslint-config.

**‚úÖ Valid**

```ts
const checkIsEmpty = (value: unknown) => remeda.isEmpty(value);

const checkWasRemoved = (obj?: Record<string, unknown>): obj is undefined =>
    obj === undefined;

const validateID = (id: string) =>
    id.length === 2;
```

```ts
class Example {
  public checkIsEmpty = (value: unknown) => remeda.isEmpty(value);

  public checkWasRemoved = (obj?: Record<string, unknown>): obj is undefined =>
    obj === undefined;

  public validateID = (obj: { id?: string }) => Boolean(obj.id);
}
```

**‚ùå Invalid**

```ts
// –∏–º—è –∏–¥–µ–Ω—Ç–∏—á–Ω–æ –ø–µ—Ä–µ–º–µ–Ω–Ω–æ–π
const isEmpty = (value: unknown) => remeda.isEmpty(value);

// –∏–º—è –∏–¥–µ–Ω—Ç–∏—á–Ω–æ –ø–µ—Ä–µ–º–µ–Ω–Ω–æ–π
const hasRemoved = (obj?: Record<string, unknown>): obj is undefined =>
    obj === undefined;

// –Ω–µ—Ç –≥–ª–∞–≥–æ–ª–∞
const removed = (obj?: Record<string, unknown>): obj is undefined =>
    obj === undefined;
```

```tsx
// –ø—Ä–∏—Ö–æ–¥–∏—Ç—Å—è –ø–µ—Ä–µ–∏–º–µ–Ω–æ–≤—ã–≤–∞—Ç—å —Ñ—É–Ω–∫—Ü–∏—é –∏–∑-–∑–∞ –ø–µ—Ä–µ—Å–µ—á–µ–Ω–∏–π —Å –ø–∞—Ä–∞–º–µ—Ç—Ä–∞–º–∏ –∏ –ø–µ—Ä–µ–º–µ–Ω–Ω—ã–º–∏
import { isEmpty as checkIsEmpty } from 'utils';

export const UserInfo = ({ data, isEmpty }: Props) => {
  const isEmptyData = isEmpty && checkIsEmpty(data);

  return (
    <Grid>
      {isEmpty && <NoData />}
    </Grid>
  );
};
```

## –î–ª—è —Å—É—â–µ—Å—Ç–≤–∏—Ç–µ–ª—å–Ω—ã—Ö –≤–æ –º–Ω–æ–∂–µ—Å—Ç–≤–µ–Ω–Ω–æ–º —á–∏—Å–ª–µ –Ω–µ–æ–±—Ö–æ–¥–∏–º–æ –¥–æ–±–∞–≤–ª—è—Ç—å —É—Ç–æ—á–Ω—è—é—â–µ–µ —Å–ª–æ–≤–æ

–ù–µ–æ–±—Ö–æ–¥–∏–º–æ –¥–æ–±–∞–≤–ª—è—Ç—å —É—Ç–æ—á–Ω—è—é—â–µ–µ —Å–ª–æ–≤–æ: `getDrafts` -> `getDraftList`.

**‚ú® –ú–æ—Ç–∏–≤–∞—Ü–∏—è**

–ò–∑-–∑–∞ —Å—É—â–µ—Å—Ç–≤–∏—Ç–µ–ª—å–Ω—ã—Ö –≤–æ –º–Ω–æ–∂–µ—Å—Ç–≤–µ–Ω–Ω–æ–º —á–∏—Å–ª–µ —Å –¥–æ–±–∞–≤–ª–µ–Ω–∏–µ–º –æ–∫–æ–Ω—á–∞–Ω–∏—è (–Ω–∞–ø—Ä–∏–º–µ—Ä, `s`) —Å–ª–æ–∂–Ω–æ –æ—Ç–ª–∏—á–∏—Ç—å —Ñ—É–Ω–∫—Ü–∏—é, —Ä–∞–±–æ—Ç–∞—é—â—É—é —Å –æ–¥–Ω–æ–π —Å—É—â–Ω–æ—Å—Ç—å—é –∏ –Ω–µ—Å–∫–æ–ª—å–∫–∏–º–∏

**ü§ñ –ê–≤—Ç–æ–º–∞—Ç–∏–∑–∞—Ü–∏—è**

–ù–µ –∏–º–ø–ª–µ–º–µ–Ω—Ç–∏—Ä–æ–≤–∞–Ω–æ –≤ eslint-config.

**‚úÖ Valid**

```ts
class Example {
  public getDraft = () => {};
  public getDraftList = () => {};
  
  public sendDraft = () => {};
  public sendDraftList = () => {};

  public calcDraftSum = () => {};
  public calcDraftListSum = () => {};
}
```

---

```tsx
const DraftScreen = () => {};

const DraftListScreen = () => {};
```

**‚ùå Invalid**

–°–ª–æ–∂–Ω–æ –≤–∏–∑—É–∞–ª—å–Ω–æ –Ω–∞–π—Ç–∏ –æ—Ç–ª–∏—á–∏—è –≤ –Ω–∞–∑–≤–∞–Ω–∏—è—Ö –∏–∑-–∑–∞ —Ä–∞–∑–Ω–∏—Ü—ã –≤ –æ–¥–Ω—É –±—É–∫–≤—É.

```ts
class Example {
  public getDraft = () => {};
  public getDrafts = () => {};
  
  public sendDraft = () => {};
  public sendDrafts = () => {};

  public calcDraftSum = () => {};
  public calcDraftsSum = () => {};
}
```

---

```tsx
const DraftScreen = () => {};

const DraftsScreen = () => {};
```

### –†–µ–∫–æ–º–µ–Ω–¥—É–µ—Ç—Å—è –∏—Å–ø–æ–ª—å–∑–æ–≤–∞—Ç—å –ø–æ—Å—Ç—Ñ–∏–∫—Å `List` –¥–ª—è —Å—É—â–µ—Å—Ç–≤–∏—Ç–µ–ª—å–Ω—ã—Ö –≤–æ –º–Ω–æ–∂–µ—Å—Ç–≤–µ–Ω–Ω–æ–º —á–∏—Å–ª–µ

–í –±–æ–ª—å—à–∏–Ω—Å—Ç–≤–µ —Å–ª—É—á–∞–µ–≤ –¥–ª—è –æ–ø–∏—Å–∞–Ω–∏—è —Å—É—â–µ—Å—Ç–≤–∏—Ç–µ–ª—å–Ω–æ–≥–æ –≤–æ –º–Ω–æ–∂–µ—Å—Ç–≤–µ–Ω–Ω–æ–º —á–∏—Å–ª–µ –ø–æ–¥–æ–π–¥–µ—Ç `List`, —ç—Ç–æ–º—É –ø–æ—Å—Ç—Ñ–∏–∫—Å—É –Ω–µ–æ–±—Ö–æ–¥–∏–º–æ –æ—Ç–¥–∞–≤–∞—Ç—å –ø—Ä–µ–¥–ø–æ—á—Ç–µ–Ω–∏–µ.

**‚ú® –ú–æ—Ç–∏–≤–∞—Ü–∏—è**

–ò—Å–ø–æ–ª—å–∑–æ–≤–∞–Ω–∏–µ —Ä–∞–∑–ª–∏—á–Ω—ã—Ö –ø–æ—Å—Ç—Ñ–∏–∫—Å–æ–≤ –º–æ–∂–µ—Ç –ø—Ä–∏–≤–µ—Å—Ç–∏ –∫ –Ω–µ–æ–¥–Ω–æ—Ä–æ–¥–Ω–æ—Å—Ç–∏ –∫–æ–¥–∞. –ù–∞–ø—Ä–∏–º–µ—Ä, –∏—Å–ø–æ–ª—å–∑–æ–≤–∞–Ω–∏–µ —Å–ª–æ–≤ `Many`, `One`, `Sindle`, `Multiple`...

## –ò—Å–ø–æ–ª—å–∑–æ–≤–∞–Ω–∏–µ `Remove` –∏ `Delete` –≤ –∏–º–µ–Ω–∞—Ö

**‚ú® –ú–æ—Ç–∏–≤–∞—Ü–∏—è**

–ò—Å–∫–ª—é—á–µ–Ω–∏–µ —Å–ø–æ—Ä–æ–≤ –æ –Ω–µ–π–º–∏–Ω–≥–µ —Ñ—É–Ω–∫—Ü–∏–π –∏ –º–µ—Ç–æ–¥–æ–≤ —É–¥–∞–ª–µ–Ω–∏—è.

**ü§ñ –ê–≤—Ç–æ–º–∞—Ç–∏–∑–∞—Ü–∏—è**

–ù–µ –∏–º–ø–ª–µ–º–µ–Ω—Ç–∏—Ä–æ–≤–∞–Ω–æ –≤ eslint-config.

### –î–ª—è –±–∏–∑–Ω–µ—Å-–ª–æ–≥–∏–∫–∏

–ï—Å–ª–∏ –æ–ø–µ—Ä–∞—Ü–∏—è —É–¥–∞–ª–µ–Ω–∏—è –ø—Ä–æ–∏–∑–≤–æ–¥–∏—Ç—Å—è —Å –≤–æ–∑–º–æ–∂–Ω–æ—Å—Ç—å—é –≤–æ—Å—Å—Ç–∞–Ω–æ–≤–ª–µ–Ω–∏—è, —Ç–æ –Ω–µ–æ–±—Ö–æ–¥–∏–º–æ –∏—Å–ø–æ–ª—å–∑–æ–≤–∞—Ç—å `Remove`.
–ï—Å–ª–∏ –≤–æ—Å—Å—Ç–∞–Ω–æ–≤–ª–µ–Ω–∏–µ –Ω–µ –≤–æ–∑–º–æ–∂–Ω–æ, —Ç–æ –Ω–µ–æ–±—Ö–æ–¥–∏–º–æ –∏—Å–ø–æ–ª—å–∑–æ–≤–∞—Ç—å `Delete`.

**‚úÖ Valid**

`removeAccount` - —Ñ—É–Ω–∫—Ü–∏—è —É–¥–∞–ª–µ–Ω–∏—è –∞–∫–∫–∞—É–Ω—Ç–∞ –ø–æ–ª—å–∑–æ–≤–∞—Ç–µ–ª—è —Å –≤–æ–∑–º–æ–∂–Ω–æ—Å—Ç—å—é –≤ –±—É–¥—É—â–µ–º –µ–≥–æ –≤–æ—Å—Å—Ç–∞–Ω–æ–≤–ª–µ–Ω–∏—è.

`removeRequest` - —Ñ—É–Ω–∫—Ü–∏—è, —É–¥–∞–ª—è—é—â–∞—è –∏–∑ —Ä–µ–µ—Å—Ç—Ä–∞ –∑–∞—è–≤–ª–µ–Ω–∏–µ, –Ω–æ –ø—Ä–∏ —ç—Ç–æ–º –ø–µ—Ä–µ–º–µ—â–∞—é—â–∞—è –µ–≥–æ –≤ "–∞—Ä—Ö–∏–≤".

`deleteRequest` - —Ñ—É–Ω–∫—Ü–∏—è –ø–æ–ª–Ω–æ—Å—Ç—å—é —É–¥–∞–ª—è—é—â–∞—è –∑–∞—è–≤–ª–µ–Ω–∏–µ –∏–∑ —Å–∏—Å—Ç–µ–º—ã.

### –î–ª—è –∏–Ω—Ñ—Ä–∞—Å—Ç—Ä—É–∫—Ç—É—Ä–Ω–æ–≥–æ –∫–æ–¥–∞

–î–ª—è –º—É—Ç–∏—Ä—É—é—â–∏—Ö –æ–ø–µ—Ä–∞—Ü–∏–π –∏—Å–ø–æ–ª—å–∑—É–µ—Ç—Å—è `Delete`.
–î–ª—è –∏–º–º—É—Ç–∞–±–µ–ª—å–Ω—ã—Ö –æ–ø–µ—Ä–∞—Ü–∏–π –∏—Å–ø–æ–ª—å–∑—É–µ—Ç—Å—è `Remove`.

**‚úÖ Valid**

```ts
// –ú—É—Ç–∏—Ä—É—é—â–∞—è –æ–ø–µ—Ä–∞—Ü–∏–∏
const deleteID = (user: User) => {
  delete user.id;
};

// –ò–º–º—É—Ç–∞–±–µ–ª—å–Ω–∞—è –æ–ø–µ—Ä–∞—Ü–∏—è
const removeID = ({ id, ...user }: User) => user;

// –ú—É—Ç–∏—Ä—É—é—â–∞—è –æ–ø–µ—Ä–∞—Ü–∏–∏
const deleteItem = (array: unknown[]) => {
    array.splice(2);
};

// –ò–º–º—É—Ç–∞–±–µ–ª—å–Ω–∞—è –æ–ø–µ—Ä–∞—Ü–∏—è
const removeItem = (array: unknown[]) => array.slice(2);
```

```ts
class Example {
  private id?: string;
  
  private deleteID = () => {
    this.id = undefined;
  }
}
```
