# ÐÐµÐ¹Ð¼Ð¸Ð½Ð³ Ð¿Ð°Ñ€Ð°Ð¼ÐµÑ‚Ñ€Ð¾Ð², ÑÐ²Ð¾Ð¹ÑÑ‚Ð²

## Boolean ÑÐ²Ð¾Ð¹ÑÑ‚Ð²Ð° Ð¸ Ð¿Ð°Ñ€Ð°Ð¼ÐµÑ‚Ñ€Ñ‹ Ð´Ð¾Ð»Ð¶Ð½Ñ‹ Ð¸Ð¼ÐµÑ‚ÑŒ Ð¿Ñ€ÐµÑ„Ð¸ÐºÑ, ÑƒÐºÐ°Ð·Ñ‹Ð²Ð°ÑŽÑ‰Ð¸Ð¹ Ð½Ð° Ð¿Ñ€Ð¸Ð½Ð°Ð´Ð»ÐµÐ¶Ð½Ð¾ÑÑ‚ÑŒ Ðº Boolean

**âœ¨ ÐœÐ¾Ñ‚Ð¸Ð²Ð°Ñ†Ð¸Ñ**

Ð˜Ð´ÐµÐ½Ñ‚Ð¸Ñ„Ð¸ÐºÐ°Ñ†Ð¸Ñ boolean ÑÐ²Ð¾Ð¹ÑÑ‚Ð² Ð¸ Ð¿Ð°Ñ€Ð°Ð¼ÐµÑ‚Ñ€Ð¾Ð² Ð¿Ð¾ Ð½Ð°Ð·Ð²Ð°Ð½Ð¸ÑŽ.

**ðŸ¤– ÐÐ²Ñ‚Ð¾Ð¼Ð°Ñ‚Ð¸Ð·Ð°Ñ†Ð¸Ñ**

ÐÐµ Ð¸Ð¼Ð¿Ð»ÐµÐ¼ÐµÐ½Ñ‚Ð¸Ñ€Ð¾Ð²Ð°Ð½Ð¾ Ð² eslint-config.

**âœ… Valid**

```ts
const calcSum = (product: Product, hasDiscount: boolean) => {}
```

```ts
type Props = {
  isOpen: boolean;
  isShow: boolean;
  wasRemoved: boolean;
  hasState: boolean;
};
```

```ts
class Product {
  hasDiscount: boolean = false;

  get isEmpty() {}
}
```

**âŒ Invalid**

```ts
const calcSum = (product: Product, discountIsPresent: boolean) => {}
```

```ts
type Props = {
  open: boolean;
  show: boolean;
  removed: boolean;
  stateIsPresent: boolean;
};
```

```ts
class Product {
  discountIsPresent: boolean = false;

  get empty() {}
}
```
