# –°—Ç—Ä—É–∫—Ç—É—Ä–∞ —Ç–µ—Å—Ç–æ–≤

## –î–ª—è —É—Ç–≤–µ—Ä–∂–¥–µ–Ω–∏–π –∏—Å–ø–æ–ª—å–∑—É–µ—Ç—Å—è it

**ü§ñ –ê–≤—Ç–æ–º–∞—Ç–∏–∑–∞—Ü–∏—è**

–ò–º–ø–ª–µ–º–µ–Ω—Ç–∏—Ä–æ–≤–∞–Ω–æ –≤ [@astral/eslint-config-vitest](https://www.npmjs.com/package/@astral/eslint-config-vitest).

**‚ú® –ú–æ—Ç–∏–≤–∞—Ü–∏—è**

–û–¥–Ω–æ—Ä–æ–¥–Ω–æ—Å—Ç—å —Ç–µ—Å—Ç–æ–≤.

**‚úÖ Valid**

```ts
describe('formatPhoneToView', () => {
  it('–ú–∞—Å–∫–∞ —Å–æ–¥–µ—Ä–∂–∏—Ç –ø–æ-—É–º–æ–ª—á–∞–Ω–∏—é +7');

  it('–í –º–∞—Å–∫—É –¥–æ–±–∞–≤–ª—è–µ—Ç—Å—è 8 –ø—Ä–∏ isStartWithPlus=false');
});
```

**‚ùå Invalid**

```ts
describe('formatPhoneToView', () => {
  test('–ú–∞—Å–∫–∞ —Å–æ–¥–µ—Ä–∂–∏—Ç –ø–æ-—É–º–æ–ª—á–∞–Ω–∏—é +7');

  test('–í –º–∞—Å–∫—É –¥–æ–±–∞–≤–ª—è–µ—Ç—Å—è 8 –ø—Ä–∏ isStartWithPlus=false');
});
```


## –î–æ–ø—É—Å—Ç–∏–º –ø–µ—Ä–≤—ã–π —É—Ä–æ–≤–µ–Ω—å –≤–ª–æ–∂–µ–Ω–Ω–æ—Å—Ç–∏ `describe`

**ü§ñ –ê–≤—Ç–æ–º–∞—Ç–∏–∑–∞—Ü–∏—è**

–ò–º–ø–ª–µ–º–µ–Ω—Ç–∏—Ä–æ–≤–∞–Ω–æ –≤ [@astral/eslint-config-vitest](https://www.npmjs.com/package/@astral/eslint-config-vitest).

**‚ú® –ú–æ—Ç–∏–≤–∞—Ü–∏—è**

–ü–æ–∑–≤–æ–ª—è–µ—Ç –∏—Å–∫–ª—é—á–∏—Ç—å —É—Å–ª–æ–∂–Ω–µ–Ω–∏—è —á—Ç–µ–Ω–∏—è —Ç–µ—Å—Ç–æ–≤ –ø—Ä–∏ –±–æ–ª—å—à–æ–π –≤–ª–æ–∂–µ–Ω–Ω–æ—Å—Ç–∏.

**‚úÖ Valid**

```ts
describe('CartRepository', () => {
  describe('–û–±—â–∏–π —Å—á–µ—Ç—á–∏–∫ —Ç–æ–≤–∞—Ä–æ–≤', () => {
    it('–£–≤–µ–ª–∏—á–∏–≤–∞–µ—Ç—Å—è –ø—Ä–∏ –¥–æ–±–∞–≤–ª–µ–Ω–∏–∏ —Ç–æ–≤–∞—Ä–æ–≤ –¥–æ —É—Å–ø–µ—à–Ω–æ–≥–æ –≤—ã–ø–æ–ª–Ω–µ–Ω–∏—è –∑–∞–ø—Ä–æ—Å–∞');

    it(
        '–û—Ç–∫–∞—Ç—ã–≤–∞–µ—Ç—Å—è –≤ –∏—Å—Ö–æ–¥–Ω–æ–µ –ø—Ä–∏ –æ—à–∏–±–∫–µ –∑–∞–ø—Ä–æ—Å–∞ –Ω–∞ –¥–æ–±–∞–≤–ª–µ–Ω–∏—è —Ç–æ–≤–∞—Ä–æ–≤ –≤ –∫–æ—Ä–∑–∏–Ω—É',
    );

    it('–£–º–µ–Ω—å—à–∞–µ—Ç—Å—è –ø—Ä–∏ —É–¥–∞–ª–µ–Ω–∏–∏ —Ç–æ–≤–∞—Ä–æ–≤ –¥–æ —É—Å–ø–µ—à–Ω–æ–≥–æ –≤—ã–ø–æ–ª–Ω–µ–Ω–∏—è –∑–∞–ø—Ä–æ—Å–∞');

    it(
        '–û—Ç–∫–∞—Ç—ã–≤–∞–µ—Ç—Å—è –≤ –∏—Å—Ö–æ–¥–Ω–æ–µ –ø—Ä–∏ –æ—à–∏–±–∫–µ –∑–∞–ø—Ä–æ—Å–∞ –Ω–∞ —É–¥–∞–ª–µ–Ω–∏–µ —Ç–æ–≤–∞—Ä–æ–≤ –∏–∑ –∫–æ—Ä–∑–∏–Ω—ã',
    );
  });
});
```

**‚ùå Invalid**

```ts
describe('CartRepository', () => {
  describe('–û–±—â–∏–π —Å—á–µ—Ç—á–∏–∫ —Ç–æ–≤–∞—Ä–æ–≤', () => {
    describe('–ü—Ä–∏ –¥–æ–±–∞–≤–ª–µ–Ω–∏–∏ —Ç–æ–≤–∞—Ä–æ–≤', () => {
      it('–£–≤–µ–ª–∏—á–∏–≤–∞–µ—Ç—Å—è –¥–æ —É—Å–ø–µ—à–Ω–æ–≥–æ –≤—ã–ø–æ–ª–Ω–µ–Ω–∏—è –∑–∞–ø—Ä–æ—Å–∞');
      it('–û—Ç–∫–∞—Ç—ã–≤–∞–µ—Ç—Å—è –≤ –∏—Å—Ö–æ–¥–Ω–æ–µ –ø—Ä–∏ –æ—à–∏–±–∫–µ –∑–∞–ø—Ä–æ—Å–∞');
    });

    describe('–ü—Ä–∏ —É–¥–∞–ª–µ–Ω–∏–∏ —Ç–æ–≤–∞—Ä–æ–≤', () => {
      it('–£–º–µ–Ω—å—à–∞–µ—Ç—Å—è –¥–æ —É—Å–ø–µ—à–Ω–æ–≥–æ –≤—ã–ø–æ–ª–Ω–µ–Ω–∏—è –∑–∞–ø—Ä–æ—Å–∞');
      it('–û—Ç–∫–∞—Ç—ã–≤–∞–µ—Ç—Å—è –≤ –∏—Å—Ö–æ–¥–Ω–æ–µ –ø—Ä–∏ –æ—à–∏–±–∫–µ –∑–∞–ø—Ä–æ—Å–∞');
    });
  });
});
```


## –¢–µ—Å—Ç–∏—Ä—É–µ–º–∞—è —Å—É—â–Ω–æ—Å—Ç—å –ø–æ–º–µ—á–∞–µ—Ç—Å—è –≤ —Ç–µ—Å—Ç–∞—Ö –∫–∞–∫ SUT

SUT - system under test.

**ü§ñ –ê–≤—Ç–æ–º–∞—Ç–∏–∑–∞—Ü–∏—è**

–ù–µ –∏–º–ø–ª–µ–º–µ–Ω—Ç–∏—Ä–æ–≤–∞–Ω–æ –≤ eslint-config.

**‚ú® –ú–æ—Ç–∏–≤–∞—Ü–∏—è**

–ü–æ–∑–≤–æ–ª—è–µ—Ç –≤–∏–∑—É–∞–ª—å–Ω–æ –≤ —Ç–µ—Å—Ç–µ –≤—ã–¥–µ–ª—è—Ç—å —Ç–µ—Å—Ç–∏—Ä—É–µ–º—É—é —Å—É—â–Ω–æ—Å—Ç—å.

**‚úÖ Valid**

```ts
describe('CreateBookScreenStore', () => {
  it('–£—Å–ø–µ—à–Ω–æ–µ —Å–æ–∑–¥–∞–Ω–∏–µ –∫–Ω–∏–≥–∏ –ø–æ–∫–∞–∑—ã–≤–∞–µ—Ç —É–≤–µ–¥–æ–º–ª–µ–Ω–∏–µ –æ–± —É—Å–ø–µ—à–Ω–æ—Å—Ç–∏', async () => {
    const notifyMock = mock<typeof notify>();
    const fakeBookFormValues = makeFakeBookFormValues({ name: '–ß–∏—Å—Ç—ã–π –∫–æ–¥' });
    const sut = new CreateBookScreenStore(notifyMock);

    await sut.createBook(fakeBookFormValues);

    expect(notifyMock.success).toBeCalledWith('–ß–∏—Å—Ç—ã–π –∫–æ–¥ —É—Å–ø–µ—à–Ω–æ —Å–æ–∑–¥–∞–Ω–∞');
  });
});
```

**‚ùå Invalid**

```ts
describe('CreateBookScreenStore', () => {
  it('–£—Å–ø–µ—à–Ω–æ–µ —Å–æ–∑–¥–∞–Ω–∏–µ –∫–Ω–∏–≥–∏ –ø–æ–∫–∞–∑—ã–≤–∞–µ—Ç —É–≤–µ–¥–æ–º–ª–µ–Ω–∏–µ –æ–± —É—Å–ø–µ—à–Ω–æ—Å—Ç–∏', async () => {
    const notifyMock = mock<typeof notify>();
    const fakeBookFormValues = makeFakeBookFormValues({ name: '–ß–∏—Å—Ç—ã–π –∫–æ–¥' });
    const bookScreenStore = new CreateBookScreenStore(notifyMock);

    await bookScreenStore.createBook(fakeBookFormValues);

    expect(notifyMock.success).toBeCalledWith('–ß–∏—Å—Ç—ã–π –∫–æ–¥ —É—Å–ø–µ—à–Ω–æ —Å–æ–∑–¥–∞–Ω–∞');
  });
});
```

### –†–µ–∑—É–ª—å—Ç–∞—Ç –≤—ã–ø–æ–ª–Ω–µ–Ω–∏—è —Ñ—É–Ω–∫—Ü–∏–∏ –Ω–µ —è–≤–ª—è–µ—Ç—Å—è SUT

**‚ùå Invalid**

```ts
describe('formatPriceToView', () => {
  it('–í —Ä–µ–∑—É–ª—å—Ç–∏—Ä—É—é—â—É—é —Å—Ç—Ä–æ–∫—É –¥–æ–±–∞–≤–ª—è–µ—Ç—Å—è –ø–æ—Å—Ç—Ñ–∏–∫—Å', () => {
    const sut = formatPriceToView(100);
      
    expect(sut).toBe('100 —Ä—É–±.');
  });
});
```

**‚úÖ Valid**

```ts
describe('formatPriceToView', () => {
  it('–í —Ä–µ–∑—É–ª—å—Ç–∏—Ä—É—é—â—É—é —Å—Ç—Ä–æ–∫—É –¥–æ–±–∞–≤–ª—è–µ—Ç—Å—è –ø–æ—Å—Ç—Ñ–∏–∫—Å', () => {
    const result = formatPriceToView(100);

    expect(result).toBe('100 —Ä—É–±.');
  });
});
```

## –°—Ç—Ä—É–∫—Ç—É—Ä–∞ —Ç–µ—Å—Ç–∞ –¥–æ–ª–∂–Ω—ã —Å–æ–æ—Ç–≤–µ—Ç—Å—Ç–≤–æ–≤–∞—Ç—å AAA (Arrange-Act-Assert)

- –ü–æ—Å–ª–µ–¥–æ–≤–∞—Ç–µ–ª—å–Ω–æ—Å—Ç—å AAA –¥–æ–ª–∂–Ω–∞ —Å–æ–±–ª—é–¥–∞—Ç—å—Å—è
- –ö–æ–º–º–µ–Ω—Ç–∞—Ä–∏–∏ –¥–ª—è –∫–∞–∂–¥–æ–≥–æ –±–ª–æ–∫–∞ AAA –¥–µ–ª–∞—Ç—å –Ω–µ –Ω–∞–¥–æ
- –û—Ç—Å—Ç—É–ø—ã –º–µ–∂–¥—É –±–ª–æ–∫–∞–º–∏ AAA - –Ω–µ –æ–±—è–∑–∞—Ç–µ–ª—å–Ω–æ–µ —É—Å–ª–æ–≤–∏–µ

**ü§ñ –ê–≤—Ç–æ–º–∞—Ç–∏–∑–∞—Ü–∏—è**

–ù–µ –∏–º–ø–ª–µ–º–µ–Ω—Ç–∏—Ä–æ–≤–∞–Ω–æ –≤ eslint-config.

**‚ú® –ú–æ—Ç–∏–≤–∞—Ü–∏—è**

–ü–æ–∑–≤–æ–ª—è–µ—Ç —Å—Ç–∞–Ω–¥–∞—Ä—Ç–∏–∑–∏—Ä–æ–≤–∞—Ç—å —Å—Ç—Ä—É–∫—Ç—É—Ä—ã —Ç–µ—Å—Ç–æ–≤.

**‚úÖ Valid**

```ts
it('–°–ø–∏—Å–æ–∫ —Ç–æ–≤–∞—Ä–æ–≤ –∏ —Å—á–µ—Ç—á–∏–∫ —Å–±—Ä–∞—Å—ã–≤–∞–µ—Ç—Å—è –ø–æ—Å–ª–µ —Ä–µ—Å–µ—Ç–∞', () => {
  const cartSourcesStub = mock<CartNetworkSources>();
  const sut = new CartRepository(cartSourcesStub, createCacheService());
  const goodsCountQuery = sut.getGoodsCountQuery();
  const goodsQuery = sut.getGoodsQuery();

  sut.resetCartCache();
  expect(goodsCountQuery.data).toBe(0);
  expect(goodsQuery.data).toEqual([]);
});
```

```tsx
it('OnChange –í—ã–∑—ã–≤–∞–µ—Ç—Å—è –±–µ–∑ –ø–µ—Ä–µ–¥–∞–Ω–Ω–æ–≥–æ isExpanded', async () => {
  const onChangeSpy = vi.fn();

  renderWithTheme(
    <Accordion
      summary="–¢–µ—Å—Ç"
      startAdorment={<InfoFillMd color="info" />}
      onChange={onChangeSpy}
    >
      <Typography>–ö–æ–Ω—Ç–µ–Ω—Ç</Typography>
    </Accordion>,
  );

  const title = screen.getByRole('button', { name: '–¢–µ—Å—Ç' });

  await userEvents.click(title);
  expect(onChangeSpy).toHaveBeenCalled();
});
```

**‚ùå Invalid**

–û—Å—Ç–∞–≤–ª—è—Ç—å –∫–æ–º–º–µ–Ω—Ç–∞—Ä–∏–∏ –¥–ª—è –±–ª–æ–∫–æ–≤ AAA –Ω–µ –Ω—É–∂–Ω–æ.

```ts
it('–°–ø–∏—Å–æ–∫ —Ç–æ–≤–∞—Ä–æ–≤ –∏ —Å—á–µ—Ç—á–∏–∫ —Å–±—Ä–∞—Å—ã–≤–∞–µ—Ç—Å—è –ø–æ—Å–ª–µ —Ä–µ—Å–µ—Ç–∞', () => {
  // Arrange
  const cartSourcesStub = mock<CartNetworkSources>();
  const sut = new CartRepository(cartSourcesStub, createCacheService());
  const goodsCountQuery = sut.getGoodsCountQuery();
  const goodsQuery = sut.getGoodsQuery();

  // Act
  sut.resetCartCache();

  // Assert
  expect(goodsCountQuery.data).toBe(0);
  expect(goodsQuery.data).toEqual([]);
});
```
