# Test Double


## –ú–æ–∫–∏ –∏ —Å—Ç–∞–±—ã –∏–º–µ—é—Ç –ø–æ—Å—Ç—Ñ–∏–∫—Å mock

**ü§ñ –ê–≤—Ç–æ–º–∞—Ç–∏–∑–∞—Ü–∏—è**

–ù–µ –∏–º–ø–ª–µ–º–µ–Ω—Ç–∏—Ä–æ–≤–∞–Ω–æ –≤ eslint-config.

**‚ú® –ú–æ—Ç–∏–≤–∞—Ü–∏—è**

–ò—Å–ø–æ–ª—å–∑–æ–≤–∞–Ω–∏–µ –¥–ª—è –º–æ–∫–æ–≤ –∏ —Å—Ç–∞–±–æ–≤ –æ–¥–Ω–æ –∏ —Ç–æ –∂–µ –æ–±–æ–∑–Ω–∞—á–µ–Ω–∏–µ –ø–æ–∑–≤–æ–ª—è–µ—Ç —É–ø—Ä–æ—Å—Ç–∏—Ç—å —Ä–µ–∞–ª–∏–∑–∞—Ü–∏—é —Ç–µ—Å—Ç–æ–≤.
–ù–µ –ø—Ä–∏—Ö–æ–¥–∏—Ç—Å—è –∑–∞–¥—É–º—ã–≤–∞—Ç—å –Ω–∞–¥ —Ä–∞–∑–ª–∏—á–∏–µ–º –ø—Ä–∏ —Ä–µ–∞–ª–∏–∑–∞—Ü–∏–∏ —Ç–µ—Å—Ç–æ–≤.

**‚úÖ Valid**

```ts
describe('CreateBookScreenStore', () => {
  it('–£—Å–ø–µ—à–Ω–æ–µ —Å–æ–∑–¥–∞–Ω–∏–µ –∫–Ω–∏–≥–∏ –ø–æ–∫–∞–∑—ã–≤–∞–µ—Ç —É–≤–µ–¥–æ–º–ª–µ–Ω–∏–µ –æ–± —É—Å–ø–µ—à–Ω–æ—Å—Ç–∏', async () => {
    const notifyMock = mock<typeof notify>();
    const fakeBookFormValues = makeFakeBookFormValues({ name: '–ß–∏—Å—Ç—ã–π –∫–æ–¥' });
    const sut = new CreateBookScreenStore(notifyMock);

    await sut.createBook(fakeBookFormValues);

    expect(notifyMock.success).toBeCalledWith('–ß–∏—Å—Ç—ã–π –∫–æ–¥ —É—Å–ø–µ—à–Ω–æ —Å–æ–∑–¥–∞–Ω–∞');
  });
});
```

–í –ø—Ä–∏–º–µ—Ä–µ –Ω–∏–∂–µ cartSourcesMock —è–≤–ª—è–µ—Ç—Å—è stub, –Ω–æ –¥–ª—è —É–ø—Ä–æ—â–µ–Ω–∏—è –Ω–µ–æ–±—Ö–æ–¥–∏–º–æ –∏—Å–ø–æ–ª—å–∑–æ–≤–∞—Ç—å –ø–æ—Å—Ç—Ñ–∏–∫—Å mock.

```ts
describe('CartRepository', () => {
  it('–°–ø–∏—Å–æ–∫ —Ç–æ–≤–∞—Ä–æ–≤ –∏ —Å—á–µ—Ç—á–∏–∫ —Å–±—Ä–∞—Å—ã–≤–∞–µ—Ç—Å—è –ø–æ—Å–ª–µ —Ä–µ—Å–µ—Ç–∞', () => {
    const cartSourcesMock = mock<CartNetworkSources>();

    const sut = new CartRepository(cartSourcesMock, createCacheService());
    const goodsCountQuery = sut.getGoodsCountQuery();
    const goodsQuery = sut.getGoodsQuery();

    sut.resetCartCache();
    expect(goodsCountQuery.data).toBe(0);
    expect(goodsQuery.data).toEqual([]);
  });
});
```

## Spy (—à–ø–∏–æ–Ω—ã) –∏–º–µ—é—Ç –ø–æ—Å—Ç—Ñ–∏–∫—Å `spy`

**ü§ñ –ê–≤—Ç–æ–º–∞—Ç–∏–∑–∞—Ü–∏—è**

–ù–µ –∏–º–ø–ª–µ–º–µ–Ω—Ç–∏—Ä–æ–≤–∞–Ω–æ –≤ eslint-config.

**‚ú® –ú–æ—Ç–∏–≤–∞—Ü–∏—è**

–ü–æ–∑–≤–æ–ª—è–µ—Ç –æ—Ç–ª–∏—á–∞—Ç—å spy –æ—Ç –ø—Ä–æ—Å—Ç—ã—Ö —Ñ—É–Ω–∫—Ü–∏–π –∏ –¥—Ä—É–≥–∏—Ö —Ç–µ—Å—Ç–æ–≤—ã—Ö –¥–≤–æ–π–Ω–∏–∫–æ–≤.

**‚úÖ Valid**

```tsx
it('OnChange –≤—ã–∑—ã–≤–∞–µ—Ç—Å—è –±–µ–∑ –ø–µ—Ä–µ–¥–∞–Ω–Ω–æ–≥–æ isExpanded', async () => {
  const onChangeSpy = vi.fn();

  renderWithTheme(
    <Accordion
      summary="–¢–µ—Å—Ç"
      startAdorment={<InfoFillMd color="info" />}
      onChange={onChangeSpy}
    >
      <Typography>–ö–æ–Ω—Ç–µ–Ω—Ç</Typography>
    </Accordion>,
  );

  const title = screen.getByRole('button', { name: '–¢–µ—Å—Ç' });

  await userEvents.click(title);
  expect(onChangeSpy).toHaveBeenCalled();
});
```

**‚ùå Invalid**

```tsx
it('OnChange –≤—ã–∑—ã–≤–∞–µ—Ç—Å—è –±–µ–∑ –ø–µ—Ä–µ–¥–∞–Ω–Ω–æ–≥–æ isExpanded', async () => {
  const onChange = vi.fn();

  renderWithTheme(
    <Accordion
        summary="–¢–µ—Å—Ç"
        startAdorment={<InfoFillMd color="info" />}
        onChange={onChange}
    >
      <Typography>–ö–æ–Ω—Ç–µ–Ω—Ç</Typography>
    </Accordion>,
  );

  const title = screen.getByRole('button', { name: '–¢–µ—Å—Ç' });

  await userEvents.click(title);
  expect(onChange).toHaveBeenCalled();
});
```
