# UIStore

ÐŸÑ€Ð°Ð²Ð¸Ð»Ð° Ð´Ð»Ñ stores, ÑÐ¾Ð´ÐµÑ€Ð¶Ð°Ñ‰Ð¸Ñ… Ð»Ð¾Ð³Ð¸ÐºÑƒ ÐºÐ¾Ð¼Ð¿Ð¾Ð½ÐµÐ½Ñ‚Ð°.

## UIStore Ð´Ð¾Ð»Ð¶ÐµÐ½ Ð½Ð°Ñ…Ð¾Ð´Ð¸Ñ‚ÑÑ Ð²Ð½ÑƒÑ‚Ñ€Ð¸ Ð´Ð¸Ñ€ÐµÐºÑ‚Ð¾Ñ€Ð¸Ð¸ ÐºÐ¾Ð¼Ð¿Ð¾Ð½ÐµÐ½Ñ‚Ð° Ð² Ð¾Ñ‚Ð´ÐµÐ»ÑŒÐ½Ð¾Ð¹ Ð´Ð¸Ñ€ÐµÐºÑ‚Ð¾Ñ€Ð¸Ð¸

**âœ¨ ÐœÐ¾Ñ‚Ð¸Ð²Ð°Ñ†Ð¸Ñ**

- ÐŸÐ¾Ð·Ð²Ð¾Ð»ÑÐµÑ‚ ÑƒÑÑ‚Ð°Ð½Ð°Ð²Ð»Ð¸Ð²Ð°Ñ‚ÑŒ Ð²Ð¸Ð·ÑƒÐ°Ð»ÑŒÐ½ÑƒÑŽ ÑÐ²ÑÐ·ÑŒ react-ÐºÐ¾Ð¼Ð¿Ð¾Ð½ÐµÐ½Ñ‚Ð° Ð¸ store Ð´Ð»Ñ Ð½ÐµÐ³Ð¾
- ÐŸÑ€Ð¸ Ð¿ÐµÑ€ÐµÐ½Ð¾ÑÐµ ÐºÐ¾Ð¼Ð¿Ð¾Ð½ÐµÐ½Ñ‚Ð° Ð² Ð´Ñ€ÑƒÐ³ÑƒÑŽ Ð´Ð¸Ñ€ÐµÐºÑ‚Ð¾Ñ€Ð¸ÑŽ Ð²Ð¼ÐµÑÑ‚Ðµ Ñ Ð½Ð¸Ð¼ Ð±ÑƒÐ´ÐµÑ‚ Ð¿ÐµÑ€ÐµÐ½ÐµÑÐµÐ½ Ð¸ store
- ÐŸÐ¾Ð·Ð²Ð¾Ð»ÑÐµÑ‚ Ð¿ÐµÑ€ÐµÐ½Ð¾ÑÐ¸Ñ‚ÑŒ store ÑÐ¾ ÑÐ²ÑÐ·Ð°Ð½Ð½Ñ‹Ð¼Ð¸ ÑÑƒÑ‰Ð½Ð¾ÑÑ‚ÑÐ¼Ð¸ Ð² Ð´Ñ€ÑƒÐ³Ð¸Ðµ ÑÐ»Ð¾Ð¸ Ð¿Ñ€Ð¸Ð»Ð¾Ð¶ÐµÐ½Ð¸Ñ Ð±ÐµÐ· Ð½ÐµÐ¾Ð±Ñ…Ð¾Ð´Ð¸Ð¼Ð¾ÑÑ‚Ð¸ Ñ€ÐµÑ„Ð°ÐºÑ‚Ð¾Ñ€Ð¸Ð½Ð³Ð°

**ðŸ¤– ÐÐ²Ñ‚Ð¾Ð¼Ð°Ñ‚Ð¸Ð·Ð°Ñ†Ð¸Ñ**

- [Codemode](https://github.com/kaluga-astral/style-guide/tree/main/codemodes/UIStore/README.md) Ð¿Ð¾Ð·Ð²Ð¾Ð»Ð¸Ñ‚ Ð°Ð²Ñ‚Ð¾Ð¼Ð°Ñ‚Ð¸Ð·Ð¸Ñ€Ð¾Ð²Ð°Ñ‚ÑŒ Ñ€ÐµÑ„Ð°ÐºÑ‚Ð¾Ñ€Ð¸Ð½Ð³ Ð¿Ñ€Ð¸ Ð¿ÐµÑ€ÐµÑ…Ð¾Ð´Ðµ Ð½Ð° `UIStore`
- ÐÐµ Ð¸Ð¼Ð¿Ð»ÐµÐ¼ÐµÐ½Ñ‚Ð¸Ñ€Ð¾Ð²Ð°Ð½Ð¾ Ð² eslint-config

**âœ… Valid**

```
â”œâ”€â”€ Cart/
|    â”œâ”€â”€ UIStore/ 
|    |    |â”€â”€ UIStore.ts
|    |    â””â”€â”€ index.ts
|    â”œâ”€â”€ Cart.tsx
|    â””â”€â”€ index.ts
```

**âŒ Invalid**

```
â”œâ”€â”€ CartStore/
|    â”œâ”€â”€ CartStore.ts
|    â””â”€â”€ index.ts
â”œâ”€â”€ Cart/
|    â”œâ”€â”€ Cart.tsx
|    â””â”€â”€ index.ts
```

```
â”œâ”€â”€ Cart/
|    â”œâ”€â”€ stores
|    |    â”œâ”€â”€ UIStore/
|    |    â””â”€â”€ index.ts
|    â”œâ”€â”€ Cart.tsx
|    â””â”€â”€ index.ts
```

```
â”œâ”€â”€ Cart/
|    â”œâ”€â”€ store.ts
|    â”œâ”€â”€ Cart.tsx
|    â””â”€â”€ index.ts
```

## UIStore Ð¸ ÐµÐ³Ð¾ Ð´Ð¸Ñ€ÐµÐºÑ‚Ð¾Ñ€Ð¸Ñ Ð´Ð¾Ð»Ð¶Ð½Ñ‹ Ð½Ð°Ð·Ñ‹Ð²Ð°Ñ‚ÑŒÑÑ `UIStore`

**âœ¨ ÐœÐ¾Ñ‚Ð¸Ð²Ð°Ñ†Ð¸Ñ**

ÐŸÐ¾Ð·Ð²Ð¾Ð»ÑÐµÑ‚ Ð¸Ð·Ð±ÐµÐ¶Ð°Ñ‚ÑŒ Ð´Ð»Ð¸Ð½Ð½Ñ‹Ñ… Ð¸Ð¼ÐµÐ½ Ð´Ð»Ñ stores, ÑÐ¾Ð´ÐµÑ€Ð¶Ð°Ñ‰Ð¸Ñ… UI Ð»Ð¾Ð³Ð¸ÐºÑƒ.

**ðŸ¤– ÐÐ²Ñ‚Ð¾Ð¼Ð°Ñ‚Ð¸Ð·Ð°Ñ†Ð¸Ñ**

- [Codemode](https://github.com/kaluga-astral/style-guide/tree/main/codemodes/UIStore/README.md) Ð¿Ð¾Ð·Ð²Ð¾Ð»Ð¸Ñ‚ Ð°Ð²Ñ‚Ð¾Ð¼Ð°Ñ‚Ð¸Ð·Ð¸Ñ€Ð¾Ð²Ð°Ñ‚ÑŒ Ñ€ÐµÑ„Ð°ÐºÑ‚Ð¾Ñ€Ð¸Ð½Ð³ Ð¿Ñ€Ð¸ Ð¿ÐµÑ€ÐµÑ…Ð¾Ð´Ðµ Ð½Ð° `UIStore`
- ÐÐµ Ð¸Ð¼Ð¿Ð»ÐµÐ¼ÐµÐ½Ñ‚Ð¸Ñ€Ð¾Ð²Ð°Ð½Ð¾ Ð² eslint-config

**âœ… Valid**

```
â”œâ”€â”€ Cart/
|    â”œâ”€â”€ UIStore/ 
|    |    |â”€â”€ UIStore.ts
|    |    â””â”€â”€ index.ts
|    â”œâ”€â”€ Cart.tsx
|    â””â”€â”€ index.ts
```

```Cart/UIStore/UIStore.ts```
```ts
export class UIStore {};

export const createUIStore = () => new UIStore();
```

**âŒ Invalid**

```
â”œâ”€â”€ Cart/
|    â”œâ”€â”€ store/ 
|    |    â””â”€â”€ store.ts
|    |    â””â”€â”€ index.ts
|    â”œâ”€â”€ Cart.tsx
|    â””â”€â”€ index.ts
```

```Cart/store/store.ts```
```ts
export class Store {};
```

---

```
â”œâ”€â”€ Cart/
|    â”œâ”€â”€ CartStore/ 
|    |    â””â”€â”€ CartStore.ts
|    |    â””â”€â”€ index.ts
|    â”œâ”€â”€ Cart.tsx
|    â””â”€â”€ index.ts
```

```Cart/CartStore/CartStore.ts```
```ts
export class CartStore {};
```

## Stores, ÑÐ²Ð»ÑÑŽÑ‰Ð¸ÐµÑÑ Ñ‡Ð°ÑÑ‚ÑŒÑŽ Ð¸Ð¼Ð¿Ð»ÐµÐ¼ÐµÐ½Ñ‚Ð°Ñ†Ð¸Ð¸ UIStore Ð´Ð¾Ð»Ð¶Ð½Ñ‹ Ð½Ð°Ñ…Ð¾Ð´Ð¸Ñ‚ÑÑ Ð² Ð´Ð¸Ñ€ÐµÐºÑ‚Ð¾Ñ€Ð¸Ð¸ UIStore

**âœ¨ ÐœÐ¾Ñ‚Ð¸Ð²Ð°Ñ†Ð¸Ñ**

Ð£ÑÑ‚Ð°Ð½Ð¾Ð²ÐºÐ° Ð¾Ð´Ð½Ð¾Ð·Ð½Ð°Ñ‡Ð½Ð¾Ð¹ ÑÐ²ÑÐ·Ð¸ UIStore Ð¸ stores, ÑÐ²Ð»ÑÑŽÑ‰Ð¸Ñ…ÑÑ Ñ‡Ð°ÑÑ‚ÑŒÑŽ Ð¸Ð¼Ð¿Ð»ÐµÐ¼ÐµÐ½Ñ‚Ð°Ñ†Ð¸Ð¸ UIStore.

**ðŸ¤– ÐÐ²Ñ‚Ð¾Ð¼Ð°Ñ‚Ð¸Ð·Ð°Ñ†Ð¸Ñ**

ÐÐµ Ð¸Ð¼Ð¿Ð»ÐµÐ¼ÐµÐ½Ñ‚Ð¸Ñ€Ð¾Ð²Ð°Ð½Ð¾ Ð² eslint-config

**âœ… Valid**

```
â”œâ”€â”€ Cart/
|    â”œâ”€â”€ UIStore/ 
|    |    |â”€â”€ stores/
|    |    |    |â”€â”€ SwitchStore/
|    |    |    â””â”€â”€ index.ts
|    |    |â”€â”€ UIStore.ts
|    |    â””â”€â”€ index.ts
|    â”œâ”€â”€ Cart.tsx
|    â””â”€â”€ index.ts
```

```Cart/UIStore/UIStore.ts```
```ts
import { SwitchStore } from './stores';

export class UIStore {
  constructor(private readonly switchStore: SwitchStore) {}
};
```

**âŒ Invalid**

Stores ÑÐ²Ð»ÑÑŽÑ‚ÑÑ Ñ‡Ð°ÑÑ‚ÑŒÑŽ Ð¸Ð¼Ð¿Ð»ÐµÐ¼ÐµÐ½Ñ‚Ð°Ñ†Ð¸Ð¸ `UIStore`, Ð½Ð¾ Ð¿Ñ€Ð¸ ÑÑ‚Ð¾Ð¼ Ð½Ð°Ñ…Ð¾Ð´ÑÑ‚ÑÑ Ð² ÐºÐ¾Ñ€Ð½Ðµ ÐºÐ¾Ð¼Ð¿Ð¾Ð½ÐµÐ½Ñ‚Ð°:

```
â”œâ”€â”€ Cart/
|    |â”€â”€ stores/
|    |    |â”€â”€ SwitchStore/
|    |    â””â”€â”€ index.ts
|    â”œâ”€â”€ UIStore/ 
|    |    |â”€â”€ UIStore.ts
|    |    â””â”€â”€ index.ts
|    â”œâ”€â”€ Cart.tsx
|    â””â”€â”€ index.ts
```

```Cart/UIStore/UIStore.ts```
```ts
import { SwitchStore } from '../stores';

export class Store {
    constructor(private readonly switchStore: SwitchStore) {}
};
```

## Ð˜Ð¼ÐµÐ½Ð° Ð¼ÐµÑ‚Ð¾Ð´Ð¾Ð² UIStore Ð½Ðµ Ð´Ð¾Ð»Ð¶Ð½Ñ‹ Ð¿Ð¾Ð´Ñ€Ð°Ð¶Ð°Ñ‚ÑŒ Ð½ÐµÐ¹Ð¼Ð¸Ð½Ð³Ñƒ Ð¾Ð±Ñ€Ð°Ð±Ð¾Ñ‚Ñ‡Ð¸ÐºÐ¾Ð² ÐºÐ¾Ð¼Ð¿Ð¾Ð½ÐµÐ½Ñ‚Ð¾Ð²

Ð’ react-ÐºÐ¾Ð¼Ð¿Ð¾Ð½ÐµÐ½Ñ‚Ð°Ñ… Ð´Ð»Ñ Ð¾Ð±Ñ€Ð°Ð±Ð¾Ñ‚Ñ‡Ð¸ÐºÐ¾Ð² [Ð¸ÑÐ¿Ð¾Ð»ÑŒÐ·ÑƒÑŽÑ‚ÑÑ Ð¿Ñ€ÐµÑ„Ð¸ÐºÑÑ‹ on Ð¸ handle](../../../react/logic#Ð¾Ð±Ñ€Ð°Ð±Ð¾Ñ‚Ñ‡Ð¸ÐºÐ¸-Ñ-Ð¿Ñ€ÐµÑ„Ð¸ÐºÑÐ¾Ð¼-handle).
Ð’ UIStore ÑÑ‚Ð¸ Ð¿Ñ€ÐµÑ„Ð¸ÐºÑÑ‹ Ð´Ð¾Ð»Ð¶Ð½Ñ‹ Ð¸ÑÐ¿Ð¾Ð»ÑŒÐ·Ð¾Ð²Ð°Ñ‚ÑŒÑÑ Ñ‚Ð¾Ð»ÑŒÐºÐ¾ Ð¿Ñ€Ð¸ ÑÐµÐ¼Ð°Ð½Ñ‚Ð¸Ñ‡ÐµÑÐºÐ¾Ð¹ Ð½ÐµÐ¾Ð±Ñ…Ð¾Ð´Ð¸Ð¼Ð¾ÑÑ‚Ð¸.

**âœ¨ ÐœÐ¾Ñ‚Ð¸Ð²Ð°Ñ†Ð¸Ñ**

UIStore Ð½Ðµ Ð·Ð°Ð²Ð¸ÑÐ¸Ñ‚ Ð¾Ñ‚ react-ÐºÐ¾Ð¼Ð¿Ð¾Ð½ÐµÐ½Ñ‚Ð°.
ÐœÐµÑ‚Ð¾Ð´Ñ‹ UIStore Ð½Ðµ ÑÐ²Ð»ÑÑŽÑ‚ÑÑ Ð¾Ð±Ñ€Ð°Ð±Ð¾Ñ‚Ñ‡Ð¸ÐºÐ°Ð¼Ð¸ Ð´Ð»Ñ react-ÐºÐ¾Ð¼Ð¿Ð¾Ð½ÐµÐ½Ñ‚Ð¾Ð².

**ðŸ¤– ÐÐ²Ñ‚Ð¾Ð¼Ð°Ñ‚Ð¸Ð·Ð°Ñ†Ð¸Ñ**

ÐÐµ Ð¸Ð¼Ð¿Ð»ÐµÐ¼ÐµÐ½Ñ‚Ð¸Ñ€Ð¾Ð²Ð°Ð½Ð¾ Ð² eslint-config

**âœ… Valid**

```
â”œâ”€â”€ Cart/
|    â”œâ”€â”€ UIStore/ 
|    |    |â”€â”€ UIStore.ts
|    |    â””â”€â”€ index.ts
|    â”œâ”€â”€ Cart.tsx
|    â””â”€â”€ index.ts
```

```Cart/UIStore/UIStore.ts```
```ts
import { SwitchStore } from './stores';

class UIStore {
  constructor() {}
  
  public deleteItem = () => {};
};

export const createUIStore = () => new UIStore();
```

```Cart/Cart/Cart.tsx```
```tsx
import { useState } from 'react';
import { createUIStore } from './UIStore';

const DeleteItem = () => {
  const [{ deleteItem }] = useState(createUIStore);

  return (
    <Button onClick={deleteItem}>Ð£Ð´Ð°Ð»Ð¸Ñ‚ÑŒ</Button>
  );
};
```

**âŒ Invalid**

```
â”œâ”€â”€ Cart/
|    â”œâ”€â”€ UIStore/ 
|    |    |â”€â”€ UIStore.ts
|    |    â””â”€â”€ index.ts
|    â”œâ”€â”€ Cart.tsx
|    â””â”€â”€ index.ts
```

```Cart/UIStore/UIStore.ts```
```ts
import { SwitchStore } from './stores';

class UIStore {
  constructor() {}
  
  // "handle" Ð¿Ñ€ÐµÑ„Ð¸ÐºÑ Ð² Ð´Ð°Ð½Ð½Ð¾Ð¼ ÑÐ»ÑƒÑ‡Ð°Ðµ ÑƒÐºÐ°Ð·Ñ‹Ð²Ð°ÐµÑ‚ Ð½Ð° Ñ‚Ð¾, Ñ‡Ñ‚Ð¾ ÑÑ‚Ð¾ Ð¾Ð±Ñ€Ð°Ð±Ð¾Ñ‚Ñ‡Ð¸Ðº Ð´Ð»Ñ react-ÐºÐ¾Ð¼Ð¿Ð¾Ð½ÐµÐ½Ñ‚Ð°
  // UIStore Ð½Ð¸Ñ‡ÐµÐ³Ð¾ Ð½Ðµ Ð·Ð½Ð°ÐµÑ‚ Ð¾ UI, Ð¿Ð¾ÑÑ‚Ð¾Ð¼Ñƒ Ð¿Ñ€ÐµÑ„Ð¸ÐºÑÐ° Ð² Ð´Ð°Ð½Ð½Ð¾Ð¼ ÑÐ»ÑƒÑ‡Ð°Ðµ Ð±Ñ‹Ñ‚ÑŒ Ð½Ðµ Ð´Ð¾Ð»Ð¶Ð½Ð¾
  public handleDeleteItem = () => {};
};

export const createUIStore = () => new UIStore();
```

```Cart/Cart/Cart.tsx```
```tsx
import { useState } from 'react';
import { createUIStore } from './UIStore';

const DeleteItem = () => {
  const [{ handleDeleteItem }] = useState(createUIStore);

  return (
    <Button onClick={handleDeleteItem}>Ð£Ð´Ð°Ð»Ð¸Ñ‚ÑŒ</Button>
  );
};
```
