# ะกะพะทะดะฐะฝะธะต ะบะพะผะฟะพะฝะตะฝัะพะฒ

## ะะฐะถะดัะน ะบะพะผะฟะพะฝะตะฝั ะดะพะปะถะตะฝ ะฝะฐัะพะดะธััั ะฒ ะพัะดะตะปัะฝะพะน ะดะธัะตะบัะพัะธะธ

**โจ ะะพัะธะฒะฐัะธั**

ะัะดะตะปัะฝะฐั ะดะธัะตะบัะพัะธั ะดะปั ะบะฐะถะดะพะณะพ ะบะพะผะฟะพะฝะตะฝัะฐ ะฟะพะทะฒะพะปัะตั:
- ะะธะดะตัั ะพะดะฝะพะทะฝะฐัะฝัั ัะฒัะทั ะบะพะผะฟะพะฝะตะฝัะฐ ัะพ ัะฒัะทะฐะฝะฝัะผะธ ัััะฝะพัััะผะธ:
  - Utils
  - Stores
  - Hooks
  - Constants
  - Styles
  - Types
  - ะะปะพะถะตะฝะฝัะต ะบะพะผะฟะพะฝะตะฝัั
  - ...
- ะะธะดะตัั ะพะดะฝะพะทะฝะฐัะฝัั ัะฒัะทั ะบะพะผะฟะพะฝะตะฝัะฐ ั ะตะณะพ ัะตััะฐะผะธ
- ะะตะบะพะผะฟะพะทะธัะพะฒะฐัั ะธ ัะฐััะธัััั ัััะฝะพััะธ, ะฒัะพะดััะธะต ะฒ ัะพััะฐะฒ ะบะพะผะฟะพะฝะตะฝัะฐ, ะฑะตะท ะฝะตะพะฑัะพะดะธะผะพััะธ ัะตัะฐะบัะพัะธะฝะณะฐ (ัะพะทะดะฐะฝะธะต ะพัะดะตะปัะฝะพะน ะดะธัะตะบัะพัะธะธ ะดะปั ะบะพะผะฟะพะฝะตะฝัะฐ)
- ะะตัะตะฝะพัะธัั ะบะพะผะฟะพะฝะตะฝัั ัะพ ัะฒัะทะฐะฝะฝัะผะธ ัััะฝะพัััะผะธ ะฒ ะดััะณะธะต ัะปะพะธ ะฟัะธะปะพะถะตะฝะธั ะฑะตะท ะฝะตะพะฑัะพะดะธะผะพััะธ ัะตัะฐะบัะพัะธะฝะณะฐ ะบะพะผะฟะพะฝะตะฝัะฐ

**๐ค ะะฒัะพะผะฐัะธะทะฐัะธั**

ะะต ะธะผะฟะปะตะผะตะฝัะธัะพะฒะฐะฝะพ ะฒ eslint-config.

**โ Valid**

```
โโโ UserInfo/
|    โโโ Header/
|    |    |โโ hooks/
|    |    |โโ utils/
|    |    |โโ Header.tsx
|    |    โโโ index.ts
|    โโโ UserInfo.tsx
|    โโโ UserInfo.test.tsx
|    โโโ constants.ts
|    โโโ types.ts
|    โโโ index.ts
โโโ Button/
|    โโโ Button.tsx
|    โโโ index.ts
```

## ะัะต utils ะบะพะผะฟะพะฝะตะฝัะฐ ะฒัะฝะพััััั ะฒ ะดะธัะตะบัะพัะธั `utils`

ะะพะปะฝัะต ะฟัะฐะฒะธะปะฐ ะดะปั utils ัะผะพััะธัะต [ะทะดะตัั](../utils).

**โจ ะะพัะธะฒะฐัะธั**

ะะพะทะฒะพะปัะตั ัะฐะทะณัะฐะฝะธัะธัั ะฟัะพะณัะฐะผะผะฝัะต ัััะฝะพััะธ ะบะพะผะฟะพะฝะตะฝัะฐ.

**๐ค ะะฒัะพะผะฐัะธะทะฐัะธั**

ะะต ะธะผะฟะปะตะผะตะฝัะธัะพะฒะฐะฝะพ ะฒ eslint-config.

**โ Valid**

```
โโโ UserInfo/
|    โโโ utils/
|    |    |โโ formatPriceToView/
|    |    |    |โโ formatPriceToView.ts
|    |    |    โโโ index.ts
|    |    โโโ index.ts
|    โโโ UserInfo.tsx
|    โโโ index.ts
```

**โ Invalid**

```
โโโ UserInfo/
|    โโโ formatPriceToView.ts
|    โโโ UserInfo.tsx
|    โโโ index.ts
```

```
โโโ UserInfo/
|    โโโ utils.ts
|    โโโ UserInfo.tsx
|    โโโ index.ts
```

```
โโโ UserInfo/
|    โโโ utils/
|    |    |โโ utils.ts
|    |    โโโ index.ts
|    โโโ UserInfo.tsx
|    โโโ index.ts
```

## ะัะต hooks ะบะพะผะฟะพะฝะตะฝัะฐ ะฒัะฝะพััััั ะฒ ะดะธัะตะบัะพัะธั `hooks`

ะัะต hooks ะบะพะผะฟะพะฝะตะฝัะฐ ะฒัะฝะพััััั ะฒ ะดะธัะตะบัะพัะธั `hooks`, ะฟัะธ ััะพะผ **ะบะฐะถะดัะน hook ะฝะฐัะพะดะธััั ะฒ ัะฒะพะตะน ะดะธัะตะบัะพัะธะธ**.

**โจ ะะพัะธะฒะฐัะธั**

ะัะดะตะปัะฝะฐั ะดะธัะตะบัะพัะธั ะดะปั ะบะฐะถะดะพะณะพ hook ะฟะพะทะฒะพะปัะตั:
- ะะธะดะตัั ะพะดะฝะพะทะฝะฐัะฝัั ัะฒัะทั hook ัะพ ัะฒัะทะฐะฝะฝัะผะธ ัััะฝะพัััะผะธ:
  - Utils
  - Constants
  - Types
  - ...
- ะะธะดะตัั ะพะดะฝะพะทะฝะฐัะฝัั ัะฒัะทั hook ั ะตะณะพ ัะตััะฐะผะธ
- ะะตะบะพะผะฟะพะทะธัะพะฒะฐัั ะธ ัะฐััะธัััั ัััะฝะพััะธ, ะฒัะพะดััะธะต ะฒ ัะพััะฐะฒ ะบะพะผะฟะพะฝะตะฝัะฐ, ะฑะตะท ะฝะตะพะฑัะพะดะธะผะพััะธ ัะตัะฐะบัะพัะธะฝะณะฐ (ัะพะทะดะฐะฝะธะต ะพัะดะตะปัะฝะพะน ะดะธัะตะบัะพัะธะธ ะดะปั hook)
- ะะตัะตะฝะพัะธัั hook ัะพ ัะฒัะทะฐะฝะฝัะผะธ ัััะฝะพัััะผะธ ะฒ ะดััะณะธะต ัะปะพะธ ะฟัะธะปะพะถะตะฝะธั ะฑะตะท ะฝะตะพะฑัะพะดะธะผะพััะธ ัะตัะฐะบัะพัะธะฝะณะฐ

**๐ค ะะฒัะพะผะฐัะธะทะฐัะธั**

ะะต ะธะผะฟะปะตะผะตะฝัะธัะพะฒะฐะฝะพ ะฒ eslint-config.

**โ Valid**

```
โโโ UserInfo/
|    โโโ hooks/
|    |    |โโ useUserData/
|    |    |    |โโ utils/
|    |    |    |โโ useUserData.ts
|    |    |    |โโ useUserData.test.ts
|    |    |    |โโ constants.ts
|    |    |    |โโ types.ts
|    |    |    โโโ index.ts
|    |    โโโ index.ts
|    โโโ UserInfo.tsx
|    โโโ index.ts
```


**โ Invalid**

```
โโโ UserInfo/
|    โโโ useUserData.ts
|    โโโ UserInfo.tsx
|    โโโ index.ts
```

```
โโโ UserInfo/
|    โโโ hooks.ts
|    โโโ UserInfo.tsx
|    โโโ index.ts
```

```
โโโ UserInfo/
|    โโโ hooks/
|    |    |โโ hooks.ts
|    |    โโโ index.ts
|    โโโ UserInfo.tsx
|    โโโ index.ts
```
