# –°–æ–∑–¥–∞–Ω–∏–µ –∫–æ–º–ø–æ–Ω–µ–Ω—Ç–æ–≤

## –ö–∞–∂–¥—ã–π –∫–æ–º–ø–æ–Ω–µ–Ω—Ç –¥–æ–ª–∂–µ–Ω –Ω–∞—Ö–æ–¥–∏—Ç—Å—è –≤ –æ—Ç–¥–µ–ª—å–Ω–æ–π –¥–∏—Ä–µ–∫—Ç–æ—Ä–∏–∏

**‚ú® –ú–æ—Ç–∏–≤–∞—Ü–∏—è**

–û—Ç–¥–µ–ª—å–Ω–∞—è –¥–∏—Ä–µ–∫—Ç–æ—Ä–∏—è –¥–ª—è –∫–∞–∂–¥–æ–≥–æ –∫–æ–º–ø–æ–Ω–µ–Ω—Ç–∞ –ø–æ–∑–≤–æ–ª—è–µ—Ç:
- –í–∏–¥–µ—Ç—å –æ–¥–Ω–æ–∑–Ω–∞—á–Ω—É—é —Å–≤—è–∑—å –∫–æ–º–ø–æ–Ω–µ–Ω—Ç–∞ —Å–æ —Å–≤—è–∑–∞–Ω–Ω—ã–º–∏ —Å—É—â–Ω–æ—Å—Ç—è–º–∏:
  - Utils
  - Stores
  - Hooks
  - Constants
  - Styles
  - Types
  - –í–ª–æ–∂–µ–Ω–Ω—ã–µ –∫–æ–º–ø–æ–Ω–µ–Ω—Ç—ã
  - ...
- –í–∏–¥–µ—Ç—å –æ–¥–Ω–æ–∑–Ω–∞—á–Ω—É—é —Å–≤—è–∑—å –∫–æ–º–ø–æ–Ω–µ–Ω—Ç–∞ —Å –µ–≥–æ —Ç–µ—Å—Ç–∞–º–∏
- –î–µ–∫–æ–º–ø–æ–∑–∏—Ä–æ–≤–∞—Ç—å –∏ —Ä–∞—Å—à–∏—Ä—è—Ç—å —Å—É—â–Ω–æ—Å—Ç–∏, –≤—Ö–æ–¥—è—â–∏–µ –≤ —Å–æ—Å—Ç–∞–≤ –∫–æ–º–ø–æ–Ω–µ–Ω—Ç–∞, –±–µ–∑ –Ω–µ–æ–±—Ö–æ–¥–∏–º–æ—Å—Ç–∏ —Ä–µ—Ñ–∞–∫—Ç–æ—Ä–∏–Ω–≥–∞ (—Å–æ–∑–¥–∞–Ω–∏–µ –æ—Ç–¥–µ–ª—å–Ω–æ–π –¥–∏—Ä–µ–∫—Ç–æ—Ä–∏–∏ –¥–ª—è –∫–æ–º–ø–æ–Ω–µ–Ω—Ç–∞)
- –ü–µ—Ä–µ–Ω–æ—Å–∏—Ç—å –∫–æ–º–ø–æ–Ω–µ–Ω—Ç—ã —Å–æ —Å–≤—è–∑–∞–Ω–Ω—ã–º–∏ —Å—É—â–Ω–æ—Å—Ç—è–º–∏ –≤ –¥—Ä—É–≥–∏–µ —Å–ª–æ–∏ –ø—Ä–∏–ª–æ–∂–µ–Ω–∏—è –±–µ–∑ –Ω–µ–æ–±—Ö–æ–¥–∏–º–æ—Å—Ç–∏ —Ä–µ—Ñ–∞–∫—Ç–æ—Ä–∏–Ω–≥–∞ –∫–æ–º–ø–æ–Ω–µ–Ω—Ç–∞

**ü§ñ –ê–≤—Ç–æ–º–∞—Ç–∏–∑–∞—Ü–∏—è**

–ù–µ –∏–º–ø–ª–µ–º–µ–Ω—Ç–∏—Ä–æ–≤–∞–Ω–æ –≤ eslint-config.

**‚úÖ Valid**

```
‚îú‚îÄ‚îÄ UserInfo/
|    ‚îú‚îÄ‚îÄ Header/
|    |    |‚îÄ‚îÄ hooks/
|    |    |‚îÄ‚îÄ utils/
|    |    |‚îÄ‚îÄ Header.tsx
|    |    ‚îî‚îÄ‚îÄ index.ts
|    ‚îú‚îÄ‚îÄ UserInfo.tsx
|    ‚îú‚îÄ‚îÄ UserInfo.test.tsx
|    ‚îú‚îÄ‚îÄ constants.ts
|    ‚îú‚îÄ‚îÄ types.ts
|    ‚îî‚îÄ‚îÄ index.ts
‚îú‚îÄ‚îÄ Button/
|    ‚îú‚îÄ‚îÄ Button.tsx
|    ‚îî‚îÄ‚îÄ index.ts
```

## –ò–º–µ–Ω–∞ –∫–æ–º–ø–æ–Ω–µ–Ω—Ç–æ–≤ –∏ –∏—Ö –¥–∏—Ä–µ–∫—Ç–æ—Ä–∏–π –¥–æ–ª–∂–Ω—ã —Å–æ–æ—Ç–≤–µ—Ç—Å—Ç–≤–æ–≤–∞—Ç—å PascalCase

**‚ú® –ú–æ—Ç–∏–≤–∞—Ü–∏—è**

–°—Ç–∞–Ω–¥–∞—Ä—Ç–∏–∑–∞—Ü–∏—è –∏–º–µ–Ω –∫–æ–º–ø–æ–Ω–µ–Ω—Ç–æ–≤.

**ü§ñ –ê–≤—Ç–æ–º–∞—Ç–∏–∑–∞—Ü–∏—è**

–ù–µ –∏–º–ø–ª–µ–º–µ–Ω—Ç–∏—Ä–æ–≤–∞–Ω–æ –≤ eslint-config.

**‚úÖ Valid**

```
‚îú‚îÄ‚îÄ UserInfo/
|    ‚îú‚îÄ‚îÄ UserInfo.tsx
|    ‚îî‚îÄ‚îÄ index.ts
‚îú‚îÄ‚îÄ ButtonGroup/
|    ‚îú‚îÄ‚îÄ ButtonGroup.tsx
|    ‚îî‚îÄ‚îÄ index.ts
```

**‚ùå Invalid**

```
‚îú‚îÄ‚îÄ userInfo/
|    ‚îú‚îÄ‚îÄ userInfo.tsx
|    ‚îî‚îÄ‚îÄ index.ts
‚îú‚îÄ‚îÄ button-group/
|    ‚îú‚îÄ‚îÄ button-group.tsx
|    ‚îî‚îÄ‚îÄ index.ts
```

## –î–ª—è `React.Fragment` –ø—Ä–µ–¥–ø–æ—á—Ç–∏—Ç–µ–ª—å–Ω–∞ –∫—Ä–∞—Ç–∫–∞—è –Ω–æ—Ç–∞—Ü–∏—è

–ò—Å–∫–ª—é—á–µ–Ω–∏–µ, –∫–æ–≥–¥–∞ –¥–æ–ª–∂–Ω–∞ –∏—Å–ø–æ–ª—å–∑–æ–≤–∞—Ç—å—Å—è —Ä–∞—Å—à–∏—Ä–µ–Ω–Ω–∞—è –Ω–æ—Ç–∞—Ü–∏—è - –Ω–µ–æ–±—Ö–æ–¥–∏–º `key` prop.

**‚ú® –ú–æ—Ç–∏–≤–∞—Ü–∏—è**

–£–º–µ–Ω—å—à–µ–Ω–∏–µ –∫–æ–ª–∏—á–µ—Å—Ç–≤–∞ –∫–æ–¥–∞ –≤ –∫–æ–º–ø–æ–Ω–µ–Ω—Ç–µ.

**ü§ñ –ê–≤—Ç–æ–º–∞—Ç–∏–∑–∞—Ü–∏—è**

–ò–º–ø–ª–µ–º–µ–Ω—Ç–∏—Ä–æ–≤–∞–Ω–æ –≤ [@astral/eslint-config-react-ts](https://www.npmjs.com/package/@astral/eslint-config-react-ts).

**‚úÖ Valid**

```tsx
const Info = () => {
  return (
    <>
      <Typography>–ò–º—è</Typography>
      <Typography>–§–∞–º–∏–ª–∏—è</Typography>
    </>
  );
};
```

```tsx
export const List = ({ list }: Props) => {
  return list.map(({ name, surname }) => {
    <Fragment key={name}>
      <Typography>–ò–º—è</Typography>
      <Typography>–§–∞–º–∏–ª–∏—è</Typography>
    </Fragment>;
  });
};
```

**‚ùå Invalid**

```tsx
export const Info = () => {
  return (
    <Fragment>
      <Typography>–ò–º—è</Typography>
      <Typography>–§–∞–º–∏–ª–∏—è</Typography>
    </Fragment>
  );
};
```

