# Hooks

## Ð’ÑÐµ hooks ÐºÐ¾Ð¼Ð¿Ð¾Ð½ÐµÐ½Ñ‚Ð° Ð²Ñ‹Ð½Ð¾ÑÑÑ‚ÑÑ Ð² Ð´Ð¸Ñ€ÐµÐºÑ‚Ð¾Ñ€Ð¸ÑŽ `hooks`

Ð’ÑÐµ hooks ÐºÐ¾Ð¼Ð¿Ð¾Ð½ÐµÐ½Ñ‚Ð° Ð²Ñ‹Ð½Ð¾ÑÑÑ‚ÑÑ Ð² Ð´Ð¸Ñ€ÐµÐºÑ‚Ð¾Ñ€Ð¸ÑŽ `hooks`, Ð¿Ñ€Ð¸ ÑÑ‚Ð¾Ð¼ **ÐºÐ°Ð¶Ð´Ñ‹Ð¹ hook Ð½Ð°Ñ…Ð¾Ð´Ð¸Ñ‚ÑÑ Ð² ÑÐ²Ð¾ÐµÐ¹ Ð´Ð¸Ñ€ÐµÐºÑ‚Ð¾Ñ€Ð¸Ð¸**.

**âœ¨ ÐœÐ¾Ñ‚Ð¸Ð²Ð°Ñ†Ð¸Ñ**

ÐžÑ‚Ð´ÐµÐ»ÑŒÐ½Ð°Ñ Ð´Ð¸Ñ€ÐµÐºÑ‚Ð¾Ñ€Ð¸Ñ Ð´Ð»Ñ ÐºÐ°Ð¶Ð´Ð¾Ð³Ð¾ hook Ð¿Ð¾Ð·Ð²Ð¾Ð»ÑÐµÑ‚:
- Ð’Ð¸Ð´ÐµÑ‚ÑŒ Ð¾Ð´Ð½Ð¾Ð·Ð½Ð°Ñ‡Ð½ÑƒÑŽ ÑÐ²ÑÐ·ÑŒ hook ÑÐ¾ ÑÐ²ÑÐ·Ð°Ð½Ð½Ñ‹Ð¼Ð¸ ÑÑƒÑ‰Ð½Ð¾ÑÑ‚ÑÐ¼Ð¸:
  - Utils
  - Constants
  - Types
  - ...
- Ð’Ð¸Ð´ÐµÑ‚ÑŒ Ð¾Ð´Ð½Ð¾Ð·Ð½Ð°Ñ‡Ð½ÑƒÑŽ ÑÐ²ÑÐ·ÑŒ hook Ñ ÐµÐ³Ð¾ Ñ‚ÐµÑÑ‚Ð°Ð¼Ð¸
- Ð”ÐµÐºÐ¾Ð¼Ð¿Ð¾Ð·Ð¸Ñ€Ð¾Ð²Ð°Ñ‚ÑŒ Ð¸ Ñ€Ð°ÑÑˆÐ¸Ñ€ÑÑ‚ÑŒ ÑÑƒÑ‰Ð½Ð¾ÑÑ‚Ð¸, Ð²Ñ…Ð¾Ð´ÑÑ‰Ð¸Ðµ Ð² ÑÐ¾ÑÑ‚Ð°Ð² ÐºÐ¾Ð¼Ð¿Ð¾Ð½ÐµÐ½Ñ‚Ð°, Ð±ÐµÐ· Ð½ÐµÐ¾Ð±Ñ…Ð¾Ð´Ð¸Ð¼Ð¾ÑÑ‚Ð¸ Ñ€ÐµÑ„Ð°ÐºÑ‚Ð¾Ñ€Ð¸Ð½Ð³Ð° (ÑÐ¾Ð·Ð´Ð°Ð½Ð¸Ðµ Ð¾Ñ‚Ð´ÐµÐ»ÑŒÐ½Ð¾Ð¹ Ð´Ð¸Ñ€ÐµÐºÑ‚Ð¾Ñ€Ð¸Ð¸ Ð´Ð»Ñ hook)
- ÐŸÐµÑ€ÐµÐ½Ð¾ÑÐ¸Ñ‚ÑŒ hook ÑÐ¾ ÑÐ²ÑÐ·Ð°Ð½Ð½Ñ‹Ð¼Ð¸ ÑÑƒÑ‰Ð½Ð¾ÑÑ‚ÑÐ¼Ð¸ Ð² Ð´Ñ€ÑƒÐ³Ð¸Ðµ ÑÐ»Ð¾Ð¸ Ð¿Ñ€Ð¸Ð»Ð¾Ð¶ÐµÐ½Ð¸Ñ Ð±ÐµÐ· Ð½ÐµÐ¾Ð±Ñ…Ð¾Ð´Ð¸Ð¼Ð¾ÑÑ‚Ð¸ Ñ€ÐµÑ„Ð°ÐºÑ‚Ð¾Ñ€Ð¸Ð½Ð³Ð°

**ðŸ¤– ÐÐ²Ñ‚Ð¾Ð¼Ð°Ñ‚Ð¸Ð·Ð°Ñ†Ð¸Ñ**

ÐÐµ Ð¸Ð¼Ð¿Ð»ÐµÐ¼ÐµÐ½Ñ‚Ð¸Ñ€Ð¾Ð²Ð°Ð½Ð¾ Ð² eslint-config.

**âœ… Valid**

```
â”œâ”€â”€ UserInfo/
|    â”œâ”€â”€ hooks/
|    |    |â”€â”€ useUserData/
|    |    |    |â”€â”€ utils/
|    |    |    |â”€â”€ useUserData.ts
|    |    |    |â”€â”€ useUserData.test.ts
|    |    |    |â”€â”€ constants.ts
|    |    |    |â”€â”€ types.ts
|    |    |    â””â”€â”€ index.ts
|    |    â””â”€â”€ index.ts
|    â”œâ”€â”€ UserInfo.tsx
|    â””â”€â”€ index.ts
```


**âŒ Invalid**

```
â”œâ”€â”€ UserInfo/
|    â”œâ”€â”€ useUserData.ts
|    â”œâ”€â”€ UserInfo.tsx
|    â””â”€â”€ index.ts
```

```
â”œâ”€â”€ UserInfo/
|    â”œâ”€â”€ hooks.ts
|    â”œâ”€â”€ UserInfo.tsx
|    â””â”€â”€ index.ts
```

```
â”œâ”€â”€ UserInfo/
|    â”œâ”€â”€ hooks/
|    |    |â”€â”€ hooks.ts
|    |    â””â”€â”€ index.ts
|    â”œâ”€â”€ UserInfo.tsx
|    â””â”€â”€ index.ts
```

## Ð¢Ð¸Ð¿ Ð²Ñ…Ð¾Ð´Ð½Ñ‹Ñ… Ð¿Ð°Ñ€Ð°Ð¼ÐµÑ‚Ñ€Ð¾Ð² Ñ…ÑƒÐºÐ° Ð´Ð¾Ð»Ð¶ÐµÐ½ Ð¸Ð¼ÐµÑ‚ÑŒ Ð¿Ð¾ÑÑ‚Ñ„Ð¸ÐºÑ `Params`

**âœ¨ ÐœÐ¾Ñ‚Ð¸Ð²Ð°Ñ†Ð¸Ñ**

ÐžÐ´Ð½Ð¾Ð·Ð½Ð°Ñ‡Ð½Ð°Ñ Ð¸Ð´ÐµÐ½Ñ‚Ð¸Ñ„Ð¸ÐºÐ°Ñ†Ð¸Ñ Ð²Ñ…Ð¾Ð´Ð½Ñ‹Ñ… Ð¿Ð°Ñ€Ð°Ð¼ÐµÑ‚Ñ€Ð¾Ð² Ñ…ÑƒÐºÐ°.

**ðŸ¤– ÐÐ²Ñ‚Ð¾Ð¼Ð°Ñ‚Ð¸Ð·Ð°Ñ†Ð¸Ñ**

ÐÐµ Ð¸Ð¼Ð¿Ð»ÐµÐ¼ÐµÐ½Ñ‚Ð¸Ñ€Ð¾Ð²Ð°Ð½Ð¾ Ð² eslint-config.

**âœ… Valid**

```ts
type UseLogicParams = {
  data: unknown[];
};

const useLogic = (params: UseLogicParams) => {};
```

```ts
// Ð”Ð¾Ð¿ÑƒÑÑ‚Ð¸Ð¼Ð¾ Ð¿Ñ€Ð¾ÑÑ‚Ð¾ "Params", ÐµÑÐ»Ð¸ Ñ‚Ð¸Ð¿ Ð½Ðµ ÑÐºÑÐ¿Ð¾Ñ€Ñ‚Ð¸Ñ€ÑƒÐµÑ‚ÑÑ Ð¸Ð»Ð¸ Ð½ÐµÑ‚ ÐºÐ¾Ð»Ð»Ð¸Ð·Ð¸Ð¹ Ð¸Ð¼ÐµÐ½ Ð²Ð½ÑƒÑ‚Ñ€Ð¸ es Ð¼Ð¾Ð´ÑƒÐ»Ñ
type Params = {
  data: unknown[];
};

const useLogic = (params: Params) => {};
```

## Ð¢Ð¸Ð¿ Ð²Ð¾Ð·Ð²Ñ€Ð°Ñ‰Ð°ÐµÐ¼Ð¾Ð³Ð¾ Ð·Ð½Ð°Ñ‡ÐµÐ½Ð¸Ñ Ñ…ÑƒÐºÐ° Ð´Ð¾Ð»Ð¶ÐµÐ½ Ð¸Ð¼ÐµÑ‚ÑŒ Ð¿Ð¾ÑÑ‚Ñ„Ð¸ÐºÑ `Result`

**âœ¨ ÐœÐ¾Ñ‚Ð¸Ð²Ð°Ñ†Ð¸Ñ**

ÐžÐ´Ð½Ð¾Ð·Ð½Ð°Ñ‡Ð½Ð°Ñ Ð¸Ð´ÐµÐ½Ñ‚Ð¸Ñ„Ð¸ÐºÐ°Ñ†Ð¸Ñ Ð²Ð¾Ð·Ð²Ñ€Ð°Ñ‰Ð°ÐµÐ¼Ð¾Ð³Ð¾ Ð·Ð½Ð°Ñ‡ÐµÐ½Ð¸Ñ Ñ…ÑƒÐºÐ°.

**ðŸ¤– ÐÐ²Ñ‚Ð¾Ð¼Ð°Ñ‚Ð¸Ð·Ð°Ñ†Ð¸Ñ**

ÐÐµ Ð¸Ð¼Ð¿Ð»ÐµÐ¼ÐµÐ½Ñ‚Ð¸Ñ€Ð¾Ð²Ð°Ð½Ð¾ Ð² eslint-config.

**âœ… Valid**

```ts
type UseLogicResult = {
  isShow: boolean;
};

const useLogic = (params: UseLogicParams): UseLogicResult => { ... };
```

```ts
// Ð”Ð¾Ð¿ÑƒÑÑ‚Ð¸Ð¼Ð¾ Ð¿Ñ€Ð¾ÑÑ‚Ð¾ "Result", ÐµÑÐ»Ð¸ Ñ‚Ð¸Ð¿ Ð½Ðµ ÑÐºÑÐ¿Ð¾Ñ€Ñ‚Ð¸Ñ€ÑƒÐµÑ‚ÑÑ Ð¸Ð»Ð¸ Ð½ÐµÑ‚ ÐºÐ¾Ð»Ð»Ð¸Ð·Ð¸Ð¹ Ð¸Ð¼ÐµÐ½ Ð²Ð½ÑƒÑ‚Ñ€Ð¸ es Ð¼Ð¾Ð´ÑƒÐ»Ñ
type Result = {
    isShow: boolean;
};

const useLogic = (params: Params): Result => { ... };
```
