# Hooks

## ะัะต hooks ะบะพะผะฟะพะฝะตะฝัะฐ ะฒัะฝะพััััั ะฒ ะดะธัะตะบัะพัะธั `hooks`

ะัะต hooks ะบะพะผะฟะพะฝะตะฝัะฐ ะฒัะฝะพััััั ะฒ ะดะธัะตะบัะพัะธั `hooks`, ะฟัะธ ััะพะผ **ะบะฐะถะดัะน hook ะฝะฐัะพะดะธััั ะฒ ัะฒะพะตะน ะดะธัะตะบัะพัะธะธ**.

**โจ ะะพัะธะฒะฐัะธั**

ะัะดะตะปัะฝะฐั ะดะธัะตะบัะพัะธั ะดะปั ะบะฐะถะดะพะณะพ hook ะฟะพะทะฒะพะปัะตั:
- ะะธะดะตัั ะพะดะฝะพะทะฝะฐัะฝัั ัะฒัะทั hook ัะพ ัะฒัะทะฐะฝะฝัะผะธ ัััะฝะพัััะผะธ:
  - Utils
  - Constants
  - Types
  - ...
- ะะธะดะตัั ะพะดะฝะพะทะฝะฐัะฝัั ัะฒัะทั hook ั ะตะณะพ ัะตััะฐะผะธ
- ะะตะบะพะผะฟะพะทะธัะพะฒะฐัั ะธ ัะฐััะธัััั ัััะฝะพััะธ, ะฒัะพะดััะธะต ะฒ ัะพััะฐะฒ ะบะพะผะฟะพะฝะตะฝัะฐ, ะฑะตะท ะฝะตะพะฑัะพะดะธะผะพััะธ ัะตัะฐะบัะพัะธะฝะณะฐ (ัะพะทะดะฐะฝะธะต ะพัะดะตะปัะฝะพะน ะดะธัะตะบัะพัะธะธ ะดะปั hook)
- ะะตัะตะฝะพัะธัั hook ัะพ ัะฒัะทะฐะฝะฝัะผะธ ัััะฝะพัััะผะธ ะฒ ะดััะณะธะต ัะปะพะธ ะฟัะธะปะพะถะตะฝะธั ะฑะตะท ะฝะตะพะฑัะพะดะธะผะพััะธ ัะตัะฐะบัะพัะธะฝะณะฐ

**๐ค ะะฒัะพะผะฐัะธะทะฐัะธั**

ะะต ะธะผะฟะปะตะผะตะฝัะธัะพะฒะฐะฝะพ ะฒ eslint-config.

**โ Valid**

```
โโโ UserInfo/
|    โโโ hooks/
|    |    |โโ useUserData/
|    |    |    |โโ utils/
|    |    |    |โโ useUserData.ts
|    |    |    |โโ useUserData.test.ts
|    |    |    |โโ constants.ts
|    |    |    |โโ types.ts
|    |    |    โโโ index.ts
|    |    โโโ index.ts
|    โโโ UserInfo.tsx
|    โโโ index.ts
```


**โ Invalid**

```
โโโ UserInfo/
|    โโโ useUserData.ts
|    โโโ UserInfo.tsx
|    โโโ index.ts
```

```
โโโ UserInfo/
|    โโโ hooks.ts
|    โโโ UserInfo.tsx
|    โโโ index.ts
```

```
โโโ UserInfo/
|    โโโ hooks/
|    |    |โโ hooks.ts
|    |    โโโ index.ts
|    โโโ UserInfo.tsx
|    โโโ index.ts
```
