# –õ–æ–≥–∏–∫–∞ –≤ –∫–æ–º–ø–æ–Ω–µ–Ω—Ç–∞—Ö

## –û–±—Ä–∞–±–æ—Ç—á–∏–∫–∏ —Å –ø—Ä–µ—Ñ–∏–∫—Å–æ–º `handle`

**‚ú® –ú–æ—Ç–∏–≤–∞—Ü–∏—è**

–ò–¥–µ–Ω—Ç–∏—Ñ–∏–∫–∞—Ü–∏—è –æ–±—Ä–∞–±–æ—Ç—á–∏–∫–æ–≤ react.

**ü§ñ –ê–≤—Ç–æ–º–∞—Ç–∏–∑–∞—Ü–∏—è**

–ù–µ –∏–º–ø–ª–µ–º–µ–Ω—Ç–∏—Ä–æ–≤–∞–Ω–æ –≤ eslint-config.

**‚úÖ Valid**

```tsx
const Cart = ({ list, onSetSum }: Props) => {
  // calcSum - –Ω–µ –æ–±—Ä–∞–±–æ—Ç—á–∏–∫, –ø–æ—ç—Ç–æ–º—É –ø—Ä–µ—Ñ–∏–∫—Å –Ω–µ –Ω—É–∂–µ–Ω
  const calcSum = () => list.reduce((sum, { price }) => sum + price, 0);

  const handleClickPay = () => {
    onSetSum(calcSum(list));
  };

  return (
    <section>
      <Button onClick={handleClickButton}>–û–ø–ª–∞—Ç–∏—Ç—å</Button>
    </section>
  );
};

```

**‚ùå Invalid**

```tsx

const Cart = ({ list, onSetSum }: Props) => {
  // –≠—Ç–æ –Ω–µ –æ–±—Ä–∞–±–æ—Ç—á–∏–∫
  const hadleCalcSum = () => list.reduce((sum, { price }) => sum + price, 0);

  // –û–±—Ä–∞–±–æ—Ç—á–∏–∫ –Ω–µ –ø–æ —Ñ–æ—Ä–º–∞—Ç—É
  const clickPayHandler = () => {
    onSetSum(calcSum(list));
  };

  return (
    <section>
      <Button onClick={clickPayHandler}>–û–ø–ª–∞—Ç–∏—Ç—å</Button>
    </section>
  );
};
```

## –ó–∞–ø—Ä–µ—â–µ–Ω—ã –¥–ª–∏–Ω–Ω—ã–µ —É—Å–ª–æ–≤–∏—è –≤ `jsx` —Ä–∞–∑–º–µ—Ç–∫–µ

**‚ú® –ú–æ—Ç–∏–≤–∞—Ü–∏—è**

–î–ª–∏–Ω–Ω—ã–µ —É—Å–ª–æ–≤–∏—è —É—Å–ª–æ–∂–Ω—è—é—Ç –≤–æ—Å–ø—Ä–∏—è—Ç–∏–µ `jsx` –∫–æ–º–ø–æ–Ω–µ–Ω—Ç–∞.

**ü§ñ –ê–≤—Ç–æ–º–∞—Ç–∏–∑–∞—Ü–∏—è**

–ù–µ –∏–º–ø–ª–µ–º–µ–Ω—Ç–∏—Ä–æ–≤–∞–Ω–æ –≤ eslint-config.

**‚ùå Invalid**

```tsx
const Cart = ({ list, isSuccess, userName, onPay }: Props) => {
  return (
    <section>
      {(Boolean(list.length) && isSuccess) ||
        (userName.startsWith('Vasya') && (
          <Button onClick={onPay}>–û–ø–ª–∞—Ç–∏—Ç—å</Button>
        ))}
    </section>
  );
};
```

**‚úÖ Valid**

```tsx
const Cart = ({ list, isSuccess, userName, onPay }: Props) => {
  const isVasya = userName.startsWith('Vasya');
  const isShowPayButton = (Boolean(list.length) && isSuccess) || isVasya;

  return (
    <section>
      {isShowPayButton && <Button onClick={onPay}>–û–ø–ª–∞—Ç–∏—Ç—å</Button>}
    </section>
  );
};
```


## –ó–∞–ø—Ä–µ—â–µ–Ω–∞ –ª–æ–≥–∏–∫–∞ –≤ —Ü–∏–∫–ª–∞—Ö –≤–Ω—É—Ç—Ä–∏ `jsx` —Ä–∞–∑–º–µ—Ç–∫–∏

–õ–æ–≥–∏–∫—É –≤ —Ü–∏–∫–ª–∞—Ö –Ω–µ–æ–±—Ö–æ–¥–∏–º–æ –ø–µ—Ä–µ–Ω–æ—Å–∏—Ç—å –≤ –æ—Ç–¥–µ–ª—å–Ω—ã–π –∫–æ–º–ø–æ–Ω–µ–Ω—Ç –∏–ª–∏ –≤ —Ç–µ–ª–æ —Ñ—É–Ω–∫—Ü–∏–∏ –∫–æ–º–ø–æ–Ω–µ–Ω—Ç–∞.

**‚ú® –ú–æ—Ç–∏–≤–∞—Ü–∏—è**

–£–ø—Ä–æ—â–µ–Ω–∏–µ –≤–æ—Å–ø—Ä–∏—è—Ç–∏—è –∏ –ø–æ–¥–¥–µ—Ä–∂–∫–∏ `jsx` –∫–æ–º–ø–æ–Ω–µ–Ω—Ç–∞.

**ü§ñ –ê–≤—Ç–æ–º–∞—Ç–∏–∑–∞—Ü–∏—è**

–ù–µ –∏–º–ø–ª–µ–º–µ–Ω—Ç–∏—Ä–æ–≤–∞–Ω–æ –≤ eslint-config.


**‚ùå Invalid**

```tsx
const MainPage = ({ list }: Props) => {
  return (
    <section>
      <Header />
      <Filters />
      <ul>
        {list.map((item) => {
          // –°–ª–æ–∂–Ω–æ—Å—Ç—å –≤–æ—Å–ø—Ä–∏—è—Ç–∏—è –∏ –ø–æ–¥–¥–µ—Ä–∂–∫–∏ –±—É–¥–µ—Ç —É–≤–µ–ª–∏—á–∏–≤–∞—Ç—å—Å—è –ø–æ –º–µ—Ä–µ —Ä–æ—Å—Ç–∞ –ª–æ–≥–∏–∫–∏ –∫–æ–º–ø–æ–Ω–µ–Ω—Ç–∞
          const price = formatPriceToView(item.price);
          const nickName = [item.name, item.surname].join(' ');

          return (
            <li>
              <Typography>{price}</Typography>
              <Typography>{nickName}</Typography>
            </li>
          );
        })}
      </ul>
      </section>
  );
};
```

**‚úÖ Valid**

```tsx
const ListItem = ({ itemInfo }: ListItemProps) => {
  const price = formatPriceToView(item.price);
  const nickName = [item.name, item.surname].join(' ');

  return (
    <li>
      <Typography>{price}</Typography>
      <Typography>{nickName}</Typography>
    </li>
  );
};

const MainPage = ({ list }: MainPageProps) => {
  return (
    <section>
      <Header />
      <Filters />
      <ul>
        {list.map((item) => (
            <ListItem itemInfo={item} />
        ))}
      </ul>
    </section>
  );
};
```

