# Props

## –¢–∏–ø—ã Props –¥–æ–ª–∂–Ω—ã –Ω–∞—Ö–æ–¥–∏—Ç—Å—è –≤ —Ñ–∞–π–ª–µ –∫–æ–º–ø–æ–Ω–µ–Ω—Ç–∞

**‚ú® –ú–æ—Ç–∏–≤–∞—Ü–∏—è**

–ù–∞—Ö–æ–∂–¥–µ–Ω–∏–µ Props —Ç–∏–ø–æ–≤ –≤ —Ñ–∞–π–ª–µ –∫–æ–º–ø–æ–Ω–µ–Ω—Ç–∞ –ø–æ–∑–≤–æ–ª—è–µ—Ç:
- –ò–∑–±–µ–∂–∞—Ç—å –æ–≤–µ—Ä—Ö–µ–¥–∞ –ø–æ —Å–æ–∑–¥–∞–Ω–∏—é –æ—Ç–¥–µ–ª—å–Ω–æ–≥–æ —Ñ–∞–π–ª–∞ –¥–ª—è —Ç–∏–ø–æ–≤ Props
- –ò–∑–±–µ–∂–∞—Ç—å [–ø—Ä–æ–±–ª–µ–º –æ—à–∏–±–æ—á–Ω–æ–≥–æ –Ω–∞–ø—Ä–∞–≤–ª–µ–Ω–∏—è –∑–∞–≤–∏—Å–∏–º–æ—Å—Ç–µ–π –∫–æ–º–ø–æ–Ω–µ–Ω—Ç–æ–≤](#–ø—Ä–æ–±–ª–µ–º–∞-–Ω–µ–ø—Ä–∞–≤–∏–ª—å–Ω–æ–≥–æ-–Ω–∞–ø—Ä–∞–≤–ª–µ–Ω–∏—è-–∑–∞–≤–∏—Å–∏–º–æ—Å—Ç–µ–π)

**ü§ñ –ê–≤—Ç–æ–º–∞—Ç–∏–∑–∞—Ü–∏—è**

–ù–µ –∏–º–ø–ª–µ–º–µ–Ω—Ç–∏—Ä–æ–≤–∞–Ω–æ –≤ eslint-config.

**‚úÖ Valid**

```UserInfo.tsx```
```tsx
type Props = {
  title: string;
  userName: string;
  onClick: () => void;
};

export const UserInfo = ({ title, userName, onClick }: Props) => {
  return (
    <Grid>
      <Typography>{title}</Typography>
      <Typography>{userName}</Typography>
      <Button onClick={onClick}>–ü–æ–∫–∞–∑–∞—Ç—å</Button>
    </Grid>
  );
};
```

**‚ùå Invalid**

```UserInfo/UserInfo.tsx```
```tsx
import { type UserInfoProps } from './types';

export const UserInfo = ({ title, userName, onClick }: ButtonProps) => {
  return (
    <Grid>
      <Typography>{title}</Typography>
      <Typography>{userName}</Typography>
      <Button onClick={onClick}>–ü–æ–∫–∞–∑–∞—Ç—å</Button>
    </Grid>
  );
};
```

```UserInfo/types.ts```
```ts
export type UserInfoProps = {
  title: string;
  userName: string;
  onClick: () => void;
};
```

### –ü—Ä–æ–±–ª–µ–º–∞ –Ω–µ–ø—Ä–∞–≤–∏–ª—å–Ω–æ–≥–æ –Ω–∞–ø—Ä–∞–≤–ª–µ–Ω–∏—è –∑–∞–≤–∏—Å–∏–º–æ—Å—Ç–µ–π –∫–æ–º–ø–æ–Ω–µ–Ω—Ç–æ–≤

–í—ã–Ω–µ—Å–µ–Ω–∏–µ Props –≤ –æ—Ç–¥–µ–ª—å–Ω—ã–π —Ñ–∞–π–ª ```types.ts``` –º–æ–∂–µ—Ç –ø—Ä–∏–≤–æ–¥–∏—Ç—å –∫ —Ç–æ–º—É, —á—Ç–æ Props –¥–æ—á–µ—Ä–Ω–∏—Ö –∫–æ–º–ø–æ–Ω–µ–Ω—Ç–æ–≤ –±—É–¥—É—Ç –∑–∞–≤–∏—Å–µ—Ç—å –æ—Ç Props —Ä–æ–¥–∏—Ç–µ–ª—è:

```
‚îú‚îÄ‚îÄ UserCard/
|    ‚îú‚îÄ‚îÄ Header/ 
|    |    ‚îî‚îÄ‚îÄ Header.tsx
|    |    ‚îî‚îÄ‚îÄ types.ts
|    |    ‚îî‚îÄ‚îÄ index.ts
|    ‚îú‚îÄ‚îÄ UserCard.tsx
|    ‚îú‚îÄ‚îÄ types.ts
|    ‚îî‚îÄ‚îÄ index.ts
```

```UserCard/types.ts```
```ts
export type UserCardProps = {
  headerTitle: ReactNode;
};
```

```HeaderProps``` –∑–∞–≤–∏—Å–∏—Ç –æ—Ç Props —Ä–æ–¥–∏—Ç–µ–ª—å—Å–∫–æ–≥–æ –∫–æ–º–ø–æ–Ω–µ–Ω—Ç–∞ ```UserCardProps```:

```UserCard/Header/types.ts```
```ts
import { type UserCardProps } from '../types';

export type HeaderProps = {
    title: UserCardProps['headerTitle'];
};
```

–ü—Ä–∏ —ç—Ç–æ–º –∫–æ–º–ø–æ–Ω–µ–Ω—Ç ```UserCard``` –∑–∞–≤–∏—Å–∏—Ç –æ—Ç –∫–æ–º–ø–æ–Ω–µ–Ω—Ç–∞ ```Header``` (–∏—Å–ø–æ–ª—å–∑—É–µ—Ç –µ–≥–æ –≤–Ω—É—Ç—Ä–∏):
```tsx
export const UserCard = ({ title }: UserCardProps) => {
    return (
        <Header title={title} />
    );
};
```

–¢–∞–∫–∏–µ —Å–≤—è–∑–∏ –ø–æ—Ä–æ–∂–¥–∞—é—Ç **—Ü–∏–∫–ª–∏—á–µ—Å–∫–∏–µ –∑–∞–≤–∏—Å–∏–º–æ—Å—Ç–∏** –Ω–µ —Ç–æ–ª—å–∫–æ –Ω–∞ —É—Ä–æ–≤–Ω–µ –∏–º–ø–æ—Ä—Ç–∞ es –º–æ–¥—É–ª–µ–π, –Ω–æ –∏ –Ω–∞ —É—Ä–æ–≤–Ω–µ –ø—Ä–æ–µ–∫—Ç–∏—Ä–æ–≤–∞–Ω–∏—è –∫–æ–º–ø–æ–Ω–µ–Ω—Ç–æ–≤:
- –í–æ–∑–Ω–∏–∫–Ω—É—Ç —Å–ª–æ–∂–Ω—ã–µ —Å–≤—è–∑–∏
- –í–æ–∑–Ω–∏–∫–Ω–µ—Ç –≤—ã—Å–æ–∫–æ–µ –∑–∞—Ü–µ–ø–ª–µ–Ω–∏–µ
- –°–ª–æ–∂–Ω–æ –≤—ã–Ω–µ—Å—Ç–∏ –¥–æ—á–µ—Ä–Ω–∏–π –∫–æ–º–ø–æ–Ω–µ–Ω—Ç –¥–ª—è –ø–µ—Ä–µ–∏—Å–ø–æ–ª—å–∑–æ–≤–∞–Ω–∏—è

–ü—É–±–ª–∏—á–Ω–æ–µ API —Ä–æ–¥–∏—Ç–µ–ª—å—Å–∫–æ–≥–æ –∫–æ–º–ø–æ–Ω–µ–Ω—Ç–∞ –¥–æ–ª–∂–Ω–æ —Å–æ–±–∏—Ä–∞—Ç—å—Å—è –∏–∑ —Ç–∏–ø–æ–≤ –¥–æ—á–µ—Ä–Ω–∏—Ö –∫–æ–º–ø–æ–Ω–µ–Ω—Ç–æ–≤ —Å –æ–¥–Ω–æ–Ω–∞–ø—Ä–∞–≤–ª–µ–Ω–Ω–æ–π —Å–≤—è–∑—å—é.

## –í—Å–µ Props —Å –º–∞–ª–µ–Ω—å–∫–æ–π –±—É–∫–≤—ã

**‚ú® –ú–æ—Ç–∏–≤–∞—Ü–∏—è**

–£–ø—Ä–æ—â–µ–Ω–∏–µ —Ñ–æ—Ä–º–∏—Ä–æ–≤–∞–Ω–∏—è –∏–º–µ–Ω –¥–ª—è props.

**ü§ñ –ê–≤—Ç–æ–º–∞—Ç–∏–∑–∞—Ü–∏—è**

–ù–µ –∏–º–ø–ª–µ–º–µ–Ω—Ç–∏—Ä–æ–≤–∞–Ω–æ –≤ eslint-config.

**‚úÖ Valid**

```ts
type Props = {
  title: string;
  userName: string;
  onClick: () => void;
  footer: ReactNode;
  header: FunctionComponent;
};
```

**‚ùå Invalid**

```ts
type Props = {
  title: string;
  userName: string;
  onClick: () => void;
  Footer: ReactNode;
  Header: FunctionComponent;
};
```

## –í—Å–µ props –æ–±—Ä–∞–±–æ—Ç—á–∏–∫–∏ —Å –ø—Ä–µ—Ñ–∏–∫—Å–æ–º ```on```

**‚ú® –ú–æ—Ç–∏–≤–∞—Ü–∏—è**

–ò–¥–µ–Ω—Ç–∏—Ñ–∏–∫–∞—Ü–∏—è –æ–±—Ä–∞–±–æ—Ç—á–∏–∫–æ–≤ –ø–æ –Ω–∞–∑–≤–∞–Ω–∏—é.

**ü§ñ –ê–≤—Ç–æ–º–∞—Ç–∏–∑–∞—Ü–∏—è**

–ù–µ –∏–º–ø–ª–µ–º–µ–Ω—Ç–∏—Ä–æ–≤–∞–Ω–æ –≤ eslint-config.

**‚úÖ Valid**

```ts
type Props = {
  onClick: () => void;
  onChange: () => void;
  onDelete: () => void;
};
```

**‚ùå Invalid**

```ts
type Props = {
  handleClick: () => void;
  remove: () => void;
  clickHandler: () => void;
};
```

## Boolean props –¥–æ–ª–∂–Ω—ã –∏–º–µ—Ç—å –ø—Ä–µ—Ñ–∏–∫—Å, —É–∫–∞–∑—ã–≤–∞—é—â–∏–π –Ω–∞ –ø—Ä–∏–Ω–∞–¥–ª–µ–∂–Ω–æ—Å—Ç—å –∫ Boolean

–°–æ–æ—Ç–≤–µ—Ç—Å—Ç–≤—É–µ—Ç [–µ–¥–∏–Ω–æ–º—É –ø—Ä–∞–≤–∏–ª—É –Ω–µ–π–º–∏–Ω–≥–∞ Boolean –ø–µ—Ä–µ–º–µ–Ω–Ω—ã—Ö](../naming/vars#boolean-–ø–µ—Ä–µ–º–µ–Ω–Ω—ã–µ-–¥–æ–ª–∂–Ω—ã-–∏–º–µ—Ç—å-–ø—Ä–µ—Ñ–∏–∫—Å-—É–∫–∞–∑—ã–≤–∞—é—â–∏–π-–Ω–∞-–ø—Ä–∏–Ω–∞–¥–ª–µ–∂–Ω–æ—Å—Ç—å-–∫-boolean).

## –°—Ç—Ä–æ–∫–æ–≤—ã–µ props –ø–µ—Ä–µ–¥–∞—é—Ç—Å—è —á–µ—Ä–µ–∑ –¥–≤–æ–π–Ω—ã–µ –∫–∞–≤—ã—á–∫–∏

**‚ú® –ú–æ—Ç–∏–≤–∞—Ü–∏—è**

–°—Ç–∞–Ω–¥–∞—Ä—Ç–∏–∑–∞—Ü–∏—è —Å–ø–æ—Å–æ–±–∞ –ø–µ—Ä–µ–¥–∞—á–∏ props.

**ü§ñ –ê–≤—Ç–æ–º–∞—Ç–∏–∑–∞—Ü–∏—è**

–ù–µ –∏–º–ø–ª–µ–º–µ–Ω—Ç–∏—Ä–æ–≤–∞–Ω–æ –≤ eslint-config.

**‚úÖ Valid**

```tsx
<Dialog title="–°–æ–∑–¥–∞–Ω–∏–µ –∑–∞—è–≤–ª–µ–Ω–∏—è" />
<Dialog title={isOrg ? '–°–æ–∑–¥–∞–Ω–∏–µ –∑–∞—è–≤–ª–µ–Ω–∏–µ –Æ–õ' : '–°–æ–∑–¥–∞–Ω–∏–µ –∑–∞—è–≤–ª–µ–Ω–∏–µ –ò–ü'} />
```

**‚ùå Invalid**

```tsx
<Dialog title={'–°–æ–∑–¥–∞–Ω–∏–µ –∑–∞—è–≤–ª–µ–Ω–∏—è'} />
```
