# Props

## Ð¢Ð¸Ð¿Ñ‹ Props Ð´Ð¾Ð»Ð¶Ð½Ñ‹ Ð½Ð°Ñ…Ð¾Ð´Ð¸Ñ‚ÑÑ Ð² Ñ„Ð°Ð¹Ð»Ðµ ÐºÐ¾Ð¼Ð¿Ð¾Ð½ÐµÐ½Ñ‚Ð°

**âœ¨ ÐœÐ¾Ñ‚Ð¸Ð²Ð°Ñ†Ð¸Ñ**

ÐÐ°Ñ…Ð¾Ð¶Ð´ÐµÐ½Ð¸Ðµ Props Ñ‚Ð¸Ð¿Ð¾Ð² Ð² Ñ„Ð°Ð¹Ð»Ðµ ÐºÐ¾Ð¼Ð¿Ð¾Ð½ÐµÐ½Ñ‚Ð° Ð¿Ð¾Ð·Ð²Ð¾Ð»ÑÐµÑ‚:
- Ð˜Ð·Ð±ÐµÐ¶Ð°Ñ‚ÑŒ Ð¾Ð²ÐµÑ€Ñ…ÐµÐ´Ð° Ð¿Ð¾ ÑÐ¾Ð·Ð´Ð°Ð½Ð¸ÑŽ Ð¾Ñ‚Ð´ÐµÐ»ÑŒÐ½Ð¾Ð³Ð¾ Ñ„Ð°Ð¹Ð»Ð° Ð´Ð»Ñ Ñ‚Ð¸Ð¿Ð¾Ð² Props
- Ð˜Ð·Ð±ÐµÐ¶Ð°Ñ‚ÑŒ [Ð¿Ñ€Ð¾Ð±Ð»ÐµÐ¼ Ð¾ÑˆÐ¸Ð±Ð¾Ñ‡Ð½Ð¾Ð³Ð¾ Ð½Ð°Ð¿Ñ€Ð°Ð²Ð»ÐµÐ½Ð¸Ñ Ð·Ð°Ð²Ð¸ÑÐ¸Ð¼Ð¾ÑÑ‚ÐµÐ¹ ÐºÐ¾Ð¼Ð¿Ð¾Ð½ÐµÐ½Ñ‚Ð¾Ð²](#Ð¿Ñ€Ð¾Ð±Ð»ÐµÐ¼Ð°-Ð½ÐµÐ¿Ñ€Ð°Ð²Ð¸Ð»ÑŒÐ½Ð¾Ð³Ð¾-Ð½Ð°Ð¿Ñ€Ð°Ð²Ð»ÐµÐ½Ð¸Ñ-Ð·Ð°Ð²Ð¸ÑÐ¸Ð¼Ð¾ÑÑ‚ÐµÐ¹)

**ðŸ¤– ÐÐ²Ñ‚Ð¾Ð¼Ð°Ñ‚Ð¸Ð·Ð°Ñ†Ð¸Ñ**

ÐÐµ Ð¸Ð¼Ð¿Ð»ÐµÐ¼ÐµÐ½Ñ‚Ð¸Ñ€Ð¾Ð²Ð°Ð½Ð¾ Ð² eslint-config.

**âœ… Valid**

```UserInfo.tsx```
```tsx
type Props = {
  title: string;
  userName: string;
  onClick: () => void;
};

export const UserInfo = ({ title, userName, onClick }: Props) => {
  return (
    <Grid>
      <Typography>{title}</Typography>
      <Typography>{userName}</Typography>
      <Button onClick={onClick}>ÐŸÐ¾ÐºÐ°Ð·Ð°Ñ‚ÑŒ</Button>
    </Grid>
  );
};
```

**âŒ Invalid**

```UserInfo/UserInfo.tsx```
```tsx
import { type UserInfoProps } from './types';

export const UserInfo = ({ title, userName, onClick }: ButtonProps) => {
  return (
    <Grid>
      <Typography>{title}</Typography>
      <Typography>{userName}</Typography>
      <Button onClick={onClick}>ÐŸÐ¾ÐºÐ°Ð·Ð°Ñ‚ÑŒ</Button>
    </Grid>
  );
};
```

```UserInfo/types.ts```
```ts
export type UserInfoProps = {
  title: string;
  userName: string;
  onClick: () => void;
};
```

### ÐŸÑ€Ð¾Ð±Ð»ÐµÐ¼Ð° Ð½ÐµÐ¿Ñ€Ð°Ð²Ð¸Ð»ÑŒÐ½Ð¾Ð³Ð¾ Ð½Ð°Ð¿Ñ€Ð°Ð²Ð»ÐµÐ½Ð¸Ñ Ð·Ð°Ð²Ð¸ÑÐ¸Ð¼Ð¾ÑÑ‚ÐµÐ¹ ÐºÐ¾Ð¼Ð¿Ð¾Ð½ÐµÐ½Ñ‚Ð¾Ð²

Ð’Ñ‹Ð½ÐµÑÐµÐ½Ð¸Ðµ Props Ð² Ð¾Ñ‚Ð´ÐµÐ»ÑŒÐ½Ñ‹Ð¹ Ñ„Ð°Ð¹Ð» ```types.ts``` Ð¼Ð¾Ð¶ÐµÑ‚ Ð¿Ñ€Ð¸Ð²Ð¾Ð´Ð¸Ñ‚ÑŒ Ðº Ñ‚Ð¾Ð¼Ñƒ, Ñ‡Ñ‚Ð¾ Props Ð´Ð¾Ñ‡ÐµÑ€Ð½Ð¸Ñ… ÐºÐ¾Ð¼Ð¿Ð¾Ð½ÐµÐ½Ñ‚Ð¾Ð² Ð±ÑƒÐ´ÑƒÑ‚ Ð·Ð°Ð²Ð¸ÑÐµÑ‚ÑŒ Ð¾Ñ‚ Props Ñ€Ð¾Ð´Ð¸Ñ‚ÐµÐ»Ñ:

```
â”œâ”€â”€ UserCard/
|    â”œâ”€â”€ Header/ 
|    |    â””â”€â”€ Header.tsx
|    |    â””â”€â”€ types.ts
|    |    â””â”€â”€ index.ts
|    â”œâ”€â”€ UserCard.tsx
|    â”œâ”€â”€ types.ts
|    â””â”€â”€ index.ts
```

```UserCard/types.ts```
```ts
export type UserCardProps = {
  headerTitle: ReactNode;
};
```

```HeaderProps``` Ð·Ð°Ð²Ð¸ÑÐ¸Ñ‚ Ð¾Ñ‚ Props Ñ€Ð¾Ð´Ð¸Ñ‚ÐµÐ»ÑŒÑÐºÐ¾Ð³Ð¾ ÐºÐ¾Ð¼Ð¿Ð¾Ð½ÐµÐ½Ñ‚Ð° ```UserCardProps```:

```UserCard/Header/types.ts```
```ts
import { type UserCardProps } from '../types';

export type HeaderProps = {
    title: UserCardProps['headerTitle'];
};
```

ÐŸÑ€Ð¸ ÑÑ‚Ð¾Ð¼ ÐºÐ¾Ð¼Ð¿Ð¾Ð½ÐµÐ½Ñ‚ ```UserCard``` Ð·Ð°Ð²Ð¸ÑÐ¸Ñ‚ Ð¾Ñ‚ ÐºÐ¾Ð¼Ð¿Ð¾Ð½ÐµÐ½Ñ‚Ð° ```Header``` (Ð¸ÑÐ¿Ð¾Ð»ÑŒÐ·ÑƒÐµÑ‚ ÐµÐ³Ð¾ Ð²Ð½ÑƒÑ‚Ñ€Ð¸):
```tsx
export const UserCard = ({ title }: UserCardProps) => {
    return (
        <Header title={title} />
    );
};
```

Ð¢Ð°ÐºÐ¸Ðµ ÑÐ²ÑÐ·Ð¸ Ð¿Ð¾Ñ€Ð¾Ð¶Ð´Ð°ÑŽÑ‚ **Ñ†Ð¸ÐºÐ»Ð¸Ñ‡ÐµÑÐºÐ¸Ðµ Ð·Ð°Ð²Ð¸ÑÐ¸Ð¼Ð¾ÑÑ‚Ð¸** Ð½Ðµ Ñ‚Ð¾Ð»ÑŒÐºÐ¾ Ð½Ð° ÑƒÑ€Ð¾Ð²Ð½Ðµ Ð¸Ð¼Ð¿Ð¾Ñ€Ñ‚Ð° es Ð¼Ð¾Ð´ÑƒÐ»ÐµÐ¹, Ð½Ð¾ Ð¸ Ð½Ð° ÑƒÑ€Ð¾Ð²Ð½Ðµ Ð¿Ñ€Ð¾ÐµÐºÑ‚Ð¸Ñ€Ð¾Ð²Ð°Ð½Ð¸Ñ ÐºÐ¾Ð¼Ð¿Ð¾Ð½ÐµÐ½Ñ‚Ð¾Ð²:
- Ð’Ð¾Ð·Ð½Ð¸ÐºÐ½ÑƒÑ‚ ÑÐ»Ð¾Ð¶Ð½Ñ‹Ðµ ÑÐ²ÑÐ·Ð¸
- Ð’Ð¾Ð·Ð½Ð¸ÐºÐ½ÐµÑ‚ Ð²Ñ‹ÑÐ¾ÐºÐ¾Ðµ Ð·Ð°Ñ†ÐµÐ¿Ð»ÐµÐ½Ð¸Ðµ
- Ð¡Ð»Ð¾Ð¶Ð½Ð¾ Ð²Ñ‹Ð½ÐµÑÑ‚Ð¸ Ð´Ð¾Ñ‡ÐµÑ€Ð½Ð¸Ð¹ ÐºÐ¾Ð¼Ð¿Ð¾Ð½ÐµÐ½Ñ‚ Ð´Ð»Ñ Ð¿ÐµÑ€ÐµÐ¸ÑÐ¿Ð¾Ð»ÑŒÐ·Ð¾Ð²Ð°Ð½Ð¸Ñ

ÐŸÑƒÐ±Ð»Ð¸Ñ‡Ð½Ð¾Ðµ API Ñ€Ð¾Ð´Ð¸Ñ‚ÐµÐ»ÑŒÑÐºÐ¾Ð³Ð¾ ÐºÐ¾Ð¼Ð¿Ð¾Ð½ÐµÐ½Ñ‚Ð° Ð´Ð¾Ð»Ð¶Ð½Ð¾ ÑÐ¾Ð±Ð¸Ñ€Ð°Ñ‚ÑŒÑÑ Ð¸Ð· Ñ‚Ð¸Ð¿Ð¾Ð² Ð´Ð¾Ñ‡ÐµÑ€Ð½Ð¸Ñ… ÐºÐ¾Ð¼Ð¿Ð¾Ð½ÐµÐ½Ñ‚Ð¾Ð² Ñ Ð¾Ð´Ð½Ð¾Ð½Ð°Ð¿Ñ€Ð°Ð²Ð»ÐµÐ½Ð½Ð¾Ð¹ ÑÐ²ÑÐ·ÑŒÑŽ.

## Ð’ÑÐµ Props Ñ Ð¼Ð°Ð»ÐµÐ½ÑŒÐºÐ¾Ð¹ Ð±ÑƒÐºÐ²Ñ‹

**âœ¨ ÐœÐ¾Ñ‚Ð¸Ð²Ð°Ñ†Ð¸Ñ**

Ð£Ð¿Ñ€Ð¾Ñ‰ÐµÐ½Ð¸Ðµ Ñ„Ð¾Ñ€Ð¼Ð¸Ñ€Ð¾Ð²Ð°Ð½Ð¸Ñ Ð¸Ð¼ÐµÐ½ Ð´Ð»Ñ props.

**ðŸ¤– ÐÐ²Ñ‚Ð¾Ð¼Ð°Ñ‚Ð¸Ð·Ð°Ñ†Ð¸Ñ**

ÐÐµ Ð¸Ð¼Ð¿Ð»ÐµÐ¼ÐµÐ½Ñ‚Ð¸Ñ€Ð¾Ð²Ð°Ð½Ð¾ Ð² eslint-config.

**âœ… Valid**

```ts
type Props = {
  title: string;
  userName: string;
  onClick: () => void;
  footer: ReactNode;
  header: FunctionComponent;
};
```

**âŒ Invalid**

```ts
type Props = {
  title: string;
  userName: string;
  onClick: () => void;
  Footer: ReactNode;
  Header: FunctionComponent;
};
```
