# Stores

ĞŸÑ€Ğ°Ğ²Ğ¸Ğ»Ğ° Ğ´Ğ»Ñ **stores**, Ğ½Ğ°Ñ…Ğ¾Ğ´ÑÑ‰Ğ¸Ñ…ÑÑ Ğ²Ğ½ÑƒÑ‚Ñ€Ğ¸ Ğ´Ğ¸Ñ€ĞµĞºÑ‚Ğ¾Ñ€Ğ¸Ğ¸ react-ĞºĞ¾Ğ¼Ğ¿Ğ¾Ğ½ĞµĞ½Ñ‚Ğ°.

Ğ”Ğ»Ñ **UIStore** Ğ¾Ğ¿Ğ¸ÑĞ°Ğ½Ñ‹ [Ğ¾Ñ‚Ğ´ĞµĞ»ÑŒĞ½Ñ‹Ğµ Ğ¿Ñ€Ğ°Ğ²Ğ¸Ğ»Ğ°](./UIstore).

## Stores Ğ´Ğ¾Ğ»Ğ¶Ğ½Ñ‹ Ğ½Ğ°Ñ…Ğ¾Ğ´Ğ¸Ñ‚ÑÑ Ğ² Ğ¾Ñ‚Ğ´ĞµĞ»ÑŒĞ½Ğ¾Ğ¹ Ğ´Ğ¸Ñ€ĞµĞºÑ‚Ğ¾Ñ€Ğ¸Ğ¸ Ğ²Ğ½ÑƒÑ‚Ñ€Ğ¸ ĞºĞ¾Ğ¼Ğ¿Ğ¾Ğ½ĞµĞ½Ñ‚Ğ°

Ğ’ÑĞµ stores, Ñ€ĞµĞ°Ğ»Ğ¸Ğ·Ğ¾Ğ²Ğ°Ğ½Ğ½Ñ‹Ğµ Ğ² ĞºĞ¾Ğ½Ñ‚ĞµĞºÑÑ‚Ğµ ĞºĞ¾Ğ¼Ğ¿Ğ¾Ğ½ĞµĞ½Ñ‚Ğ° Ğ¸ Ğ½Ğµ ÑĞ²Ğ»ÑÑÑ‰Ğ¸ĞµÑÑ VMStore, Ğ´Ğ¾Ğ»Ğ¶Ğ½Ñ‹ Ğ½Ğ°Ñ…Ğ¾Ğ´Ğ¸Ñ‚ÑÑ Ğ² Ğ´Ğ¸Ñ€ĞµĞºÑ‚Ğ¾Ñ€Ğ¸Ğ¸ `stores`.
ĞšĞ°Ğ¶Ğ´Ñ‹Ğ¹ store Ğ´Ğ¾Ğ»Ğ¶ĞµĞ½ Ğ½Ğ°Ñ…Ğ¾Ğ´Ğ¸Ñ‚ÑŒÑÑ Ğ² ÑĞ²Ğ¾ĞµĞ¹ Ğ´Ğ¸Ñ€ĞµĞºÑ‚Ğ¾Ñ€Ğ¸Ğ¸, Ğ²Ğ¼ĞµÑÑ‚Ğµ ÑĞ¾ ÑĞ²ÑĞ·Ğ°Ğ½Ğ½Ñ‹Ğ¼Ğ¸ ÑÑƒÑ‰Ğ½Ğ¾ÑÑ‚ÑĞ¼Ğ¸ (constants, utils...).

**âœ¨ ĞœĞ¾Ñ‚Ğ¸Ğ²Ğ°Ñ†Ğ¸Ñ**

Ğ“Ñ€ÑƒĞ¿Ğ¿Ğ¸Ñ€Ğ¾Ğ²ĞºĞ° stores Ğ²Ğ½ÑƒÑ‚Ñ€Ğ¸ ĞºĞ¾Ğ¼Ğ¿Ğ¾Ğ½ĞµĞ½Ñ‚Ğ°.

**ğŸ¤– ĞĞ²Ñ‚Ğ¾Ğ¼Ğ°Ñ‚Ğ¸Ğ·Ğ°Ñ†Ğ¸Ñ**

ĞĞµ Ğ¸Ğ¼Ğ¿Ğ»ĞµĞ¼ĞµĞ½Ñ‚Ğ¸Ñ€Ğ¾Ğ²Ğ°Ğ½Ğ¾ Ğ² eslint-config.

**âœ… Valid**

```
â”œâ”€â”€ Cart/
|    â”œâ”€â”€ stores/ 
|    |    |â”€â”€ PaymentStore/
|    |    |    |â”€â”€ utils/
|    |    |    |â”€â”€ PaymentStore.ts
|    |    |    |â”€â”€ PaymentStore.test.ts
|    |    |    |â”€â”€ constants.ts
|    |    |    |â”€â”€ types.ts
|    |    |    â””â”€â”€ index.ts
|    |    â””â”€â”€ index.ts
|    â”œâ”€â”€ VMStore/ 
|    |    â””â”€â”€ VMStore.ts
|    |    â””â”€â”€ index.ts
|    â”œâ”€â”€ Cart.tsx
|    â””â”€â”€ index.ts
```

**âŒ Invalid**

```
â”œâ”€â”€ Cart/
|    â”œâ”€â”€ PaymentStore/ 
|    |    â””â”€â”€ PaymentStore.ts
|    |    â””â”€â”€ index.ts
|    â”œâ”€â”€ Cart.tsx
|    â””â”€â”€ index.ts
```

```
â”œâ”€â”€ Cart/
|    â”œâ”€â”€ store/ 
|    |    â””â”€â”€ store.ts
|    |    â””â”€â”€ index.ts
|    â”œâ”€â”€ Cart.tsx
|    â””â”€â”€ index.ts
```

```
â”œâ”€â”€ Cart/
|    â”œâ”€â”€ PaymentStore.ts
|    â”œâ”€â”€ Cart.tsx
|    â””â”€â”€ index.ts
```


