# –°—Ç–∏–ª–∏–∑–∞—Ü–∏—è —á–µ—Ä–µ–∑ styled

## –°—Ç–∏–ª–∏ –¥–ª—è –∫–æ–º–ø–æ–Ω–µ–Ω—Ç–∞ –≤—ã–Ω–æ—Å—è—Ç—Å—è –≤ –æ—Ç–¥–µ–ª—å–Ω—ã–π —Ñ–∞–π–ª `styles`

–í—Å–µ —Å—Ç–∏–ª–∏ –¥–ª—è –∫–æ–º–ø–æ–Ω–µ–Ω—Ç–∞ –≤—ã–Ω–æ—Å—è—Ç—Å—è –≤ –æ—Ç–¥–µ–ª—å–Ω—ã–π —Ñ–∞–π–ª `styles`, –∫–æ—Ç–æ—Ä—ã–π –Ω–∞—Ö–æ–¥–∏—Ç—Å—è –≤ –¥–∏—Ä–µ–∫—Ç–æ—Ä–∏–∏ –∫–æ–º–ø–æ–Ω–µ–Ω—Ç–∞.

**‚ú® –ú–æ—Ç–∏–≤–∞—Ü–∏—è**

- –ù–∞—Ö–æ–∂–¥–µ–Ω–∏–µ —Å—Ç–∏–ª–µ–π –≤ —Ñ–∞–π–ª–µ –∫–æ–º–ø–æ–Ω–µ–Ω—Ç–∞ "–∑–∞–≥—Ä—è–∑–Ω—è–µ—Ç" –∫–æ–¥ –∏ —É—Ö—É–¥—à–∞–µ—Ç —á–∏—Ç–∞–µ–º–æ—Å—Ç—å –∫–æ–¥–∞ –∫–æ–º–ø–æ–Ω–µ–Ω—Ç–∞
- –í—ã–¥–µ–ª–µ–Ω–∏–µ —Å—Ç–∏–ª–µ–π –≤ –æ—Ç–¥–µ–ª—å–Ω—ã–π —Ñ–∞–π–ª –ø–æ–∑–≤–æ–ª—è–µ—Ç –≤–∏–∑—É–∞–ª—å–Ω–æ –æ—Ç–¥–µ–ª–∏—Ç—å –∑–æ–Ω—É –æ—Ç–≤–µ—Ç—Å—Ç–≤–µ–Ω–Ω–æ—Å—Ç–∏

**ü§ñ –ê–≤—Ç–æ–º–∞—Ç–∏–∑–∞—Ü–∏—è**

–ù–µ –∏–º–ø–ª–µ–º–µ–Ω—Ç–∏—Ä–æ–≤–∞–Ω–æ –≤ eslint-config.

**‚úÖ Valid**

```
‚îú‚îÄ‚îÄ UserInfo/
|    ‚îú‚îÄ‚îÄ UserInfo.tsx
|    ‚îú‚îÄ‚îÄ styles.ts
|    ‚îî‚îÄ‚îÄ index.ts
```

```UserInfo.tsx```
```tsx
import { Footer, Header, Wrapper } from './styles';

const UserInfo = ({ isError }: Props) => {
  return (
    <Wrapper>
      <Header>Header</Header>
      <Footer isError={isError} />
    </Wrapper>
  );
};
```

```styles.ts```
```tsx
export const Wrapper = styled.section`
  display: flex;
`;

export const Header = styled.header`
  background-color: white;
`;

export const Footer = styled.footer`
  background-color: ${({ isError }) => (isError ? 'red' : 'white')};
`;
```

**‚ùå Invalid**

```UserInfo.tsx```
```tsx
const Wrapper = styled.section`
  display: flex;
`;

const Header = styled.header`
  background-color: white;
`;

const Footer = styled.footer`
  background-color: ${({ isError }) => (isError ? 'red' : 'white')};
`;

const UserInfo = ({ isError }: Props) => {
  return (
    <Wrapper>
      <Header>Header</Header>
      <Footer isError={isError} />
    </Wrapper>
  );
};
```

## –ù–µ–π–º–∏–Ω–≥ styled-–∫–æ–º–ø–æ–Ω–µ–Ω—Ç–æ–≤

Styled-–∫–æ–º–ø–æ–Ω–µ–Ω—Ç—ã –∏–º–µ–Ω—É—é—Ç—Å—è –≤ –ø—Ä–æ–∏–∑–≤–æ–ª—å–Ω–æ–π —Ñ–æ—Ä–º–µ, –Ω–æ [Styled –ø—Ä–µ—Ñ–∏–∫—Å—ã –∏ –ø–æ—Å—Ç—Ñ–∏–∫—Å—ã –∏—Å–ø–æ–ª—å–∑—É—é—Ç—Å—è —Ç–æ–ª—å–∫–æ –¥–ª—è styled-–æ–±–µ—Ä—Ç–æ–∫](#styled-–ø—Ä–µ—Ñ–∏–∫—Å-–¥–æ–±–∞–≤–ª—è–µ—Ç—Å—è-—Ç–æ–ª—å–∫–æ-–¥–ª—è-–æ–±–µ—Ä—Ç–æ–∫-styled-–∫–æ–º–ø–æ–Ω–µ–Ω—Ç–æ–≤);

**‚ú® –ú–æ—Ç–∏–≤–∞—Ü–∏—è**

–ü—Ä–æ–∏–∑–≤–æ–ª—å–Ω–∞—è —Ñ–æ—Ä–º–∞ –Ω–µ–π–º–∏–Ω–≥–∞ styled-–∫–æ–º–ø–æ–Ω–µ–Ω—Ç–æ–≤ —è–≤–ª—è–µ—Ç—Å—è –Ω–∞–∏–±–æ–ª–µ–µ —É–¥–æ–±–Ω–æ–π –¥–ª—è —Ä–∞–∑—Ä–∞–±–æ—Ç—á–∏–∫–æ–≤.
–í–Ω–µ–¥—Ä–µ–Ω–∏–µ –¥–æ–ø–æ–ª–Ω–∏—Ç–µ–ª—å–Ω—ã—Ö –æ–≥—Ä–∞–Ω–∏—á–µ–Ω–∏–π —É—Å–ª–æ–∂–Ω—è–µ—Ç –∏ –∑–∞–º–µ–¥–ª—è–µ—Ç –Ω–∞–ø–∏—Å–∞–Ω–∏–µ —Å—Ç–∏–ª–µ–π.

**‚úÖ Valid**

```styles.ts```
```tsx
export const Wrapper = styled.section`
  display: flex;
`;

export const Header = styled.header`
  background-color: white;
`;

export const Footer = styled.footer`
  background-color: ${({ isError }) => (isError ? 'red' : 'white')};
`;
```

```UserInfo.tsx```
```tsx
import { Footer, Header, Wrapper } from './styles';

const UserInfo = ({ isError }: Props) => {
  return (
    <Wrapper>
      <Header>Header</Header>
      <Footer isError={isError} />
    </Wrapper>
  );
};
```

**‚ùå Invalid**

```styles.ts```
```tsx
// –ü—Ä–µ—Ñ–∏–∫—Å —Ä–æ–¥–∏—Ç–µ–ª—è —É–∫–∞–∑—ã–≤–∞—Ç—å –Ω–µ –Ω—É–∂–Ω–æ
export const UserInfoWrapper = styled.section`
  display: flex;
`;

// –ü—Ä–µ—Ñ–∏–∫—Å Styled –∏—Å–ø–æ–ª—å–∑—É–µ—Ç—Å—è —Ç–æ–ª—å–∫–æ –¥–ª—è styled-–æ–±–µ—Ä—Ç–æ–∫
export const StyledHeader = styled.header`
  background-color: white;
`;

// –ü–æ—Å—Ç—Ñ–∏–∫—Å Styled –∑–∞–ø—Ä–µ—â–µ–Ω
export const FooterStyled = styled.footer`
  background-color: ${({ isError }) => (isError ? 'red' : 'white')};
`;
```

```UserInfo.tsx```
```tsx
import { FooterStyled, StyledHeader, UserInfoWrapper } from './styles';

const UserInfo = ({ isError }: Props) => {
  return (
    <UserInfoWrapper>
      <StyledHeader>Header</StyledHeader>
      <FooterStyled isError={isError} />
    </UserInfoWrapper>
  );
};
```

## –ù–µ–π–º–∏–Ω–≥ –æ–±–µ—Ä—Ç–æ–∫ styled-–∫–æ–º–ø–æ–Ω–µ–Ω—Ç–æ–≤

–ü—Ä–∏–º–µ—Ä –æ–±–µ—Ä—Ç–∫–∏ styled-–∫–æ–º–ø–æ–Ω–µ–Ω—Ç–∞:

```ts
import { Button } from '@astral/ui';

export const StyledButton = styled(Button)`
  color: red;
`;
```

### –†–µ–∫–æ–º–µ–Ω–¥—É–µ—Ç—Å—è –∏—Å–ø–æ–ª—å–∑–æ–≤–∞—Ç—å —Å–µ–º–∞–Ω—Ç–∏—á–µ—Å–∫–∏–µ –∏–º–µ–Ω–∞ –¥–ª—è –æ–±–µ—Ä—Ç–æ–∫ styled-–∫–æ–º–ø–æ–Ω–µ–Ω—Ç–æ–≤

**‚ú® –ú–æ—Ç–∏–≤–∞—Ü–∏—è**

–°–µ–º–∞–Ω—Ç–∏—á–µ—Å–∫–∏–µ –∏–º–µ–Ω–∞ styled-–∫–æ–º–ø–æ–Ω–µ–Ω—Ç–æ–≤ –ø–æ–∑–≤–æ–ª—è—é—Ç –ø–æ –Ω–∞–∑–≤–∞–Ω–∏—é –ø–æ–Ω—è—Ç—å –Ω–∞–∑–Ω–∞—á–µ–Ω–∏–µ –∫–æ–º–ø–æ–Ω–µ–Ω—Ç–∞.

**‚úÖ Valid**

```styles.ts```
```ts
import { Button } from '@astral/ui';

export const LogoutButton = styled(Button)`
  color: red;
`;
```

```UserInfo.tsx```
```tsx
import { LogoutButton, Wrapper } from './styles';

const UserInfo = () => {
  return (
    <Wrapper>
      <LogoutButton>–í—ã–π—Ç–∏</LogoutButton>
    </Wrapper>
  );
};
```

### –î–æ–ø—É—Å–∫–∞–µ—Ç—Å—è –∏—Å–ø–æ–ª—å–∑–æ–≤–∞–Ω–∏–µ `Styled` –ø—Ä–µ—Ñ–∏–∫—Å–∞ –¥–ª—è –æ–±–µ—Ä—Ç–æ–∫ styled-–∫–æ–º–ø–æ–Ω–µ–Ω—Ç–æ–≤

–ï—Å–ª–∏ –Ω–µ–≤–æ–∑–º–æ–∂–Ω–æ —É–∫–∞–∑–∞—Ç—å —Å–µ–º–∞–Ω—Ç–∏—á–µ—Å–∫–æ–µ –∏–º—è –¥–ª—è –∫–æ–º–ø–æ–Ω–µ–Ω—Ç–∞, —Ç–æ –¥–æ–ø—É—Å–∫–∞–µ—Ç—Å—è –∏—Å–ø–æ–ª—å–∑–æ–≤–∞–Ω–∏–µ `Styled` –ø—Ä–µ—Ñ–∏–∫—Å–∞.

**‚ú® –ú–æ—Ç–∏–≤–∞—Ü–∏—è**

–ù–µ –≤—Å–µ–º styled-–æ–±–µ—Ä—Ç–∫–∞–º –º–æ–∂–Ω–æ –¥–∞—Ç—å —Å–µ–º–∞–Ω—Ç–∏—á–µ—Å–∫–æ–µ –∏–º—è.

**–ü—Ä–∏–º–µ—Ä –∫–∞—Å—Ç–æ–º–∏–∑–∞—Ü–∏–∏ –∫–Ω–æ–ø–∫–∏ ui-kit**

**‚úÖ Valid**

```styles.ts```
```ts
import { Button } from '@astral/ui';

export const StyledButton = styled(Button)`
  color: red;
`;
```

```Button.tsx```
```tsx
import { StyledButton } from './styles';

const Button = () => <StyledButton />;
```

**‚ùå Invalid**

```ts
import { Button } from '@astral/ui';

// –ü–æ—Å—Ç—Ñ–∏–∫—Å –∑–∞–ø—Ä–µ—â–µ–Ω
export const ButtonStyled = styled(Button)`
  color: red;
`;
```

---

```ts
import { Button as AstralButton } from '@astral/ui';

export const Button = styled(AstralButton)`
  color: red;
`;
```

```Button.tsx```
```tsx
// –õ–∏—à–Ω–µ–µ –¥–µ–π—Å—Ç–≤–∏–µ
import { Button as StyledButton } from './styles';

const Button = () => <StyledButton />;
```

---

```ts
import { Button } from '@astral/ui';

// –°–µ–º–∞–Ω—Ç–∏—á–µ—Å–∫–∏ –æ–±–µ—Ä—Ç–∫–∞ –Ω–µ —è–≤–ª—è–µ—Ç—Å—è –∫–æ—Ä–Ω–µ–º –∫–æ–º–ø–æ–Ω–µ–Ω—Ç–∞ –∏–ª–∏ –µ–≥–æ –æ—Å–Ω–æ–≤–æ–π, —ç—Ç–æ –ø—Ä–æ—Å—Ç–æ –æ–±–µ—Ä—Ç–∫–∞ –¥–ª—è Button
export const ButtonRoot = styled(Button)`
  color: red;
`;
```

```Button.tsx```
```tsx
import { ButtonRoot } from './styles';

const Button = () => <ButtonRoot />;
```

## –¢–∏–ø—ã –¥–ª—è Props styled-–∫–æ–º–ø–æ–Ω–µ–Ω—Ç–∞ –Ω–µ –¥–æ–ª–∂–Ω—ã –∑–∞–≤–∏—Å–µ—Ç—å –æ—Ç Props –∫–æ—Ä–Ω–µ–≤–æ–≥–æ –∫–æ–º–ø–æ–Ω–µ–Ω—Ç–∞

**‚ú® –ú–æ—Ç–∏–≤–∞—Ü–∏—è**

–ó–æ–Ω–∞ –æ—Ç–≤–µ—Ç—Å—Ç–≤–µ–Ω–Ω–æ—Å—Ç–∏ styled-–∫–æ–º–ø–æ–Ω–µ–Ω—Ç–∞ - —Å—Ç–∏–ª–∏–∑–∞—Ü–∏—è.
–ü—Ä—è–º–æ–µ –∏—Å–ø–æ–ª—å–∑–æ–≤–∞–Ω–∏–µ –ø—Ä–æ–ø—Å–æ–≤ –∫–æ—Ä–Ω–µ–≤–æ–≥–æ –∫–æ–º–ø–æ–Ω–µ–Ω—Ç–∞ –≤ styled –Ω–∞—Ä—É—à–∞–µ—Ç –ø—Ä–∏–Ω—Ü–∏–ø –µ–¥–∏–Ω—Å—Ç–≤–µ–Ω–Ω–æ–π –æ—Ç–≤–µ—Ç—Å—Ç–≤–µ–Ω–Ω–æ—Å—Ç–∏ (SOLID):
styled-–∫–æ–º–ø–æ–Ω–µ–Ω—Ç –Ω–∞—á–∏–Ω–∞–µ—Ç –∑–Ω–∞—Ç—å –æ public API –∫–æ—Ä–Ω–µ–≤–æ–≥–æ –∫–æ–º–ø–æ–Ω–µ–Ω—Ç–∞ –∏ –∏–Ω–∫–∞–ø—Å—É–ª–∏—Ä–æ–≤–∞—Ç—å –∑–Ω–∞–Ω–∏—è –æ –ª–æ–≥–∏–∫–µ.

**ü§ñ –ê–≤—Ç–æ–º–∞—Ç–∏–∑–∞—Ü–∏—è**

–ù–µ –∏–º–ø–ª–µ–º–µ–Ω—Ç–∏—Ä–æ–≤–∞–Ω–æ –≤ eslint-config.

**‚úÖ Valid**

```
‚îú‚îÄ‚îÄ UserInfo/
|    ‚îú‚îÄ‚îÄ UserInfo.tsx
|    ‚îú‚îÄ‚îÄ styles.ts
|    ‚îî‚îÄ‚îÄ index.ts
```

```UserInfo.tsx```
```tsx
import { Footer, Header, Wrapper } from './styles';

type UserInfoProps = {
  error?: Error;
};

const UserInfo = ({ error }: UserInfoProps) => {
  return (
    <Wrapper>
      <Header>Header</Header>
      <Footer isError={Boolean(error)} />
    </Wrapper>
  );
};
```

```styles.ts```
```tsx
type FooterProps = {
  isError: boolean;
};

export const Footer = styled.footer<FooterProps>`
  background-color: ${({ isError }) => (isError ? 'red' : 'white')};
`;
```

**‚ùå Invalid**

```
‚îú‚îÄ‚îÄ UserInfo/
|    ‚îú‚îÄ‚îÄ UserInfo.tsx
|    ‚îú‚îÄ‚îÄ styles.ts
|    ‚îî‚îÄ‚îÄ index.ts
```

```UserInfo.tsx```
```tsx
import { Footer, Header, Wrapper } from './styles';

export type UserInfoProps = {
  error?: Error;
};

const UserInfo = ({ error }: UserInfoProps) => {
  return (
    <Wrapper>
      <Header>Header</Header>
      <Footer error={error} />
    </Wrapper>
  );
};
```

```styles.ts```
```tsx
// –¶–∏–∫–ª–∏—á–µ—Å–∫–∞—è –∑–∞–≤–∏—Å–∏–º–æ—Å—Ç—å
import { type UserInfoProps } from './UserInfo';

// error Prop –≤ –¥–∞–Ω–Ω–æ–º —Å–ª—É—á–∞–µ –Ω–µ –æ–Ω–æ—Å–∏—Ç—Å—è –∫ —Å—Ç–∏–ª–∏–∑–∞—Ü–∏–∏, –æ–Ω –æ—Ç–Ω–æ—Å–∏—Ç—Å—è –∫ –ª–æ–≥–∏–∫–µ –≤—Ä–µ—Ö–Ω–µ—É—Ä–æ–≤–Ω–µ–≥–æ –∫–æ–º–ø–æ–Ω–µ–Ω—Ç–∞
type FooterProps = Pick<UserInfoProps, 'error'>;

// Footer –∂–µ—Å—Ç–∫–æ –∑–∞—Ü–µ–ø–ª–µ–Ω —Å UserInfo. –õ—é–±–æ–µ –∏–∑–º–µ–Ω–µ–Ω–∏–µ public API –±—É–¥–µ—Ç –ø—Ä–æ–≤–æ—Ü–∏—Ä–æ–≤–∞—Ç—å –∏–∑–º–µ–Ω–µ–Ω–∏—è –≤ Footer
export const Footer = styled.footer<FooterProps>`
  background-color: ${({ error }) => (error ? 'red' : 'white')};
`;
```

### –î–ª—è styled-–æ–±–µ—Ä—Ç–æ–∫ –¥–æ–ø—É—Å—Ç–∏–º–∞ —Å–≤—è–∑—å —Å –æ–±–æ—Ä–∞—á–∏–≤–∞–µ–º—ã–º –∫–æ–º–ø–æ–Ω–µ–Ω—Ç–æ–º

**‚úÖ Valid**

```
‚îú‚îÄ‚îÄ UserInfo/
|    ‚îú‚îÄ‚îÄ UserInfo.tsx
|    ‚îú‚îÄ‚îÄ styles.ts
|    ‚îî‚îÄ‚îÄ index.ts
```

```UserInfo.tsx```
```tsx
import { Footer, StyledButton } from './styles';

type UserInfoProps = {
  error?: Error;
};

const UserInfo = ({ error }: UserInfoProps) => {
  return (
    <Wrapper>
      <Header>Header</Header>
      <StyledButton isError={Boolean(error)} variant="light">
        Click
      </StyledButton>
    </Wrapper>
  );
};
```

```styles.ts```
```tsx
import { Button, type ButtonProps } from '@astral/ui'

type StyledButtonProps = ButtonProps & {
  isError: boolean;
};

export const StyledButton = styled<StyledButtonProps>(Button)`
  color: ${({ isError }) => (isError ? 'red' : 'white')};
`;
```
