# Ğ¡Ñ‚Ğ¸Ğ»Ğ¸Ğ·Ğ°Ñ†Ğ¸Ñ Ñ‡ĞµÑ€ĞµĞ· styled

## Ğ¡Ñ‚Ğ¸Ğ»Ğ¸ Ğ´Ğ»Ñ ĞºĞ¾Ğ¼Ğ¿Ğ¾Ğ½ĞµĞ½Ñ‚Ğ° Ğ²Ñ‹Ğ½Ğ¾ÑÑÑ‚ÑÑ Ğ² Ğ¾Ñ‚Ğ´ĞµĞ»ÑŒĞ½Ñ‹Ğ¹ Ñ„Ğ°Ğ¹Ğ» `styles`

Ğ’ÑĞµ ÑÑ‚Ğ¸Ğ»Ğ¸ Ğ´Ğ»Ñ ĞºĞ¾Ğ¼Ğ¿Ğ¾Ğ½ĞµĞ½Ñ‚Ğ° Ğ²Ñ‹Ğ½Ğ¾ÑÑÑ‚ÑÑ Ğ² Ğ¾Ñ‚Ğ´ĞµĞ»ÑŒĞ½Ñ‹Ğ¹ Ñ„Ğ°Ğ¹Ğ» `styles`, ĞºĞ¾Ñ‚Ğ¾Ñ€Ñ‹Ğ¹ Ğ½Ğ°Ñ…Ğ¾Ğ´Ğ¸Ñ‚ÑÑ Ğ² Ğ´Ğ¸Ñ€ĞµĞºÑ‚Ğ¾Ñ€Ğ¸Ğ¸ ĞºĞ¾Ğ¼Ğ¿Ğ¾Ğ½ĞµĞ½Ñ‚Ğ°.

**âœ¨ ĞœĞ¾Ñ‚Ğ¸Ğ²Ğ°Ñ†Ğ¸Ñ**

- ĞĞ°Ñ…Ğ¾Ğ¶Ğ´ĞµĞ½Ğ¸Ğµ ÑÑ‚Ğ¸Ğ»ĞµĞ¹ Ğ² Ñ„Ğ°Ğ¹Ğ»Ğµ ĞºĞ¾Ğ¼Ğ¿Ğ¾Ğ½ĞµĞ½Ñ‚Ğ° "Ğ·Ğ°Ğ³Ñ€ÑĞ·Ğ½ÑĞµÑ‚" ĞºĞ¾Ğ´ Ğ¸ ÑƒÑ…ÑƒĞ´ÑˆĞ°ĞµÑ‚ Ñ‡Ğ¸Ñ‚Ğ°ĞµĞ¼Ğ¾ÑÑ‚ÑŒ ĞºĞ¾Ğ´Ğ° ĞºĞ¾Ğ¼Ğ¿Ğ¾Ğ½ĞµĞ½Ñ‚Ğ°
- Ğ’Ñ‹Ğ´ĞµĞ»ĞµĞ½Ğ¸Ğµ ÑÑ‚Ğ¸Ğ»ĞµĞ¹ Ğ² Ğ¾Ñ‚Ğ´ĞµĞ»ÑŒĞ½Ñ‹Ğ¹ Ñ„Ğ°Ğ¹Ğ» Ğ¿Ğ¾Ğ·Ğ²Ğ¾Ğ»ÑĞµÑ‚ Ğ²Ğ¸Ğ·ÑƒĞ°Ğ»ÑŒĞ½Ğ¾ Ğ¾Ñ‚Ğ´ĞµĞ»Ğ¸Ñ‚ÑŒ Ğ·Ğ¾Ğ½Ñƒ Ğ¾Ñ‚Ğ²ĞµÑ‚ÑÑ‚Ğ²ĞµĞ½Ğ½Ğ¾ÑÑ‚Ğ¸

**ğŸ¤– ĞĞ²Ñ‚Ğ¾Ğ¼Ğ°Ñ‚Ğ¸Ğ·Ğ°Ñ†Ğ¸Ñ**

ĞĞµ Ğ¸Ğ¼Ğ¿Ğ»ĞµĞ¼ĞµĞ½Ñ‚Ğ¸Ñ€Ğ¾Ğ²Ğ°Ğ½Ğ¾ Ğ² eslint-config.

**âœ… Valid**

```
â”œâ”€â”€ UserInfo/
|    â”œâ”€â”€ UserInfo.tsx
|    â”œâ”€â”€ styles.ts
|    â””â”€â”€ index.ts
```

```UserInfo.tsx```
```tsx
import { Footer, Header, Wrapper } from './styles';

const UserInfo = ({ isError }: Props) => {
  return (
    <Wrapper>
      <Header>Header</Header>
      <Footer isError={isError} />
    </Wrapper>
  );
};
```

```styles.ts```
```tsx
export const Wrapper = styled.section`
  display: flex;
`;

export const Header = styled.header`
  background-color: white;
`;

export const Footer = styled.footer`
  background-color: ${({ isError }) => (isError ? 'red' : 'white')};
`;
```

**âŒ Invalid**


```UserInfo.tsx```
```tsx
const Wrapper = styled.section`
  display: flex;
`;

const Header = styled.header`
  background-color: white;
`;

const Footer = styled.footer`
  background-color: ${({ isError }) => (isError ? 'red' : 'white')};
`;

const UserInfo = ({ isError }: Props) => {
  return (
    <Wrapper>
      <Header>Header</Header>
      <Footer isError={isError} />
    </Wrapper>
  );
};
```

## ĞĞµĞ¹Ğ¼Ğ¸Ğ½Ğ³ styled-ĞºĞ¾Ğ¼Ğ¿Ğ¾Ğ½ĞµĞ½Ñ‚Ğ¾Ğ²

Styled-ĞºĞ¾Ğ¼Ğ¿Ğ¾Ğ½ĞµĞ½ĞµĞ½Ñ‚Ñ‹ Ğ¸Ğ¼ĞµĞ½ÑƒÑÑ‚ÑÑ Ğ² Ğ¿Ñ€Ğ¾Ğ¸Ğ·Ğ²Ğ¾Ğ»ÑŒĞ½Ğ¾Ğ¹ Ñ„Ğ¾Ñ€Ğ¼Ğµ, Ğ½Ğ¾ [Styled Ğ¿Ñ€ĞµÑ„Ğ¸ĞºÑÑ‹ Ğ¸ Ğ¿Ğ¾ÑÑ‚Ñ„Ğ¸ĞºÑÑ‹ Ğ¸ÑĞ¿Ğ¾Ğ»ÑŒĞ·ÑƒÑÑ‚ÑÑ Ñ‚Ğ¾Ğ»ÑŒĞºĞ¾ Ğ´Ğ»Ñ styled-Ğ¾Ğ±ĞµÑ€Ñ‚Ğ¾Ğº](#styled-Ğ¿Ñ€ĞµÑ„Ğ¸ĞºÑ-Ğ´Ğ¾Ğ±Ğ°Ğ²Ğ»ÑĞµÑ‚ÑÑ-Ñ‚Ğ¾Ğ»ÑŒĞºĞ¾-Ğ´Ğ»Ñ-Ğ¾Ğ±ĞµÑ€Ñ‚Ğ¾Ğº-styled-ĞºĞ¾Ğ¼Ğ¿Ğ¾Ğ½ĞµĞ½Ñ‚Ğ¾Ğ²);

**âœ¨ ĞœĞ¾Ñ‚Ğ¸Ğ²Ğ°Ñ†Ğ¸Ñ**

ĞŸÑ€Ğ¾Ğ¸Ğ·Ğ²Ğ¾Ğ»ÑŒĞ½Ğ°Ñ Ñ„Ğ¾Ñ€Ğ¼Ğ° Ğ½ĞµĞ¹Ğ¼Ğ¸Ğ½Ğ³Ğ° styled-ĞºĞ¾Ğ¼Ğ¿Ğ¾Ğ½ĞµĞ½Ñ‚Ğ¾Ğ² ÑĞ²Ğ»ÑĞµÑ‚ÑÑ Ğ½Ğ°Ğ¸Ğ±Ğ¾Ğ»ĞµĞµ ÑƒĞ´Ğ¾Ğ±Ğ½Ğ¾Ğ¹ Ğ´Ğ»Ñ Ñ€Ğ°Ğ·Ñ€Ğ°Ğ±Ğ¾Ñ‚Ñ‡Ğ¸ĞºĞ¾Ğ².
Ğ’Ğ½ĞµĞ´Ñ€ĞµĞ½Ğ¸Ğµ Ğ´Ğ¾Ğ¿Ğ¾Ğ»Ğ½Ğ¸Ñ‚ĞµĞ»ÑŒĞ½Ñ‹Ñ… Ğ¾Ğ³Ñ€Ğ°Ğ½Ğ¸Ñ‡ĞµĞ½Ğ¸Ğ¹ ÑƒÑĞ»Ğ¾Ğ¶Ğ½ÑĞµÑ‚ Ğ¸ Ğ·Ğ°Ğ¼ĞµĞ´Ğ»ÑĞµÑ‚ Ğ½Ğ°Ğ¿Ğ¸ÑĞ°Ğ½Ğ¸Ğµ ÑÑ‚Ğ¸Ğ»ĞµĞ¹.

**âœ… Valid**

```styles.ts```
```tsx
export const Wrapper = styled.section`
  display: flex;
`;

export const Header = styled.header`
  background-color: white;
`;

export const Footer = styled.footer`
  background-color: ${({ isError }) => (isError ? 'red' : 'white')};
`;
```

```UserInfo.tsx```
```tsx
import { Footer, Header, Wrapper } from './styles';

const UserInfo = ({ isError }: Props) => {
  return (
    <Wrapper>
      <Header>Header</Header>
      <Footer isError={isError} />
    </Wrapper>
  );
};
```

**âŒ Invalid**

```styles.ts```
```tsx
// ĞŸÑ€ĞµÑ„Ğ¸ĞºÑ Ñ€Ğ¾Ğ´Ğ¸Ñ‚ĞµĞ»Ñ ÑƒĞºĞ°Ğ·Ñ‹Ğ²Ğ°Ñ‚ÑŒ Ğ½Ğµ Ğ½ÑƒĞ¶Ğ½Ğ¾
export const UserInfoWrapper = styled.section`
  display: flex;
`;

// ĞŸÑ€ĞµÑ„Ğ¸ĞºÑ Styled Ğ¸ÑĞ¿Ğ¾Ğ»ÑŒĞ·ÑƒĞµÑ‚ÑÑ Ñ‚Ğ¾Ğ»ÑŒĞºĞ¾ Ğ´Ğ»Ñ styled-Ğ¾Ğ±ĞµÑ€Ñ‚Ğ¾Ğº
export const StyledHeader = styled.header`
  background-color: white;
`;

// ĞŸĞ¾ÑÑ‚Ñ„Ğ¸ĞºÑ Styled Ğ·Ğ°Ğ¿Ñ€ĞµÑ‰ĞµĞ½
export const FooterStyled = styled.footer`
  background-color: ${({ isError }) => (isError ? 'red' : 'white')};
`;
```

```UserInfo.tsx```
```tsx
import { FooterStyled, StyledHeader, UserInfoWrapper } from './styles';

const UserInfo = ({ isError }: Props) => {
  return (
    <UserInfoWrapper>
      <StyledHeader>Header</StyledHeader>
      <FooterStyled isError={isError} />
    </UserInfoWrapper>
  );
};
```

## ĞĞµĞ¹Ğ¼Ğ¸Ğ½Ğ³ Ğ¾Ğ±ĞµÑ€Ñ‚Ğ¾Ğº styled-ĞºĞ¾Ğ¼Ğ¿Ğ¾Ğ½ĞµĞ½Ñ‚Ğ¾Ğ²

ĞŸÑ€Ğ¸Ğ¼ĞµÑ€ Ğ¾Ğ±ĞµÑ€Ñ‚ĞºĞ¸ styled-ĞºĞ¾Ğ¼Ğ¿Ğ¾Ğ½ĞµĞ½Ñ‚Ğ°:

```ts
import { Button } from '@astral/ui';

export const StyledButton = styled(Button)`
  color: red;
`;
```

### Ğ ĞµĞºĞ¾Ğ¼ĞµĞ½Ğ´ÑƒĞµÑ‚ÑÑ Ğ¸ÑĞ¿Ğ¾Ğ»ÑŒĞ·Ğ¾Ğ²Ğ°Ñ‚ÑŒ ÑĞµĞ¼Ğ°Ğ½Ñ‚Ğ¸Ñ‡ĞµÑĞºĞ¸Ğµ Ğ¸Ğ¼ĞµĞ½Ğ° Ğ´Ğ»Ñ Ğ¾Ğ±ĞµÑ€Ñ‚Ğ¾Ğº styled-ĞºĞ¾Ğ¼Ğ¿Ğ¾Ğ½ĞµĞ½Ñ‚Ğ¾Ğ²

**âœ¨ ĞœĞ¾Ñ‚Ğ¸Ğ²Ğ°Ñ†Ğ¸Ñ**

Ğ¡ĞµĞ¼Ğ°Ğ½Ñ‚Ğ¸Ñ‡ĞµÑĞºĞ¸Ğµ Ğ¸Ğ¼ĞµĞ½Ğ° styled-ĞºĞ¾Ğ¼Ğ¿Ğ¾Ğ½ĞµĞ½Ñ‚Ğ¾Ğ² Ğ¿Ğ¾Ğ·Ğ²Ğ¾Ğ»ÑÑÑ‚ Ğ¿Ğ¾ Ğ½Ğ°Ğ·Ğ²Ğ°Ğ½Ğ¸Ñ Ğ¿Ğ¾Ğ½ÑÑ‚ÑŒ Ğ½Ğ°Ğ·Ğ½Ğ°Ñ‡ĞµĞ½Ğ¸Ğµ ĞºĞ¾Ğ¼Ğ¿Ğ¾Ğ½ĞµĞ½Ñ‚Ğ°.

**âœ… Valid**

```styles.ts```
```ts
import { Button } from '@astral/ui';

export const LogoutButton = styled(Button)`
  color: red;
`;
```

```UserInfo.tsx```
```tsx
import { LogoutButton, Wrapper } from './styles';

const UserInfo = () => {
  return (
    <Wrapper>
      <LogoutButton>Ğ’Ñ‹Ğ¹Ñ‚Ğ¸</LogoutButton>
    </Wrapper>
  );
};
```

### Ğ”Ğ¾Ğ¿ÑƒÑĞºĞ°ĞµÑ‚ÑÑ Ğ¸ÑĞ¿Ğ¾Ğ»ÑŒĞ·Ğ¾Ğ²Ğ°Ğ½Ğ¸Ğµ `Styled` Ğ¿Ñ€ĞµÑ„Ğ¸ĞºÑĞ° Ğ´Ğ»Ñ Ğ¾Ğ±ĞµÑ€Ñ‚Ğ¾Ğº styled-ĞºĞ¾Ğ¼Ğ¿Ğ¾Ğ½ĞµĞ½Ñ‚Ğ¾Ğ²

Ğ•ÑĞ»Ğ¸ Ğ½ĞµĞ²Ğ¾Ğ·Ğ¼Ğ¾Ğ¶Ğ½Ğ¾ ÑƒĞºĞ°Ğ·Ğ°Ñ‚ÑŒ ÑĞµĞ¼Ğ°Ğ½Ñ‚Ğ¸Ñ‡ĞµÑĞºĞ¾Ğµ Ğ¸Ğ¼Ñ Ğ´Ğ»Ñ ĞºĞ¾Ğ¼Ğ¿Ğ¾Ğ½ĞµĞ½Ñ‚Ğ°, Ñ‚Ğ¾ Ğ´Ğ¾Ğ¿ÑƒÑĞºĞ°ĞµÑ‚ÑÑ Ğ¸ÑĞ¿Ğ¾Ğ»ÑŒĞ·Ğ¾Ğ²Ğ°Ğ½Ğ¸Ğµ `Styled` Ğ¿Ñ€ĞµÑ„Ğ¸ĞºÑĞ°.

**âœ¨ ĞœĞ¾Ñ‚Ğ¸Ğ²Ğ°Ñ†Ğ¸Ñ**

ĞĞµ Ğ²ÑĞµĞ¼ styled-Ğ¾Ğ±ĞµÑ€Ñ‚ĞºĞ°Ğ¼ Ğ¼Ğ¾Ğ¶Ğ½Ğ¾ Ğ´Ğ°Ñ‚ÑŒ ÑĞµĞ¼Ğ°Ğ½Ñ‚Ğ¸Ñ‡ĞµÑĞºĞ¾Ğµ Ğ¸Ğ¼Ñ.

**ĞŸÑ€Ğ¸Ğ¼ĞµÑ€ ĞºĞ°ÑÑ‚Ğ¾Ğ¼Ğ¸Ğ·Ğ°Ñ†Ğ¸Ğ¸ ĞºĞ½Ğ¾Ğ¿ĞºĞ¸ ui-kit**

**âœ… Valid**

```styles.ts```
```ts
import { Button } from '@astral/ui';

export const StyledButton = styled(Button)`
  color: red;
`;
```

```Button.tsx```
```tsx
import { StyledButton } from './styles';

const Button = () => <StyledButton />;
```

**âŒ Invalid**

```ts
import { Button } from '@astral/ui';

// ĞŸĞ¾ÑÑ‚Ñ„Ğ¸ĞºÑ Ğ·Ğ°Ğ¿Ñ€ĞµÑ‰ĞµĞ½
export const ButtonStyled = styled(Button)`
  color: red;
`;
```

---

```ts
import { Button as AstralButton } from '@astral/ui';

export const Button = styled(AstralButton)`
  color: red;
`;
```

```Button.tsx```
```tsx
// Ğ›Ğ¸ÑˆĞ½ĞµĞµ Ğ´ĞµĞ¹ÑÑ‚Ğ²Ğ¸Ğµ
import { Button as StyledButton } from './styles';

const Button = () => <StyledButton />;
```

---

```ts
import { Button } from '@astral/ui';

// Ğ¡ĞµĞ¼Ğ°Ğ½Ñ‚Ğ¸Ñ‡ĞµÑĞºĞ¸ Ğ¾Ğ±ĞµÑ€Ñ‚ĞºĞ° Ğ½Ğµ ÑĞ²Ğ»ÑĞµÑ‚ÑÑ ĞºĞ¾Ñ€Ğ½ĞµĞ¼ ĞºĞ¾Ğ¼Ğ¿Ğ¾Ğ½ĞµĞ½Ñ‚Ğ° Ğ¸Ğ»Ğ¸ ĞµĞ³Ğ¾ Ğ¾ÑĞ½Ğ¾Ğ²Ğ¾Ğ¹, ÑÑ‚Ğ¾ Ğ¿Ñ€Ğ¾ÑÑ‚Ğ¾ Ğ¾Ğ±ĞµÑ€Ñ‚ĞºĞ° Ğ´Ğ»Ñ Button
export const ButtonRoot = styled(AstralButton)`
  color: red;
`;
```

```Button.tsx```
```tsx
import { ButtonRoot } from './styles';

const Button = () => <ButtonRoot />;
```
