# UI Store

ĞŸÑ€Ğ°Ğ²Ğ¸Ğ»Ğ° Ğ´Ğ»Ñ stores, ÑĞ¾Ğ´ĞµÑ€Ğ¶Ğ°Ñ‰Ğ¸Ñ… Ğ»Ğ¾Ğ³Ğ¸ĞºÑƒ ĞºĞ¾Ğ¼Ğ¿Ğ¾Ğ½ĞµĞ½Ñ‚Ğ°.

Ğ”Ğ»Ñ stores, Ğ½Ğµ ÑĞ²Ğ»ÑÑÑ‰Ğ¸Ğ¼Ğ¸ÑÑ UIStore Ğ¾Ğ¿Ğ¸ÑĞ°Ğ½Ñ‹ [Ğ¾Ñ‚Ğ´ĞµĞ»ÑŒĞ½Ñ‹Ğµ Ğ¿Ñ€Ğ°Ğ²Ğ¸Ğ»Ğ°](./stores).

## UIStore Ğ´Ğ¾Ğ»Ğ¶ĞµĞ½ Ğ½Ğ°Ñ…Ğ¾Ğ´Ğ¸Ñ‚ÑÑ Ğ²Ğ½ÑƒÑ‚Ñ€Ğ¸ Ğ´Ğ¸Ñ€ĞµĞºÑ‚Ğ¾Ñ€Ğ¸Ğ¸ ĞºĞ¾Ğ¼Ğ¿Ğ¾Ğ½ĞµĞ½Ñ‚Ğ° Ğ² Ğ¾Ñ‚Ğ´ĞµĞ»ÑŒĞ½Ğ¾Ğ¹ Ğ´Ğ¸Ñ€ĞµĞºÑ‚Ğ¾Ñ€Ğ¸Ğ¸

**âœ¨ ĞœĞ¾Ñ‚Ğ¸Ğ²Ğ°Ñ†Ğ¸Ñ**

- ĞŸĞ¾Ğ·Ğ²Ğ¾Ğ»ÑĞµÑ‚ ÑƒÑÑ‚Ğ°Ğ½Ğ°Ğ²Ğ»Ğ¸Ğ²Ğ°Ñ‚ÑŒ Ğ²Ğ¸Ğ·ÑƒĞ°Ğ»ÑŒĞ½Ğ¾ ÑĞ²ÑĞ·ÑŒ react-ĞºĞ¾Ğ¼Ğ¿Ğ¾Ğ½ĞµĞ½Ñ‚Ğ° Ğ¸ store Ğ´Ğ»Ñ Ğ½ĞµĞ³Ğ¾
- ĞŸÑ€Ğ¸ Ğ¿ĞµÑ€ĞµĞ½Ğ¾ÑĞµ ĞºĞ¾Ğ¼Ğ¿Ğ¾Ğ½ĞµĞ½Ñ‚Ğ° Ğ² Ğ´Ñ€ÑƒĞ³ÑƒÑ Ğ´Ğ¸Ñ€ĞµĞºÑ‚Ğ¾Ñ€Ğ¸Ñ Ğ²Ğ¼ĞµÑÑ‚Ğµ Ñ Ğ½Ğ¸Ğ¼ Ğ±ÑƒĞ´ĞµÑ‚ Ğ¿ĞµÑ€ĞµĞ½ĞµÑĞµĞ½ Ğ¸ store
- ĞŸĞ¾Ğ·Ğ²Ğ¾Ğ»ÑĞµÑ‚ Ğ¿ĞµÑ€ĞµĞ½Ğ¾ÑĞ¸Ñ‚ÑŒ store ÑĞ¾ ÑĞ²ÑĞ·Ğ°Ğ½Ğ½Ñ‹Ğ¼Ğ¸ ÑÑƒÑ‰Ğ½Ğ¾ÑÑ‚ÑĞ¼Ğ¸ Ğ² Ğ´Ñ€ÑƒĞ³Ğ¸Ğµ ÑĞ»Ğ¾Ğ¸ Ğ¿Ñ€Ğ¸Ğ»Ğ¾Ğ¶ĞµĞ½Ğ¸Ñ Ğ±ĞµĞ· Ğ½ĞµĞ¾Ğ±Ñ…Ğ¾Ğ´Ğ¸Ğ¼Ğ¾ÑÑ‚Ğ¸ Ñ€ĞµÑ„Ğ°ĞºÑ‚Ğ¾Ñ€Ğ¸Ğ½Ğ³Ğ°

**ğŸ¤– ĞĞ²Ñ‚Ğ¾Ğ¼Ğ°Ñ‚Ğ¸Ğ·Ğ°Ñ†Ğ¸Ñ**

ĞĞµ Ğ¸Ğ¼Ğ¿Ğ»ĞµĞ¼ĞµĞ½Ñ‚Ğ¸Ñ€Ğ¾Ğ²Ğ°Ğ½Ğ¾ Ğ² eslint-config.

**âœ… Valid**

```
â”œâ”€â”€ Cart/
|    â”œâ”€â”€ UIStore/ 
|    |    â””â”€â”€ UIStore.ts
|    |    â””â”€â”€ index.ts
|    â”œâ”€â”€ Cart.tsx
|    â””â”€â”€ index.ts
```

**âŒ Invalid**

```
â”œâ”€â”€ CartStore/
|    â”œâ”€â”€ CartStore.ts
|    â””â”€â”€ index.ts
â”œâ”€â”€ Cart/
|    â”œâ”€â”€ Cart.tsx
|    â””â”€â”€ index.ts
```

```
â”œâ”€â”€ Cart/
|    â”œâ”€â”€ stores
|    |    â”œâ”€â”€ UIStore/
|    |    â””â”€â”€ index.ts
|    â”œâ”€â”€ Cart.tsx
|    â””â”€â”€ index.ts
```

```
â”œâ”€â”€ Cart/
|    â”œâ”€â”€ store.ts
|    â”œâ”€â”€ Cart.tsx
|    â””â”€â”€ index.ts
```

## UIStore Ğ¸ ĞµĞ³Ğ¾ Ğ´Ğ¸Ñ€ĞµĞºÑ‚Ğ¾Ñ€Ğ¸Ñ Ğ´Ğ¾Ğ»Ğ¶Ğ½Ñ‹ Ğ½Ğ°Ğ·Ñ‹Ğ²Ğ°Ñ‚ÑŒÑÑ `UIStore`

**âœ¨ ĞœĞ¾Ñ‚Ğ¸Ğ²Ğ°Ñ†Ğ¸Ñ**

Ğ˜Ğ´ĞµĞ½Ñ‚Ğ¸Ñ„Ğ¸ĞºĞ°Ñ†Ğ¸Ñ View-Model Store.

**ğŸ¤– ĞĞ²Ñ‚Ğ¾Ğ¼Ğ°Ñ‚Ğ¸Ğ·Ğ°Ñ†Ğ¸Ñ**

ĞĞµ Ğ¸Ğ¼Ğ¿Ğ»ĞµĞ¼ĞµĞ½Ñ‚Ğ¸Ñ€Ğ¾Ğ²Ğ°Ğ½Ğ¾ Ğ² eslint-config.

**âœ… Valid**

```
â”œâ”€â”€ Cart/
|    â”œâ”€â”€ UIStore/ 
|    |    â””â”€â”€ UIStore.ts
|    |    â””â”€â”€ index.ts
|    â”œâ”€â”€ Cart.tsx
|    â””â”€â”€ index.ts
```

```Cart/UIStore/UIStore.ts```
```ts
export class UIStore {};

export const createUIStore = () => new UIStore();
```

**âŒ Invalid**

```
â”œâ”€â”€ Cart/
|    â”œâ”€â”€ store/ 
|    |    â””â”€â”€ store.ts
|    |    â””â”€â”€ index.ts
|    â”œâ”€â”€ Cart.tsx
|    â””â”€â”€ index.ts
```

```Cart/store/store.ts```
```ts
export class Store {};
```

---

```
â”œâ”€â”€ Cart/
|    â”œâ”€â”€ CartStore/ 
|    |    â””â”€â”€ CartStore.ts
|    |    â””â”€â”€ index.ts
|    â”œâ”€â”€ Cart.tsx
|    â””â”€â”€ index.ts
```

```Cart/CartStore/CartStore.ts```
```ts
export class CartStore {};
```
